---
title: "metaG_in_depth"
author: "Colin G. Finlay"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document: default
  pdf_document: default
---
# Setup

```{r setup, include=FALSE}
#use to set global options for chunks e.g., echo and warning options will be applied to all chunks:
knitr::opts_chunk$set(echo = TRUE)

# Clear environment:
rm(list=ls())

# Set working directory:
setwd("~/GitHub/WetlandMesocosm_GHG_Timberlake/analyses")

# Standard error (se) and confidence interval (ci):
se <- function(x, ...){sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))}
ci <- function(x, ...){1.96 * sd(x,na.rm = TRUE)}

# Code dependencies:
source("../bin/DiversityFunctions.R")
source("../bin/MothurTools.R")
require("vegan"); require("reshape"); require("ggplot2"): require("tidyr"); require("dplyr"); require("dbstats"); require("reshape2"); require("ggpubr"); require("glue"); require("tidyverse")
```

# Load files

Import KO function files from JGI IMG/MER
```{r Import Files}
#Create vector with desired row order:
sample_order <- c("03NFC", "03PFC", "03Pre", "06NFC", "06PFC", "06Pre", "08NFC", "08PFC", "08Pre", "09NFC", "09PFC", "09Pre", "10NFC", "10PFC", "10Pre", "11NFC", "11PFC", "11Pre", "12NFC", "12PFC", "12Pre", "16NFC", "16PFC", "16Pre")
length(sample_order) #Should be 24

#Denitrification KEGG module (M00529):
denit <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/Redownloaded/CSV/Denit_CF.csv")
#Reorder rows:
denit <- denit[match(sample_order, denit$Sample), ]

#Methanogen Signature KEGG module (M00617):
meth <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/Redownloaded/CSV/Meth_CF.csv")
meth <- meth[match(sample_order, meth$Sample), ]

#All Central Carbohydrate Metabolism KEGG modules:
carb_metab <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/Redownloaded/CSV/Central_carb_metab_CF.csv")
carb_metab <- carb_metab[match(sample_order, carb_metab$Sample), ]

#Cytochrome C Oxidase KEGG module (M00155):
cytoCox <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/Redownloaded/CSV/CytCOx_CF.csv")
cytoCox <- cytoCox[match(sample_order, cytoCox$Sample), ]

#Sulfate-sulfur assimilation (M00616):
SO4_S_Assim <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/IMG_functions_in_depth/M00616_SO4_S_Assimiliation.csv")
#Reorder rows:
SO4_S_Assim <- SO4_S_Assim[match(sample_order, SO4_S_Assim$Sample), ]

#Assimilatory sulfate reduction (M00176):
Assim_SO4_red <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/IMG_functions_in_depth/M00176_Assim_SO4_reduction.csv")
#Reorder rows:
Assim_SO4_red <- Assim_SO4_red[match(sample_order, Assim_SO4_red$Sample), ]

#Dissimilatory sulfate reduction (M00596):
Dissim_SO4_red <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/IMG_functions_in_depth/M00596_Dissim_SO4_reduction.csv")
#Reorder rows:
Dissim_SO4_red <- Dissim_SO4_red[match(sample_order, Dissim_SO4_red$Sample), ]

#Thiosulfate oxidation by SOX complex (M00595):
ThioSOX <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/IMG_functions_in_depth/M00595_Thiosulfate_ox_by_SOX.csv")
#Reorder rows:
ThioSOX <- ThioSOX[match(sample_order, ThioSOX$Sample), ]

#Dissimilatory nitrate reduction (M00530):
DNRA <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/IMG_functions_in_depth/M00530_DNRA.csv")
#Reorder rows:
DNRA <- DNRA[match(sample_order, DNRA$Sample), ]

#Assimilatory nitrate reduction (M00531):
ANRA <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/IMG_functions_in_depth/M00531_ANRA.csv")
#Reorder rows:
ANRA <- ANRA[match(sample_order, ANRA$Sample), ]

#Nitrification (M00528):
Nitrific <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/IMG_functions_in_depth/M00528_Nitrification.csv")
#Reorder rows:
Nitrific <- Nitrific[match(sample_order, Nitrific$Sample), ]

#COMAMMOX (M00804):
COMAMMOX <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/IMG_functions_in_depth/M00804_COMAMMOX.csv")
#Reorder rows:
COMAMMOX <- COMAMMOX[match(sample_order, COMAMMOX$Sample), ]

#Nitrogen fixation (M00175):
Nfix <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/IMG_functions_in_depth/M00175_N_fixation.csv")
#Reorder rows:
Nfix <- Nfix[match(sample_order, Nfix$Sample), ]

#Methane oxidation (M00174):
CH4_ox <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/IMG_functions_in_depth/M00174_MethaneOxidation.csv")
#Reorder rows:
CH4_ox <- CH4_ox[match(sample_order, CH4_ox$Sample), ]

#Fe reduction (mtrBC) (TIGR03507 + TIGR03509):
mtrBC_TIGR <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/IMG_functions_in_depth/FeR_mtrBC.csv")
#Reorder rows:
mtrBC_TIGR <- mtrBC_TIGR[match(sample_order, mtrBC_TIGR$Sample), ]

#Bins, all KOs:

```

```{r Import FeGenie output}
#Gene summary:
FeSummary <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/FeGenie_meta_full/FeGenie-geneSummaryRReady.csv", header = T)

FeHeatmap <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/FeGenie_meta_full/FeGenie-heatmap-data.csv", header = T)
```

```{r Wrangle FeGenie Heatmap}
# read in metagenomic metadata from IMG, that will be used to match genome IDs to Sample IDs:
metaGmetaD <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/metaGmetaData.csv", header = T)

# transpose heatmap:
FeHeatTranspose <- as.data.frame(t(FeHeatmap))
# Fix column names on FeHeatTranspose:
colnames(FeHeatTranspose) <- FeHeatTranspose[1,]
#Now get rid of column 'X' and extra column at bottom:
FeHeatTranspose <- FeHeatTranspose[-c(1,26),]

#String manipulation; cut out extra characters to match IMG taxon oid:
taxon_oid <- as.vector(str_sub(rownames(FeHeatTranspose), start = 2, end = -7))
#Add as new column to FeHeatTranspose:
FeHeatTranspose$taxon_oid <- taxon_oid

#Merge FeHeatTranspose and metaGmetaD by taxon_oid:
FeMerged <- merge(FeHeatTranspose, metaGmetaD)

# Match sample order:
#mtrBC_TIGR <- mtrBC_TIGR[match(sample_order, mtrBC_TIGR$Sample), ]
FeMerged <- FeMerged[match(sample_order, FeMerged$Sample), ]
```


```{r Import metadata and greenhouse gas data}
#Design file:
design <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/design.csv")
design2 <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/tmg_design.csv")
#Reorder rows of design:
design <- design[match(sample_order, design$Sample), ]

#Remove mock, and box samples from design2. 
rownames(design2) <- design2$sample
design2 <- design2[!(rownames(design2) %in% "MOCK"),]
design2 <- design2[!(design2$source %in% "b"),]
design2 <- design2[!(design2$source %in% "s"),] # Consider using start samples later (16S data?)

#Bin genome IDs:
Bin_genome_IDs <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/Bins/Bin_Genome_IDs.csv")

#Greenhouse gas data:
GHG <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/TMG_GHG-Flux_T1_Pub_2-2020.csv")

# Import soil physiochemical data:
soil <- read.csv("../data/Design_and_soil.csv",  header=T)
```

```{r Playing around with a list of dfs for control structures}
# Make a list of all the metaG df names:
metaG_string_list <- c("denit", "meth", "carb_metab", "cytoCox", "SO4_S_Assim", "Assim_SO4_red", "Dissim_SO4_red", "ThioSOX", "DNRA", "ANRA", "Nitrific", "COMAMMOX", "Nfix", "CH4_ox", "mtrBC_TIGR")

for (i in metaG_string_list) {
  print(i)
}

# Make names for numerical only data frames:
for (i in metaG_string_list) {
  print(paste(i, "Numerical", sep = ""))
}

# Later, you'll want to print the colnames() (KOs) of each metaG df to use for indexing Bins-all-KOs data
```

# Data Wrangling

```{r soil wrangling}
# Start and finish samples, but no field samples (SSF = soil.start.finish)
soil.start.finish <- soil %>%
  filter(time!="field")

SSF.by_time_plant_treatment <- soil.start.finish %>%
  dplyr::select(chamber, time, history, treatment, plant, nh4.mg.l, redox.percent.paint.removed, moisture.percent) %>%
  group_by(chamber)

# No field or start samples (SFO = soil.finishOnly)
soil.finishONLY <- soil %>%
  filter(time!="field")%>%
  filter(time!="start")

SFO.by_time_plant_treatment <- soil.finishONLY %>%
  dplyr::select(-c(g.soil.kcl, no2.no3.mg.l, plant.c.percent, plant.n.percent, plant.dry.mass.g, CEC., HM.)) %>%
  group_by(chamber)
```

```{r Sample IDs to rownames}
#For metagenomic data, make sample IDs rownames in a new data frame, to be further subsetted to numeric only data:

denitNumerical <- denit
rownames(denitNumerical) <- denitNumerical$Sample

methNumerical <- meth
rownames(methNumerical) <- methNumerical$Sample

carb_metabNumerical <- carb_metab
rownames(carb_metabNumerical) <- carb_metabNumerical$Sample

cytoCoxNumerical <- cytoCox
rownames(cytoCoxNumerical) <- cytoCoxNumerical$Sample

SO4_S_AssimNumerical <- SO4_S_Assim
rownames(SO4_S_AssimNumerical) <- SO4_S_AssimNumerical$Sample

Assim_SO4_redNumerical <- Assim_SO4_red
rownames(Assim_SO4_redNumerical) <- Assim_SO4_redNumerical$Sample

Dissim_SO4_redNumerical <- Dissim_SO4_red
rownames(Dissim_SO4_redNumerical) <- Dissim_SO4_redNumerical$Sample

ThioSOXNumerical <- ThioSOX
rownames(ThioSOXNumerical) <- ThioSOXNumerical$Sample

DNRANumerical <- DNRA
rownames(DNRANumerical) <- DNRANumerical$Sample

ANRANumerical <- ANRA
rownames(ANRANumerical) <- ANRANumerical$Sample

NitrificNumerical <- Nitrific
rownames(NitrificNumerical) <- NitrificNumerical$Sample

COMAMMOXNumerical <- COMAMMOX
rownames(COMAMMOXNumerical) <- COMAMMOXNumerical$Sample

NfixNumerical <- Nfix
rownames(NfixNumerical) <- NfixNumerical$Sample

CH4_oxNumerical <- CH4_ox
rownames(CH4_oxNumerical) <- CH4_oxNumerical$Sample

mtrBC_TIGRNumerical <- mtrBC_TIGR
rownames(mtrBC_TIGRNumerical) <- mtrBC_TIGRNumerical$Sample

FeGenieNumerical <- FeMerged
rownames(FeGenieNumerical) <- FeGenieNumerical$Sample 
```

```{r make numeric only data frames for functional genes}
denitNumerical <- denitNumerical[,-(1:2)]

methNumerical <- methNumerical[,-(1:2)]

carb_metabNumerical <- carb_metabNumerical[,-(1:2)]

cytoCoxNumerical <- cytoCoxNumerical[,-(1:2)]

SO4_S_AssimNumerical <- SO4_S_AssimNumerical[,-(1:2)]

Assim_SO4_redNumerical <- Assim_SO4_redNumerical[,-(1:2)]

Dissim_SO4_redNumerical <- Dissim_SO4_redNumerical[,-(1:2)]

ThioSOXNumerical <- ThioSOXNumerical[,-(1:2)]

DNRANumerical <- DNRANumerical[,-(1:2)]

ANRANumerical <- ANRANumerical[,-(1:2)]

NitrificNumerical <- NitrificNumerical[,-(1:2)]

COMAMMOXNumerical <- COMAMMOXNumerical[,-(1:2)]

NfixNumerical <- NfixNumerical[,-(1:2)]

CH4_oxNumerical <- CH4_oxNumerical[,-(1:2)]

mtrBC_TIGRNumerical <- mtrBC_TIGRNumerical[,-(1:2)]

FeGenieNumerical <- FeGenieNumerical[, 2:14]
# Must convert characters to numeric for all the data from FeGenie:
#(using a brute force method for now, because struggling to find something more efficient)
FeGenieNumerical$`iron_aquisition-iron_transport` <- as.numeric(FeGenieNumerical$`iron_aquisition-iron_transport`)
FeGenieNumerical$`iron_aquisition-heme_transport` <- as.numeric(FeGenieNumerical$`iron_aquisition-heme_transport`)
FeGenieNumerical$`iron_aquisition-heme_oxygenase` <- as.numeric(FeGenieNumerical$`iron_aquisition-heme_oxygenase`)
FeGenieNumerical$`iron_aquisition-siderophore_synthesis` <- as.numeric(FeGenieNumerical$`iron_aquisition-siderophore_synthesis`)
FeGenieNumerical$`iron_aquisition-siderophore_transport` <- as.numeric(FeGenieNumerical$`iron_aquisition-siderophore_transport`)
FeGenieNumerical$`iron_aquisition-siderophore_transport_potential` <- as.numeric(FeGenieNumerical$`iron_aquisition-siderophore_transport_potential`)
FeGenieNumerical$iron_gene_regulation <- as.numeric(FeGenieNumerical$iron_gene_regulation)
FeGenieNumerical$iron_oxidation <- as.numeric(FeGenieNumerical$iron_oxidation)
FeGenieNumerical$possible_iron_oxidation_and_possible_iron_reduction <- as.numeric(FeGenieNumerical$possible_iron_oxidation_and_possible_iron_reduction)
FeGenieNumerical$probable_iron_reduction <- as.numeric(FeGenieNumerical$probable_iron_reduction)
FeGenieNumerical$iron_reduction <- as.numeric(FeGenieNumerical$iron_reduction)
FeGenieNumerical$iron_storage <- as.numeric(FeGenieNumerical$iron_storage)
FeGenieNumerical$magnetosome_formation <- as.numeric(FeGenieNumerical$magnetosome_formation)
```

```{r for loop to generate text for RA calculations}

for (i in metaG_string_list) {
  cat(i, "RA <- ", i,"Numerical", sep = "")
  cat("\nfor(i in 1:dim(", i, "Numerical)[1]){\n  ", i, "RA[i,] <- ", i, "Numerical[i,]/sum(", i, "Numerical[i,])\n}\n\n", sep = "")
}

# Output of loop to write following chunk
```

```{r Convert gene counts to relative abundance}
#Convert denitrification KO counts to relative abundance:
denitRA <- denitNumerical
for(i in 1:dim(denitNumerical)[1]){
  denitRA[i,] <- denitNumerical[i,]/sum(denitNumerical[i,])
}

#Convert methanogen KO counts to relative abundance:
methRA <- methNumerical
for(i in 1:dim(methNumerical)[1]){
  methRA[i,] <- methNumerical[i,]/sum(methNumerical[i,])
}

#Convert central carbohydrate metabolism KO counts to relative abundance:
carb_metabRA <- carb_metabNumerical
for(i in 1:dim(carb_metabNumerical)[1]){
  carb_metabRA[i,] <- carb_metabNumerical[i,]/sum(carb_metabNumerical[i,])
}

#Convert cytochrome C oxidase KO counts to relative abundance:
cytoCoxRA <- cytoCoxNumerical
for(i in 1:dim(cytoCoxNumerical)[1]){
  cytoCoxRA[i,] <- cytoCoxNumerical[i,]/sum(cytoCoxNumerical[i,])
}

SO4_S_AssimRA <- SO4_S_AssimNumerical
for(i in 1:dim(SO4_S_AssimNumerical)[1]){
  SO4_S_AssimRA[i,] <- SO4_S_AssimNumerical[i,]/sum(SO4_S_AssimNumerical[i,])
}

Assim_SO4_redRA <- Assim_SO4_redNumerical
for(i in 1:dim(Assim_SO4_redNumerical)[1]){
  Assim_SO4_redRA[i,] <- Assim_SO4_redNumerical[i,]/sum(Assim_SO4_redNumerical[i,])
}

Dissim_SO4_redRA <- Dissim_SO4_redNumerical
for(i in 1:dim(Dissim_SO4_redNumerical)[1]){
  Dissim_SO4_redRA[i,] <- Dissim_SO4_redNumerical[i,]/sum(Dissim_SO4_redNumerical[i,])
}

ThioSOXRA <- ThioSOXNumerical
for(i in 1:dim(ThioSOXNumerical)[1]){
  ThioSOXRA[i,] <- ThioSOXNumerical[i,]/sum(ThioSOXNumerical[i,])
}

DNRARA <- DNRANumerical
for(i in 1:dim(DNRANumerical)[1]){
  DNRARA[i,] <- DNRANumerical[i,]/sum(DNRANumerical[i,])
}

ANRARA <- ANRANumerical
for(i in 1:dim(ANRANumerical)[1]){
  ANRARA[i,] <- ANRANumerical[i,]/sum(ANRANumerical[i,])
}

NitrificRA <- NitrificNumerical
for(i in 1:dim(NitrificNumerical)[1]){
  NitrificRA[i,] <- NitrificNumerical[i,]/sum(NitrificNumerical[i,])
}

COMAMMOXRA <- COMAMMOXNumerical
for(i in 1:dim(COMAMMOXNumerical)[1]){
  COMAMMOXRA[i,] <- COMAMMOXNumerical[i,]/sum(COMAMMOXNumerical[i,])
}

NfixRA <- NfixNumerical
for(i in 1:dim(NfixNumerical)[1]){
  NfixRA[i,] <- NfixNumerical[i,]/sum(NfixNumerical[i,])
}

CH4_oxRA <- CH4_oxNumerical
for(i in 1:dim(CH4_oxNumerical)[1]){
  CH4_oxRA[i,] <- CH4_oxNumerical[i,]/sum(CH4_oxNumerical[i,])
}

mtrBC_TIGRRA <- mtrBC_TIGRNumerical
for(i in 1:dim(mtrBC_TIGRNumerical)[1]){
  mtrBC_TIGRRA[i,] <- mtrBC_TIGRNumerical[i,]/sum(mtrBC_TIGRNumerical[i,])
}

FeGenieRA <- FeGenieNumerical
for(i in 1:dim(FeGenieNumerical)[1]){
  FeGenieRA[i,] <- FeGenieNumerical[i,]/sum(FeGenieNumerical[i,])
}
```

```{r lists of start and finish samples}
#Make a list for start and finish, so you don't have to re-write it each time below
start_metaG_samples <- c("03Pre", "06Pre", "08Pre", "09Pre", "10Pre", "11Pre", "12Pre", "16Pre")
finish_metaG_samples <- c("03NFC", "03PFC", "06NFC", "06PFC", "08NFC", "08PFC", "09NFC", "09PFC", "10NFC", "10PFC", "11NFC", "11PFC", "12NFC", "12PFC", "16NFC", "16PFC")
```

```{r for loop to create start and finish subsets}
for (i in metaG_string_list) {
  cat(i, "RA_start <- ", i,"RA[start_metaG_samples, ]\n", sep = "")
  cat(i, "RA_finish <- ", i,"RA[finish_metaG_samples, ]\n\n", sep = "")
}
```

```{r Subset start and finish samples from all gene sets}
denitRA_start <- denitRA[start_metaG_samples, ]
denitRA_finish <- denitRA[finish_metaG_samples, ]

methRA_start <- methRA[start_metaG_samples, ]
methRA_finish <- methRA[finish_metaG_samples, ]

carb_metabRA_start <- carb_metabRA[start_metaG_samples, ]
carb_metabRA_finish <- carb_metabRA[finish_metaG_samples, ]

cytoCoxRA_start <- cytoCoxRA[start_metaG_samples, ]
cytoCoxRA_finish <- cytoCoxRA[finish_metaG_samples, ]

SO4_S_AssimRA_start <- SO4_S_AssimRA[start_metaG_samples, ]
SO4_S_AssimRA_finish <- SO4_S_AssimRA[finish_metaG_samples, ]

Assim_SO4_redRA_start <- Assim_SO4_redRA[start_metaG_samples, ]
Assim_SO4_redRA_finish <- Assim_SO4_redRA[finish_metaG_samples, ]

Dissim_SO4_redRA_start <- Dissim_SO4_redRA[start_metaG_samples, ]
Dissim_SO4_redRA_finish <- Dissim_SO4_redRA[finish_metaG_samples, ]

ThioSOXRA_start <- ThioSOXRA[start_metaG_samples, ]
ThioSOXRA_finish <- ThioSOXRA[finish_metaG_samples, ]

DNRARA_start <- DNRARA[start_metaG_samples, ]
DNRARA_finish <- DNRARA[finish_metaG_samples, ]

ANRARA_start <- ANRARA[start_metaG_samples, ]
ANRARA_finish <- ANRARA[finish_metaG_samples, ]

NitrificRA_start <- NitrificRA[start_metaG_samples, ]
NitrificRA_finish <- NitrificRA[finish_metaG_samples, ]

COMAMMOXRA_start <- COMAMMOXRA[start_metaG_samples, ]
COMAMMOXRA_finish <- COMAMMOXRA[finish_metaG_samples, ]

NfixRA_start <- NfixRA[start_metaG_samples, ]
NfixRA_finish <- NfixRA[finish_metaG_samples, ]

CH4_oxRA_start <- CH4_oxRA[start_metaG_samples, ]
CH4_oxRA_finish <- CH4_oxRA[finish_metaG_samples, ]

mtrBC_TIGRRA_start <- mtrBC_TIGRRA[start_metaG_samples, ]
mtrBC_TIGRRA_finish <- mtrBC_TIGRRA[finish_metaG_samples, ]

FeGenieRA_start <- FeGenieRA[start_metaG_samples,]
FeGenieRA_finish <- FeGenieRA[finish_metaG_samples,]
```

```{r for loop to generate dimension checks}
# Relative abundance (RA) dimension check:
for (i in metaG_string_list) {
  cat("dim(",i, "RA)\n", sep = "")
}

# Start samples only (RA_start) dimension check:
for (i in metaG_string_list) {
  cat("dim(",i, "RA_start)\n", sep = "")
}

# Finish samples only (RA_finish) dimension check:
for (i in metaG_string_list) {
  cat("dim(",i, "RA_finish)\n", sep = "")
}
```

```{r Dimension check for RA matrices}
#Check dimensions of each relative abundance (RA) matrix:
dim(denitRA) #24 10
dim(methRA) #24 44
dim(carb_metabRA) #24 120
dim(cytoCoxRA) #24 5
dim(SO4_S_AssimRA) #24 14
dim(Assim_SO4_redRA) #24 10
dim(Dissim_SO4_redRA) #24 5
dim(ThioSOXRA) #24 7
dim(DNRARA) #24 9
dim(ANRARA) #24 6
dim(NitrificRA) #24 4
dim(COMAMMOXRA) #24 6
dim(NfixRA) #24 8
dim(CH4_oxRA) #24 12
dim(mtrBC_TIGRRA) #24 2
dim(FeGenieRA) #24 13

#Check dimensions of RA start matrices
dim(denitRA_start) #8 10
dim(methRA_start) #8 44
dim(carb_metabRA_start) #8 120
dim(cytoCoxRA_start) #8 5
dim(SO4_S_AssimRA_start) #8 14
dim(Assim_SO4_redRA_start) #8 10
dim(Dissim_SO4_redRA_start) #8 5
dim(ThioSOXRA_start) #8 7
dim(DNRARA_start) #8 9
dim(ANRARA_start) #8 6
dim(NitrificRA_start) #8 4
dim(COMAMMOXRA_start) #8 6
dim(NfixRA_start) #8 8
dim(CH4_oxRA_start) #8 12
dim(mtrBC_TIGRRA_start) #8 2
dim(FeGenieRA_start) #8 13

#Check dimensions of RA finish matrices
dim(denitRA_finish) #16 10
dim(methRA_finish) #16 44
dim(carb_metabRA_finish) #16 120
dim(cytoCoxRA_finish) #16 5
dim(SO4_S_AssimRA_finish) #16 14
dim(Assim_SO4_redRA_finish) #16 10
dim(Dissim_SO4_redRA_finish) #16 5
dim(ThioSOXRA_finish) #16 7
dim(DNRARA_finish) #16 9
dim(ANRARA_finish) #16 6
dim(NitrificRA_finish) #16 4
dim(COMAMMOXRA_finish) #16 6
dim(NfixRA_finish) #16 8
dim(CH4_oxRA_finish) #16 12
dim(mtrBC_TIGRRA_finish) #16 2
dim(FeGenieRA_finish) #16 13
```

```{r modify 'design' rownames}
#Make metaG sample ID's the rownames on 'design'. Then drop design$Sample:
rownames(design) <- design$Sample
design <- design[,2:4]
```

```{r for loop for RA & design combination}
for (i in metaG_string_list) {
  cat(i, "RA_design <- cbind(design, ", i, "RA)\n", sep = "")
}
```

```{r combine RA and design data frames}
# Make data frames from gene set relative abundance (RA) data frames and design
denitRA_design <- cbind(design, denitRA)
methRA_design <- cbind(design, methRA)
carb_metabRA_design <- cbind(design, carb_metabRA)
cytoCoxRA_design <- cbind(design, cytoCoxRA)
SO4_S_AssimRA_design <- cbind(design, SO4_S_AssimRA)
Assim_SO4_redRA_design <- cbind(design, Assim_SO4_redRA)
Dissim_SO4_redRA_design <- cbind(design, Dissim_SO4_redRA)
ThioSOXRA_design <- cbind(design, ThioSOXRA)
DNRARA_design <- cbind(design, DNRARA)
ANRARA_design <- cbind(design, ANRARA)
NitrificRA_design <- cbind(design, NitrificRA)
COMAMMOXRA_design <- cbind(design, COMAMMOXRA)
NfixRA_design <- cbind(design, NfixRA)
CH4_oxRA_design <- cbind(design, CH4_oxRA)
mtrBC_TIGRRA_design <- cbind(design, mtrBC_TIGRRA)
FeGenieRA_design <- cbind(design, FeGenieRA)
```

## Bray Curtis dissimilarity

```{r for loop for BC calculation}
# Combined loop:
for (i in metaG_string_list) {
  cat(i, "BC <- vegdist(", i, "RA, method = 'bray')\n", sep = "")
  cat(i, "BC_start <- vegdist(", i, "RA_start, method = 'bray')\n", sep = "")
  cat(i, "BC_finish <- vegdist(", i, "RA_finish, method = 'bray')\n\n", sep = "")
}
```

Calculate Bray Curtis dissimilarity:
```{r relative abundance to Bray Curtis dissimilarity}
denitBC <- vegdist(denitRA, method = 'bray')
denitBC_start <- vegdist(denitRA_start, method = 'bray')
denitBC_finish <- vegdist(denitRA_finish, method = 'bray')

methBC <- vegdist(methRA, method = 'bray')
methBC_start <- vegdist(methRA_start, method = 'bray')
methBC_finish <- vegdist(methRA_finish, method = 'bray')

carb_metabBC <- vegdist(carb_metabRA, method = 'bray')
carb_metabBC_start <- vegdist(carb_metabRA_start, method = 'bray')
carb_metabBC_finish <- vegdist(carb_metabRA_finish, method = 'bray')

cytoCoxBC <- vegdist(cytoCoxRA, method = 'bray')
cytoCoxBC_start <- vegdist(cytoCoxRA_start, method = 'bray')
cytoCoxBC_finish <- vegdist(cytoCoxRA_finish, method = 'bray')

SO4_S_AssimBC <- vegdist(SO4_S_AssimRA, method = 'bray')
SO4_S_AssimBC_start <- vegdist(SO4_S_AssimRA_start, method = 'bray')
SO4_S_AssimBC_finish <- vegdist(SO4_S_AssimRA_finish, method = 'bray')

Assim_SO4_redBC <- vegdist(Assim_SO4_redRA, method = 'bray')
Assim_SO4_redBC_start <- vegdist(Assim_SO4_redRA_start, method = 'bray')
Assim_SO4_redBC_finish <- vegdist(Assim_SO4_redRA_finish, method = 'bray')

Dissim_SO4_redBC <- vegdist(Dissim_SO4_redRA, method = 'bray')
Dissim_SO4_redBC_start <- vegdist(Dissim_SO4_redRA_start, method = 'bray')
Dissim_SO4_redBC_finish <- vegdist(Dissim_SO4_redRA_finish, method = 'bray')

ThioSOXBC <- vegdist(ThioSOXRA, method = 'bray')
ThioSOXBC_start <- vegdist(ThioSOXRA_start, method = 'bray')
ThioSOXBC_finish <- vegdist(ThioSOXRA_finish, method = 'bray')

DNRABC <- vegdist(DNRARA, method = 'bray')
DNRABC_start <- vegdist(DNRARA_start, method = 'bray')
DNRABC_finish <- vegdist(DNRARA_finish, method = 'bray')

ANRABC <- vegdist(ANRARA, method = 'bray')
ANRABC_start <- vegdist(ANRARA_start, method = 'bray')
ANRABC_finish <- vegdist(ANRARA_finish, method = 'bray')

NitrificBC <- vegdist(NitrificRA, method = 'bray')
NitrificBC_start <- vegdist(NitrificRA_start, method = 'bray')
NitrificBC_finish <- vegdist(NitrificRA_finish, method = 'bray')

COMAMMOXBC <- vegdist(COMAMMOXRA, method = 'bray')
COMAMMOXBC_start <- vegdist(COMAMMOXRA_start, method = 'bray')
COMAMMOXBC_finish <- vegdist(COMAMMOXRA_finish, method = 'bray')

NfixBC <- vegdist(NfixRA, method = 'bray')
NfixBC_start <- vegdist(NfixRA_start, method = 'bray')
NfixBC_finish <- vegdist(NfixRA_finish, method = 'bray')

CH4_oxBC <- vegdist(CH4_oxRA, method = 'bray')
CH4_oxBC_start <- vegdist(CH4_oxRA_start, method = 'bray')
CH4_oxBC_finish <- vegdist(CH4_oxRA_finish, method = 'bray')

mtrBC_TIGRBC <- vegdist(mtrBC_TIGRRA, method = 'bray')
mtrBC_TIGRBC_start <- vegdist(mtrBC_TIGRRA_start, method = 'bray')
mtrBC_TIGRBC_finish <- vegdist(mtrBC_TIGRRA_finish, method = 'bray')

FeGenieBC <- vegdist(FeGenieRA, method = 'bray')
FeGenieBC_start <- vegdist(FeGenieRA_start, method = 'bray')
FeGenieBC_finish <- vegdist(FeGenieRA_finish, method = 'bray')
```

## GHG wrangling

Wrangling greenhouse gas (GHG) data:
```{r wrangling GHG data}
#Create a new GHG data frame with less columns:
GHG_less_columns <- cbind(GHG[,1:4], GHG[24:32])

#Drop rows with NAs (non-T1 rows):
GHG_less_columns <- GHG_less_columns %>% drop_na()

#Change column name "chamber" to "box" to match with design2:
colnames(GHG_less_columns)[2] <- "box"

#Convert GHG_less_columns$plant to lowercase to match with design2$plant:
GHG_less_columns$plant <- tolower(GHG_less_columns$plant)

#Subset GHG_less_columns for 'start' time point (6/13/2016) and 'finish' time point (8/11/2016):
GHG_start <- GHG_less_columns[1:36,]
GHG_finish <- GHG_less_columns[157:192,]

#Add design2 into GHG 'start' and 'finish':
GHG_start <- left_join(GHG_start, design2, by = c("box", "plant"))
GHG_finish <- left_join(GHG_finish, design2, by = c("box", "plant"))

#Subset GHG_finish and GHG_start to only have metaG sample numbers:
GHG_start_metaG <- rbind(GHG_start[5:6,], GHG_start[11:12,], GHG_start[15:24,], GHG_start[31:32,], make.row.names = FALSE)
GHG_finish_metaG <- rbind(GHG_finish[5:6,], GHG_finish[11:12,], GHG_finish[15:24,], GHG_finish[31:32,], make.row.names = FALSE)

#To align "pre" metaG samples with a single, initial GHG value (from the start GHG measurements on 6/13/206), average together GHG samples in the same box number, then transpose and set row name to corresponding metaG Sample ID:
GHG_start_03 <- data.frame(colMeans(GHG_start_metaG[1:2, 5:13]))
GHG_start_03 <- as.data.frame(t(GHG_start_03))
row.names(GHG_start_03) <- "03Pre"

GHG_start_06 <- data.frame(colMeans(GHG_start_metaG[3:4, 5:13]))
GHG_start_06 <- as.data.frame(t(GHG_start_06))
row.names(GHG_start_06) <- "06Pre"

GHG_start_08 <- data.frame(colMeans(GHG_start_metaG[5:6, 5:13]))
GHG_start_08 <- as.data.frame(t(GHG_start_08))
row.names(GHG_start_08) <- "08Pre"

GHG_start_09 <- data.frame(colMeans(GHG_start_metaG[7:8, 5:13]))
GHG_start_09 <- as.data.frame(t(GHG_start_09))
row.names(GHG_start_09) <- "09Pre"

GHG_start_10 <- data.frame(colMeans(GHG_start_metaG[9:10, 5:13]))
GHG_start_10 <- as.data.frame(t(GHG_start_10))
row.names(GHG_start_10) <- "10Pre"

GHG_start_11 <- data.frame(colMeans(GHG_start_metaG[11:12, 5:13]))
GHG_start_11 <- as.data.frame(t(GHG_start_11))
row.names(GHG_start_11) <- "11Pre"

GHG_start_12 <- data.frame(colMeans(GHG_start_metaG[13:14, 5:13]))
GHG_start_12 <- as.data.frame(t(GHG_start_12))
row.names(GHG_start_12) <- "12Pre"

GHG_start_16 <- data.frame(colMeans(GHG_start_metaG[15:16, 5:13]))
GHG_start_16 <- as.data.frame(t(GHG_start_16))
row.names(GHG_start_16) <- "16Pre"

#Then rbind() all the dfs together
GHG_start_metaG_avg <- rbind(GHG_start_03, GHG_start_06, GHG_start_08, GHG_start_09, GHG_start_10, GHG_start_11, GHG_start_12, GHG_start_16)

#Set the row names of GHG data to the metaG sample names, so that stats functions can compare the GHG and metaG data sets:
row.names(GHG_finish_metaG) <- c("03NFC", "03PFC", "06NFC", "06PFC", "08NFC", "08PFC", "09NFC","09PFC", "10NFC","10PFC", "11NFC", "11PFC", "12NFC", "12PFC", "16NFC", "16PFC")

#Combine GHG_start_metaG_avg and GHG_finish_metaG[,5:13] to have all 24 samples:
GHG_metaG <- rbind(GHG_start_metaG_avg, GHG_finish_metaG[,5:13])
GHG_metaG <- GHG_metaG[match(sample_order, rownames(GHG_metaG)), ]
GHG_metaG <- GHG_metaG[,c(2,5,8)] #Subset to only have units in mg

#Clean up GHG_finish_metaG for envfitting:
GHG_finish_envfit <- GHG_finish_metaG %>%
  dplyr::select(-c(date, timepoint, ug.ch4c.m2, g.CH4.N.m.2, ug.co2c.m2, g.CO2.N.m.2, ug.n2on.m2, g.N2O.N.m.2, source, ttreat, hhistory))
```

```{r Combined GHG and Soil FINISH data frames for envfit}
#Subset of SFO.by_time_plant_treatment that is represented in GHG_finish_envfit:
SFO.envfit <- SFO.by_time_plant_treatment %>%
  dplyr::filter(chamber %in% GHG_finish_envfit$sample)

# Change 'box' to 'chamber' to enable merging:
colnames(GHG_finish_envfit)[6] <- "chamber"

# Match lowercase to enable merging
SFO.envfit$plant <- tolower(SFO.envfit$plant)
SFO.envfit$history <- tolower(SFO.envfit$history)
SFO.envfit$treatment <- tolower(SFO.envfit$treatment)
GHG_finish_envfit$treatment <- tolower(GHG_finish_envfit$treatment)
GHG_finish_envfit$history <- tolower(GHG_finish_envfit$history)

# Merge soil and greenhouse gas data:
envfit_FINISH <- merge(GHG_finish_envfit, SFO.envfit)

# Order by chamber, to match with metaG/GHG sample order:
envfit_FINISH <- envfit_FINISH[order(envfit_FINISH$chamber),]

# Set rownames to metaG/GHG rownames; this will be required for envfit
rownames(envfit_FINISH) <- rownames(GHG_finish_envfit)

# Now subset to only numbers:
envfit_FINISH_numerical <- envfit_FINISH %>%
  dplyr::select(where(is.numeric))
```

```{r Combine GHG and Soil START&FINISH for envfit}
#Filter SSF to only get GHG samples ("boxes")
# Using a brute-force approach:
SSF.2.0 <- SSF.by_time_plant_treatment[-c(1:4, 7:10, 13:14, 25:30, 33:40, 43:46, 49:50, 61:66, 69:72),]

# Need to average together 'start' samples again to make 'pre'. Use GHG code:
SSF_start_03 <- data.frame(colMeans(SSF.2.0[17:18, 6:8]))
SSF_start_03 <- as.data.frame(t(SSF_start_03))
row.names(SSF_start_03) <- "03Pre"

SSF_start_06 <- data.frame(colMeans(SSF.2.0[19:20, 6:8]))
SSF_start_06 <- as.data.frame(t(SSF_start_06))
row.names(SSF_start_06) <- "06Pre"

SSF_start_08 <- data.frame(colMeans(SSF.2.0[21:22, 6:8]))
SSF_start_08 <- as.data.frame(t(SSF_start_08))
row.names(SSF_start_08) <- "08Pre"

SSF_start_09 <- data.frame(colMeans(SSF.2.0[23:24, 6:8]))
SSF_start_09 <- as.data.frame(t(SSF_start_09))
row.names(SSF_start_09) <- "09Pre"

SSF_start_10 <- data.frame(colMeans(SSF.2.0[25:26, 6:8]))
SSF_start_10 <- as.data.frame(t(SSF_start_10))
row.names(SSF_start_10) <- "10Pre"

SSF_start_11 <- data.frame(colMeans(SSF.2.0[27:28, 6:8]))
SSF_start_11 <- as.data.frame(t(SSF_start_11))
row.names(SSF_start_11) <- "11Pre"

SSF_start_12 <- data.frame(colMeans(SSF.2.0[29:30, 6:8]))
SSF_start_12 <- as.data.frame(t(SSF_start_12))
row.names(SSF_start_12) <- "12Pre"

SSF_start_16 <- data.frame(colMeans(SSF.2.0[31:32, 6:8]))
SSF_start_16 <- as.data.frame(t(SSF_start_16))
row.names(SSF_start_16) <- "16Pre"

#Then rbind() all the dfs together
SSF_start_metaG_avg <- rbind(SSF_start_03, SSF_start_06, SSF_start_08, SSF_start_09, SSF_start_10, SSF_start_11, SSF_start_12, SSF_start_16)

# Subset Finish samples from SSF.2.0:
SSF.2.0.FINISH <- as.data.frame(SSF.2.0[1:16, ])

# Add column to SSF.2.0.FINISH with the metaG sample IDs:
SSF.2.0.FINISH$sample <- c("03NFC", "03PFC", "06NFC", "06PFC", "08NFC", "08PFC", "09NFC","09PFC", "10NFC","10PFC", "11NFC", "11PFC", "12NFC", "12PFC", "16NFC", "16PFC")

# Make $sample rownames:
rownames(SSF.2.0.FINISH) <- SSF.2.0.FINISH$sample

# Subset SSF.2.0.FINISH to the same dimensions as SSF_start_metaG_avg:
SSF.FINISH.forMerging <- as.data.frame(SSF.2.0.FINISH[,6:8])
rownames(SSF.FINISH.forMerging) <- SSF.2.0.FINISH$sample

#Combine GHG_start_metaG_avg and GHG_finish_metaG[,5:13] to have all 24 samples:
Soil_StartFinish_forEnvfit <- rbind(SSF_start_metaG_avg, SSF.FINISH.forMerging)
Soil_StartFinish_forEnvfit <- Soil_StartFinish_forEnvfit[match(sample_order, rownames(Soil_StartFinish_forEnvfit)), ]

# combine GHG_metaG and SSF
SoilGHG_StartFinish_Envfit <- cbind(GHG_metaG, Soil_StartFinish_forEnvfit)
```


# Statistics

## dbplsr

Distance-based partial least squares regression:
```{r distance-based partial least squares regression}
start.denit.dbplsr <- dbplsr(GHG_start_metaG_avg$mg.N2O.N.m.2 ~ as.matrix(denitBC_start), ncomp = 7, method = "GCV")
summary(start.denit.dbplsr)
#plot(start.denit.dbplsr)

finish.denit.dbplsr <- dbplsr(GHG_finish_metaG$mg.N2O.N.m.2 ~ as.matrix(denitBC_finish), ncomp = 15, method = "GCV")
summary(finish.denit.dbplsr)
#plot(finish.denit.dbplsr)

start.meth.dbplsr <- dbplsr(GHG_start_metaG_avg$mg.CH4.N.m.2 ~ as.matrix(methBC_start), ncomp = 7, method = "GCV")
summary(start.meth.dbplsr)
#plot(start.meth.dbplsr)

finish.meth.dbplsr <- dbplsr(GHG_finish_metaG$mg.CH4.N.m.2 ~ as.matrix(methBC_finish), ncomp = 15, method = "GCV")
summary(finish.meth.dbplsr)
#plot(finish.meth.dbplsr)

start.carb_metab.dbplsr <- dbplsr(GHG_start_metaG_avg$mg.CO2.N.m.2 ~ as.matrix(carb_metabBC_start), ncomp = 7, method = "GCV")
summary(start.carb_metab.dbplsr)
#plot(start.carb_metab.dbplsr)

finish.carb_metab.dbplsr <- dbplsr(GHG_finish_metaG$mg.CO2.N.m.2 ~ as.matrix(carb_metabBC_finish), ncomp = 15, method = "GCV")
summary(finish.carb_metab.dbplsr)
#plot(finish.carb_metab.dbplsr)

start.cytoCox.dbplsr <- dbplsr(GHG_start_metaG_avg$mg.CO2.N.m.2 ~ as.matrix(cytoCoxBC_start), ncomp = 7, method = "GCV")
summary(start.cytoCox.dbplsr)
#plot(start.cytoCox.dbplsr)

finish.cytoCox.dbplsr <- dbplsr(GHG_finish_metaG$mg.CO2.N.m.2 ~ as.matrix(cytoCoxBC_finish), ncomp = 15, method = "GCV")
summary(finish.cytoCox.dbplsr)
#plot(finish.cytoCox.dbplsr)

# Because of envfit() results showing significant relationship between denitrification composition and [CO2], running dbplsrs for denit and CO2:
start.denit.dbplsr.co2 <- dbplsr(GHG_start_metaG_avg$mg.CO2.N.m.2 ~ as.matrix(denitBC_start), ncomp = 7, method = "GCV")
summary(start.denit.dbplsr.co2)
#plot(start.denit.dbplsr.co2)

finish.denit.dbplsr.co2 <- dbplsr(GHG_finish_metaG$mg.CO2.N.m.2 ~ as.matrix(denitBC_finish), ncomp = 15, method = "GCV")
summary(finish.denit.dbplsr.co2)
#plot(finish.denit.dbplsr.co2)
```

```{r dbplsr continued}

```


## PERMANOVA

Permutational multivariate analysis of variance (PERMANOVA):
```{r PERMANOVA without History}
#PERMANOVA without History, used early on in analysis, but now performing PERMANOVAS with history as default:
adonis_denit = adonis2(denitRA_design[,-c(1:3)]~Plant*Treat, method = "bray", data = denitRA_design, perm=10000, set.seed=42)
adonis_denit

adonis_meth = adonis2(methRA_design[,-c(1:3)]~Plant*Treat, method = "bray", data = methRA_design, perm=10000, set.seed=42)
adonis_meth

adonis_carb_metab = adonis2(carb_metabRA_design[,-c(1:3)]~Plant*Treat, method = "bray", data = carb_metabRA_design, perm=10000, set.seed=42)
adonis_carb_metab

adonis_cytoCox = adonis2(cytoCoxRA_design[,-c(1:3)]~Plant*Treat, method = "bray", data = cytoCoxRA_design, perm=10000, set.seed=42)
adonis_cytoCox
```

```{r PERMANOVA with History}
#PERMANOVA with History:
adonis_denit_H = adonis2(denitRA_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = denitRA_design, perm=10000, set.seed=42)
adonis_denit_H

adonis_meth_H = adonis2(methRA_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = methRA_design, perm=10000, set.seed=42)
adonis_meth_H

adonis_carb_metab_H = adonis2(carb_metabRA_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = carb_metabRA_design, perm=10000, set.seed=42)
adonis_carb_metab_H

adonis_cytoCox_H = adonis2(cytoCoxRA_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = cytoCoxRA_design, perm=10000, set.seed=42)
adonis_cytoCox_H

adonis_SO4_S_Assim_H = adonis2(SO4_S_AssimRA_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = SO4_S_AssimRA_design, perm=10000, set.seed=42)
adonis_SO4_S_Assim_H

adonis_Assim_SO4_red_H = adonis2(Assim_SO4_redRA_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = Assim_SO4_redRA_design, perm=10000, set.seed=42)
adonis_Assim_SO4_red_H

adonis_Dissim_SO4_red_H = adonis2(Dissim_SO4_redRA_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = Dissim_SO4_redRA_design, perm=10000, set.seed=42)
adonis_Dissim_SO4_red_H

adonis_ThioSOX_H = adonis2(ThioSOXRA_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = ThioSOXRA_design, perm=10000, set.seed=42)
adonis_ThioSOX_H

adonis_DNRA_H = adonis2(DNRARA_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = DNRARA_design, perm=10000, set.seed=42)
adonis_DNRA_H

adonis_ANRA_H = adonis2(ANRARA_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = ANRARA_design, perm=10000, set.seed=42)
adonis_ANRA_H

adonis_Nitrific_H = adonis2(NitrificRA_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = NitrificRA_design, perm=10000, set.seed=42)
adonis_Nitrific_H

adonis_COMAMMOX_H = adonis2(COMAMMOXRA_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = COMAMMOXRA_design, perm=10000, set.seed=42)
adonis_COMAMMOX_H

adonis_Nfix_H = adonis2(NfixRA_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = NfixRA_design, perm=10000, set.seed=42)
adonis_Nfix_H

adonis_CH4_ox_H = adonis2(CH4_oxRA_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = CH4_oxRA_design, perm=10000, set.seed=42)
adonis_CH4_ox_H

adonis_mtrBC_TIGR_H = adonis2(mtrBC_TIGRRA_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = mtrBC_TIGRRA_design, perm=10000, set.seed=42)
adonis_mtrBC_TIGR_H

adonis_FeGenie_H = adonis2(FeGenieRA_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = FeGenieRA_design, perm=10000, set.seed=42)
adonis_FeGenie_H
```


## PCoA

PCoA denitrification, all factors (including hydrologic history AS SHAPE):
```{r PCoA: denitrification; all factors}
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate
pcoa_denit <- cmdscale(denitBC, k=3, eig=TRUE, add=FALSE)

explainvar1b_denit <- round(pcoa_denit$eig[1] / sum(pcoa_denit$eig), 3) * 100
explainvar2b_denit <- round(pcoa_denit$eig[2] / sum(pcoa_denit$eig), 3) * 100
sum.eigb <- sum(explainvar1b_denit, explainvar2b_denit)

explainvar1b_denit # 44.6
explainvar2b_denit # 25.1

pcoa_denit2.groups <- paste(denitRA_design$Treat, denitRA_design$Plant, denitRA_design$Hist, sep = "_")
pcoa_denit2.points <- data.frame(pcoa_denit$points, group = pcoa_denit2.groups)

# Calculate Centroids (mean and SE)
pcoa_denit2.L.centroids <- reshape2::melt(pcoa_denit2.points, id="group", measure.vars = c("X1", "X2"))
pcoa_denit2.centroids <- acast(pcoa_denit2.L.centroids, variable ~ group, mean)
pcoa_denit2.centroids.se <- acast(pcoa_denit2.L.centroids, variable ~ group, se)
pcoa_denit2.centroids.sd <- acast(pcoa_denit2.L.centroids, variable ~ group, sd)

# Combine
pcoa_denit2.cent.dataframe <- cbind(t(pcoa_denit2.centroids), t(pcoa_denit2.centroids.se))
colnames(pcoa_denit2.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")

# Adding columns with the factors. H: hydrologic history
trt_H <- c("Baseline", "Baseline", "Dry","Dry", "Dry","Dry","Wet","Wet", "Wet","Wet") 
plant_H <- c("Baseline", "Baseline", "No Plant","No Plant","Plant", "Plant", "No Plant","No Plant","Plant", "Plant")
history <- c("Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet")
history_plant_combined <- c(paste(history, plant_H, sep = "_"))

pcoa_denit2.cent.dataframe.trts <- as.data.frame(pcoa_denit2.cent.dataframe) 
pcoa_denit2.cent.dataframe.trts$trt <- as.factor(trt_H)
pcoa_denit2.cent.dataframe.trts$plant <- as.factor(plant_H)
pcoa_denit2.cent.dataframe.trts$history <- as.factor(history)
pcoa_denit2.cent.dataframe.trts$history_plant<- as.factor(history_plant_combined)

#Run envfit to see which [GHG]s (start or finish) have good fit with ordination:
fit_denit <- envfit(pcoa_denit$points, GHG_metaG, permutations = 10000)
fit_denit

A_denit <-as.list(fit_denit$vectors)
vec_denit <- as.data.frame(fit_denit$vectors$arrows*sqrt(fit_denit$vectors$r)*0.15)
p_denit <- as.data.frame(A_denit$pvals)
vec_denit <- cbind(vec_denit, p_denit)
vec_denit <- subset(vec_denit, A_denit$pvals<=0.1)

#Plot using ggplot2
df2a <- as.data.frame(pcoa_denit2.cent.dataframe.trts)

plot2a <- ggplot(df2a, aes(x=V1, y=V2, shape=history_plant, colour=trt)) + geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.005), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.005), colour="black") +   
  geom_point(aes(shape = history_plant, colour=trt), size=5, stroke=2) +  
  theme_bw() +
  scale_color_manual(values=c("black", "#997700", "#004488"), name="Treatment") +
  scale_shape_manual(values = c(7, 0, 15, 13, 1, 19), name="History_Plant") +
  xlab("PCoA 1 (44.6%)") + ylab("PCoA 2 (25.1%)") + 
  geom_segment(data=vec_denit, aes(x=0,xend=Dim1,y=0,yend=Dim2), size=1, arrow = arrow(length = unit(0.2, "cm")),colour="black",inherit.aes=F)+ 
 geom_text(data=vec_denit, aes(x=Dim1+0.03, y=Dim2), label= expression(paste("mg CO"[2],"-C m"^{-2})), size=3, inherit.aes=F)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  theme(panel.background = element_blank()) + 
  theme(axis.title = element_text(size=14), axis.text=element_text(size=14), 
          axis.text.x = element_text(size=14),  axis.text.y = element_text(size=14),
          panel.border = element_rect(colour = "black", linewidth=1.25)) +
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  theme(plot.title=element_text(size=14)) +
  theme(legend.text=element_text(size=14), legend.title = element_text(size=14))+
  guides(shape = guide_legend(override.aes = list(size = 4), order = 1), 
           colour = guide_legend(override.aes = list(pch=16, size = 4), order = 2)) +
    ggtitle("Denitrification")

plot2a
```


```{r new denit PCoA}
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate
pcoa_denit <- cmdscale(denitBC, k=2, eig=TRUE, add=TRUE) # changing k to 2 because we only want 2 dimensions anyway
  # Now trying add = TRUE because of https://www.youtube.com/watch?v=G5Qckqq5Erw
  # This doesn't change the shape of the PCoA, and it's not clear that it's necessary

explainvar1b_denit <- round(pcoa_denit$eig[1] / sum(pcoa_denit$eig), 3) * 100
explainvar2b_denit <- round(pcoa_denit$eig[2] / sum(pcoa_denit$eig), 3) * 100
sum.eigb <- sum(explainvar1b_denit, explainvar2b_denit)

explainvar1b_denit # 27.1
explainvar2b_denit # 16.3

pcoa_denit2.groups <- paste(denitRA_design$Treat, denitRA_design$Plant, denitRA_design$Hist, sep = "_")
pcoa_denit2.points <- data.frame(pcoa_denit$points, group = pcoa_denit2.groups)

# Calculate Centroids (mean and SE)
pcoa_denit2.L.centroids <- reshape2::melt(pcoa_denit2.points, id="group", measure.vars = c("X1", "X2"))
pcoa_denit2.centroids <- acast(pcoa_denit2.L.centroids, variable ~ group, mean)
pcoa_denit2.centroids.se <- acast(pcoa_denit2.L.centroids, variable ~ group, se)
pcoa_denit2.centroids.sd <- acast(pcoa_denit2.L.centroids, variable ~ group, sd)

# Combine
pcoa_denit2.cent.dataframe <- cbind(t(pcoa_denit2.centroids), t(pcoa_denit2.centroids.se))
colnames(pcoa_denit2.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")

# Adding columns with the factors. H: hydrologic history
trt_H <- c("Baseline", "Baseline", "Dry","Dry", "Dry","Dry","Wet","Wet", "Wet","Wet") 
plant_H <- c("Baseline", "Baseline", "No Plant","No Plant","Plant", "Plant", "No Plant","No Plant","Plant", "Plant")
history <- c("Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet")
history_plant_combined <- c(paste(history, plant_H, sep = "_"))

pcoa_denit2.cent.dataframe.trts <- as.data.frame(pcoa_denit2.cent.dataframe) 
pcoa_denit2.cent.dataframe.trts$trt <- as.factor(trt_H)
pcoa_denit2.cent.dataframe.trts$plant <- as.factor(plant_H)
pcoa_denit2.cent.dataframe.trts$history <- as.factor(history)
pcoa_denit2.cent.dataframe.trts$history_plant<- as.factor(history_plant_combined)
pcoa_denit2.cent.dataframe.trts$centroid_or_sample <- "centroid"

# Add in points to data frame:
pcoa_denit2.points_ForPlot <- pcoa_denit2.points
colnames(pcoa_denit2.points_ForPlot) <- c("V1", "V2", "group")
# Add labels to match ...cent.dataframe.trts
pcoa_denit2.points_ForPlot$trt <- denitRA_design$Treat %>% str_replace_all(c("w" = "Wet", "_pre" = "Baseline", "d" = "Dry"))
pcoa_denit2.points_ForPlot$plant <- denitRA_design$Plant %>% str_replace_all(c("n" = "No Plant", "y" = "Plant", "_pre" = "Baseline"))
pcoa_denit2.points_ForPlot$history <- denitRA_design$Hist %>% str_replace_all(c("d" = "Dry", "w" = "Wet"))
pcoa_denit2.points_ForPlot$history_plant <- c(paste(pcoa_denit2.points_ForPlot$history, pcoa_denit2.points_ForPlot$plant, sep = "_"))
#remove "group" column
pcoa_denit2.points_ForPlot <- pcoa_denit2.points_ForPlot[, -3]
#add V1e and V2e as zeros:
pcoa_denit2.points_ForPlot$V1e <- numeric(24)
pcoa_denit2.points_ForPlot$V2e <- numeric(24)
# Add column to distinguish sample points from centroids, and later use to set alpha in plot
pcoa_denit2.points_ForPlot$centroid_or_sample <- "sample"

#Merge:
denit_centroids_points_merged <- rbind(pcoa_denit2.cent.dataframe.trts, pcoa_denit2.points_ForPlot)
```
```{r envfit with GHGs ONLY}
#Run envfit to see which [GHG]s (start or finish) have good fit with ordination:
#fit_denit <- envfit(pcoa_denit$points, GHG_metaG, permutations = 10000)
#fit_denit

#A_denit <-as.list(fit_denit$vectors)
#vec_denit <- as.data.frame(fit_denit$vectors$arrows*sqrt(fit_denit$vectors$r)*0.15)
#p_denit <- as.data.frame(A_denit$pvals)
#vec_denit <- cbind(vec_denit, p_denit)
#vec_denit <- subset(vec_denit, A_denit$pvals<=0.1)
```
```{r denit SSF envfit and PCoA}
#Run envfit with new Soil data (SSF = Soil Start & Finish):
fit_denit_SSF <- envfit(pcoa_denit$points, SoilGHG_StartFinish_Envfit, permutations = 10000)
fit_denit_SSF

A_denit_SSF <-as.list(fit_denit_SSF$vectors)
vec_denit_SSF <- as.data.frame(fit_denit_SSF$vectors$arrows*sqrt(fit_denit_SSF$vectors$r)*0.15)
p_denit_SSF <- as.data.frame(A_denit_SSF$pvals)
vec_denit_SSF <- cbind(vec_denit_SSF, p_denit_SSF)
vec_denit_SSF <- subset(vec_denit_SSF, A_denit_SSF$pvals<=0.1)


#Plot using ggplot2
df2a <- as.data.frame(denit_centroids_points_merged)

plot2a <- ggplot(df2a, aes(x=V1, y=V2, shape=history_plant, colour=trt)) + geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.005), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.005), colour="black") +   
  geom_point(aes(shape = history_plant, colour=trt, alpha = centroid_or_sample, size = centroid_or_sample), stroke=2) +  
  theme_bw() +
  scale_size_manual(values = c(6, 3), name = "Centroid or Sample")+
  scale_alpha_manual(values = c(1, 0.3), name = "Centroid or Sample")+
  scale_color_manual(values=c("black", "#997700", "#004488"), name="Treatment") +
  scale_shape_manual(values = c(7, 0, 15, 13, 1, 19), name="History_Plant") +
  xlab(glue("PCoA 1 ({explainvar1b_denit}%)")) + ylab(glue("PCoA 2 ({explainvar2b_denit}%)")) + 
  geom_segment(data=vec_denit_SSF, aes(x=0,xend=Dim1,y=0,yend=Dim2), linewidth=1, arrow = arrow(length = unit(0.2, "cm")),colour="black",inherit.aes=F)+ 
 geom_text(data=vec_denit, aes(x=Dim1+0.03, y=Dim2), label= expression(paste("mg CO"[2],"-C m"^{-2})), size=3, inherit.aes=F)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  theme(panel.background = element_blank()) + 
  theme(axis.title = element_text(size=14), axis.text=element_text(size=14), 
          axis.text.x = element_text(size=14),  axis.text.y = element_text(size=14),
          panel.border = element_rect(colour = "black", linewidth=1.25)) +
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  theme(plot.title=element_text(size=14)) +
  theme(legend.text=element_text(size=14), legend.title = element_text(size=14))+
  guides(shape = guide_legend(override.aes = list(size = 4), order = 1), 
           colour = guide_legend(override.aes = list(pch=16, size = 4), order = 2)) +
    ggtitle("Denitrification")

plot2a

#ggsave("~/Desktop/newDenitPCoa.jpeg", plot = plot2a, device="jpeg", path=NULL, scale=1, height = 7, width = 10, limitsize = TRUE, dpi = 300)
```

```{r Denit FINAL}
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate
pcoa_denit_FINAL <- cmdscale(denitBC_finish, k=2, eig=TRUE, add=TRUE) # k = 2 because we only plot 2 dimensions, add=TRUE corrects highly negative eigenvalues


explainvar1_denit_FIN <- round(pcoa_denit_FINAL$eig[1] / sum(pcoa_denit_FINAL$eig), 3) * 100
explainvar2_denit_FIN <- round(pcoa_denit_FINAL$eig[2] / sum(pcoa_denit_FINAL$eig), 3) * 100
sum.eig_denit_FIN <- sum(explainvar1_denit_FIN, explainvar2_denit_FIN)

explainvar1_denit_FIN # 34.8
explainvar2_denit_FIN # 15.9

# Filter 'pre' samples out of *RA_design to get final samples only
denitRA_FIN_design <- denitRA_design %>% dplyr::filter(Treat != "_pre" & Plant != "_pre")

pcoa_denit_FIN.groups <- paste(denitRA_FIN_design$Treat, denitRA_FIN_design$Plant, denitRA_FIN_design$Hist, sep = "_")
pcoa_denit_FIN.points <- data.frame(pcoa_denit_FINAL$points, group = pcoa_denit_FIN.groups)

# Calculate Centroids (mean and SE)
pcoa_denit_FIN.L.centroids <- reshape2::melt(pcoa_denit_FIN.points, id="group", measure.vars = c("X1", "X2"))
pcoa_denit_FIN.centroids <- acast(pcoa_denit_FIN.L.centroids, variable ~ group, mean)
pcoa_denit_FIN.centroids.se <- acast(pcoa_denit_FIN.L.centroids, variable ~ group, se)
pcoa_denit_FIN.centroids.sd <- acast(pcoa_denit_FIN.L.centroids, variable ~ group, sd)

# Combine
pcoa_denit_FIN.cent.dataframe <- cbind(t(pcoa_denit_FIN.centroids), t(pcoa_denit_FIN.centroids.se))
colnames(pcoa_denit_FIN.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")

# Adding columns with the factors.
trt_FIN <- c("Dry","Dry", "Dry","Dry","Wet","Wet", "Wet","Wet") 
plant_FIN <- c("No Plant","No Plant","Plant", "Plant", "No Plant","No Plant","Plant", "Plant")
history_FIN <- c("Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet")
history_plant_combined_FIN <- c(paste(history_FIN, plant_FIN, sep = "_"))

pcoa_denit_FIN.cent.dataframe.trts <- as.data.frame(pcoa_denit_FIN.cent.dataframe) 
pcoa_denit_FIN.cent.dataframe.trts$trt <- as.factor(trt_FIN)
pcoa_denit_FIN.cent.dataframe.trts$plant <- as.factor(plant_FIN)
pcoa_denit_FIN.cent.dataframe.trts$history <- as.factor(history_FIN)
pcoa_denit_FIN.cent.dataframe.trts$history_plant<- as.factor(history_plant_combined_FIN)
pcoa_denit_FIN.cent.dataframe.trts$centroid_or_sample <- "centroid"

# Add in points to data frame:
pcoa_denit_FIN.points_ForPlot <- pcoa_denit_FIN.points
colnames(pcoa_denit_FIN.points_ForPlot) <- c("V1", "V2", "group")
# Add labels to match ...cent.dataframe.trts
pcoa_denit_FIN.points_ForPlot$trt <- denitRA_FIN_design$Treat %>% str_replace_all(c("w" = "Wet", "_pre" = "Baseline", "d" = "Dry"))
pcoa_denit_FIN.points_ForPlot$plant <- denitRA_FIN_design$Plant %>% str_replace_all(c("n" = "No Plant", "y" = "Plant", "_pre" = "Baseline"))
pcoa_denit_FIN.points_ForPlot$history <- denitRA_FIN_design$Hist %>% str_replace_all(c("d" = "Dry", "w" = "Wet"))
pcoa_denit_FIN.points_ForPlot$history_plant <- c(paste(pcoa_denit_FIN.points_ForPlot$history, pcoa_denit_FIN.points_ForPlot$plant, sep = "_"))
#remove "group" column
pcoa_denit_FIN.points_ForPlot <- pcoa_denit_FIN.points_ForPlot[, -3]
#add V1e and V2e as zeros:
pcoa_denit_FIN.points_ForPlot$V1e <- numeric(16)
pcoa_denit_FIN.points_ForPlot$V2e <- numeric(16)
# Add column to distinguish sample points from centroids, and later use to set alpha in plot
pcoa_denit_FIN.points_ForPlot$centroid_or_sample <- "sample"

#Merge:
denit_FIN_centroids_points_merged <- rbind(pcoa_denit_FIN.cent.dataframe.trts, pcoa_denit_FIN.points_ForPlot)


#Run envfit with new Soil data (SFO = Soil Finish Only):
fit_denit_SFO <- envfit(pcoa_denit_FINAL$points, envfit_FINISH_numerical, permutations = 10000)
fit_denit_SFO

A_denit_SFO <-as.list(fit_denit_SFO$vectors)
vec_denit_SFO <- as.data.frame(fit_denit_SFO$vectors$arrows*sqrt(fit_denit_SFO$vectors$r)*0.15)
p_denit_SFO <- as.data.frame(A_denit_SFO$pvals)
vec_denit_SFO <- cbind(vec_denit_SFO, p_denit_SFO)
vec_denit_SFO <- subset(vec_denit_SFO, A_denit_SFO$pvals<=0.1)



#Plot using ggplot2
df3a <- as.data.frame(denit_FIN_centroids_points_merged)

plot3a <- ggplot(df3a, aes(x=V1, y=V2, shape=history_plant, colour=trt)) + geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.005), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.005), colour="black") +   
  geom_point(aes(shape = history_plant, colour=trt, alpha = centroid_or_sample, size = centroid_or_sample), stroke=2) +  
  theme_bw() +
  scale_size_manual(values = c(6, 3), name = "Centroid or Sample")+
  scale_alpha_manual(values = c(1, 0.3), name = "Centroid or Sample")+
  scale_color_manual(values=c("#997700", "#004488"), name="Treatment") +
  scale_shape_manual(values = c(0, 15, 1, 19), name="History_Plant") +
  xlab(glue("PCoA 1 ({explainvar1_denit_FIN}%)")) + ylab(glue("PCoA 2 ({explainvar2_denit_FIN}%)")) + 
  geom_segment(data=vec_denit_SFO, aes(x=0,xend=Dim1,y=0,yend=Dim2), linewidth=1, arrow = arrow(length = unit(0.2, "cm")),colour="black",inherit.aes=F)+ 
 geom_text(data=vec_denit_SFO, aes(x=Dim1, y=Dim2+0.01), label= c(expression(paste("mg CO"[2],"-C m"^{-2})), expression(paste("mg N"[2],"O-N m"^{-2}))), size=3, inherit.aes=F)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  theme(panel.background = element_blank()) + 
  theme(axis.title = element_text(size=14), axis.text=element_text(size=14), 
          axis.text.x = element_text(size=14),  axis.text.y = element_text(size=14),
          panel.border = element_rect(colour = "black", linewidth=1.25)) +
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  theme(plot.title=element_text(size=14)) +
  theme(legend.text=element_text(size=14), legend.title = element_text(size=14))+
  guides(shape = guide_legend(override.aes = list(size = 4), order = 1), 
           colour = guide_legend(override.aes = list(pch=16, size = 4), order = 2)) +
    ggtitle("Denitrification")

plot3a
```



PCoA Methanogenesis, all factors, HISTORY AS SHAPE:
```{r PCoA: Methanogenesis; all factors}
pcoa_meth <- cmdscale(methBC, k=2, eig=TRUE, add=TRUE)

explainvar1b_meth <- round(pcoa_meth$eig[1] / sum(pcoa_meth$eig), 3) * 100
explainvar2b_meth <- round(pcoa_meth$eig[2] / sum(pcoa_meth$eig), 3) * 100
sum.eigb <- sum(explainvar1b_meth, explainvar2b_meth)

explainvar1b_meth # 29.7
explainvar2b_meth # 20.8

pcoa_meth2.groups <- paste(methRA_design$Treat, methRA_design$Plant, methRA_design$Hist, sep = "_")
pcoa_meth2.points <- data.frame(pcoa_meth$points, group = pcoa_meth2.groups)

# Calculate Centroids (mean and SE)
pcoa_meth2.L.centroids <- reshape2::melt(pcoa_meth2.points, id="group", measure.vars = c("X1", "X2"))
pcoa_meth2.centroids <- acast(pcoa_meth2.L.centroids, variable ~ group, mean)
pcoa_meth2.centroids.se <- acast(pcoa_meth2.L.centroids, variable ~ group, se)
pcoa_meth2.centroids.sd <- acast(pcoa_meth2.L.centroids, variable ~ group, sd)

# Combine
pcoa_meth2.cent.dataframe <- cbind(t(pcoa_meth2.centroids), t(pcoa_meth2.centroids.se))
colnames(pcoa_meth2.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")

# No need to run again if these lists have already been declared:
#trt_H <- c("Baseline", "Baseline", "Dry","Dry", "Dry","Dry","Wet","Wet", "Wet","Wet") 
#plant_H <- c("Baseline", "Baseline", "No Plant","No Plant","Plant", "Plant", "No Plant","No Plant","Plant", "Plant")
#history <- c("Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet")
#trt_plant_combined <- c(paste(trt, plant, sep = "_"))

pcoa_meth2.cent.dataframe.trts <- as.data.frame(pcoa_meth2.cent.dataframe) 
pcoa_meth2.cent.dataframe.trts$trt <- as.factor(trt_H)
pcoa_meth2.cent.dataframe.trts$plant <- as.factor(plant_H)
pcoa_meth2.cent.dataframe.trts$history <- as.factor(history)
pcoa_meth2.cent.dataframe.trts$history_plant<- as.factor(history_plant_combined)
pcoa_meth2.cent.dataframe.trts$centroid_or_sample <- "centroid"

# Add in points to data frame:
pcoa_meth2.points_ForPlot <- pcoa_meth2.points
colnames(pcoa_meth2.points_ForPlot) <- c("V1", "V2", "group")
# Add labels to match ...cent.dataframe.trts
pcoa_meth2.points_ForPlot$trt <- methRA_design$Treat %>% str_replace_all(c("w" = "Wet", "_pre" = "Baseline", "d" = "Dry"))
pcoa_meth2.points_ForPlot$plant <- methRA_design$Plant %>% str_replace_all(c("n" = "No Plant", "y" = "Plant", "_pre" = "Baseline"))
pcoa_meth2.points_ForPlot$history <- methRA_design$Hist %>% str_replace_all(c("d" = "Dry", "w" = "Wet"))
pcoa_meth2.points_ForPlot$history_plant <- c(paste(pcoa_meth2.points_ForPlot$history, pcoa_meth2.points_ForPlot$plant, sep = "_"))
#remove "group" column
pcoa_meth2.points_ForPlot <- pcoa_meth2.points_ForPlot[, -3]
#add V1e and V2e as zeros:
pcoa_meth2.points_ForPlot$V1e <- numeric(24)
pcoa_meth2.points_ForPlot$V2e <- numeric(24)
# Add column to distinguish sample points from centroids, and later use to set alpha in plot
pcoa_meth2.points_ForPlot$centroid_or_sample <- "sample"

#Merge:
meth_centroids_points_merged <- rbind(pcoa_meth2.cent.dataframe.trts, pcoa_meth2.points_ForPlot)
```
```{r envfit with GHGs ONLY}
#Run envfit to see which [GHG]s (start or finish) have good fit with ordination:
#fit_meth <- envfit(pcoa_meth$points, GHG_metaG, permutations = 10000)
#fit_meth

#A_meth <-as.list(fit_meth$vectors)
#vec_meth <- as.data.frame(fit_meth$vectors$arrows*sqrt(fit_meth$vectors$r)*0.15)
#p_meth <- as.data.frame(A_meth$pvals)
#vec_meth <- cbind(vec_meth, p_meth)
#vec_meth <- subset(vec_meth, A_meth$pvals<=0.1)
```
```{r meth SSF envfit and PCoA}
#Run envfit with new Soil data (SSF = Soil Start & Finish):
fit_meth_SSF <- envfit(pcoa_meth$points, SoilGHG_StartFinish_Envfit, permutations = 10000)
fit_meth_SSF

A_meth_SSF <-as.list(fit_meth_SSF$vectors)
vec_meth_SSF <- as.data.frame(fit_meth_SSF$vectors$arrows*sqrt(fit_meth_SSF$vectors$r)*0.15)
p_meth_SSF <- as.data.frame(A_meth_SSF$pvals)
vec_meth_SSF <- cbind(vec_meth_SSF, p_meth_SSF)
vec_meth_SSF <- subset(vec_meth_SSF, A_meth_SSF$pvals<=0.1)


#Plot using ggplot2
df2b <- as.data.frame(pcoa_meth2.cent.dataframe.trts)

plot2b <- ggplot(df2b, aes(x=V1, y=V2, shape=history_plant, colour=trt)) + geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.005), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.005), colour="black") +   
  geom_point(aes(shape = history_plant, colour=trt), size=5, stroke=2) +  
  theme_bw() +
  scale_color_manual(values=c("black", "#997700", "#004488"), name="Treatment") +
  scale_shape_manual(values = c(7, 0, 15, 13, 1, 19), name="History_Plant") +
  xlab(glue("PCoA 1 ({explainvar1b_meth}%)")) + ylab(glue("PCoA 2 ({explainvar2b_meth}%)")) + 
  geom_segment(data=vec_meth, aes(x=0,xend=Dim1,y=0,yend=Dim2), size=1, arrow = arrow(length = unit(0.2, "cm")),colour="black",inherit.aes=F)+ 
 geom_text(data=vec_meth, aes(x=-0.05, y=0.0), label= expression(paste("mg CH"[4],"-C m"^{-2})), size=3, inherit.aes=F)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  theme(panel.background = element_blank()) + 
  theme(axis.title = element_text(size=14), axis.text=element_text(size=14), 
          axis.text.x = element_text(size=14),  axis.text.y = element_text(size=14),
          panel.border = element_rect(colour = "black", linewidth=1.25)) +
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  theme(plot.title=element_text(size=14)) +
  theme(legend.text=element_text(size=14), legend.title = element_text(size=14))+
  guides(shape = guide_legend(override.aes = list(size = 4), order = 1), 
           colour = guide_legend(override.aes = list(pch=16, size = 4), order = 2)) +
    ggtitle("Methanogenesis")

plot2b
```

PCoA central carbohydrate metabolism, all factors, HISTORY AS SHAPE:
```{r PCoA: central carbohydrate metabolism; all factors}
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate
pcoa_carb_metab <- cmdscale(carb_metabBC, k=2, eig=TRUE, add=TRUE)

explainvar1b_carb_metab <- format(round(pcoa_carb_metab$eig[1] / sum(pcoa_carb_metab$eig), 3) * 100, nsmall = 1) # Using format() to get .0
explainvar2b_carb_metab <- round(pcoa_carb_metab$eig[2] / sum(pcoa_carb_metab$eig), 3) * 100
#sum.eigc <- sum(explainvar1b_carb_metab, explainvar2b_carb_metab) #Doesn't work when using format()

explainvar1b_carb_metab # 31.0
explainvar2b_carb_metab # 14.9

pcoa_carb_metab2.groups <- paste(carb_metabRA_design$Treat, carb_metabRA_design$Plant, carb_metabRA_design$Hist, sep = "_")
pcoa_carb_metab2.points <- data.frame(pcoa_carb_metab$points, group = pcoa_carb_metab2.groups)

# Calculate Centroids (mean and SE)
pcoa_carb_metab2.L.centroids <- reshape2::melt(pcoa_carb_metab2.points, id="group", measure.vars = c("X1", "X2"))
pcoa_carb_metab2.centroids <- acast(pcoa_carb_metab2.L.centroids, variable ~ group, mean)
pcoa_carb_metab2.centroids.se <- acast(pcoa_carb_metab2.L.centroids, variable ~ group, se)
pcoa_carb_metab2.centroids.sd <- acast(pcoa_carb_metab2.L.centroids, variable ~ group, sd)

# Combine
pcoa_carb_metab2.cent.dataframe <- cbind(t(pcoa_carb_metab2.centroids), t(pcoa_carb_metab2.centroids.se))
colnames(pcoa_carb_metab2.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")

# No need to run again if these lists have already been declared:
#trt_H <- c("Baseline", "Baseline", "Dry","Dry", "Dry","Dry","Wet","Wet", "Wet","Wet") 
#plant_H <- c("Baseline", "Baseline", "No Plant","No Plant","Plant", "Plant", "No Plant","No Plant","Plant", "Plant")
#history <- c("Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet")
#trt_plant_combined <- c(paste(trt, plant, sep = "_"))

pcoa_carb_metab2.cent.dataframe.trts <- as.data.frame(pcoa_carb_metab2.cent.dataframe) 
pcoa_carb_metab2.cent.dataframe.trts$trt <- as.factor(trt_H)
pcoa_carb_metab2.cent.dataframe.trts$plant <- as.factor(plant_H)
pcoa_carb_metab2.cent.dataframe.trts$history <- as.factor(history)
pcoa_carb_metab2.cent.dataframe.trts$history_plant<- as.factor(history_plant_combined)

#Run envfit to see which [GHG]s (start or finish) have good fit with ordination:
fit_carb_metab <- envfit(pcoa_carb_metab$points, GHG_metaG, permutations = 10000)
fit_carb_metab

A_carb_metab <-as.list(fit_carb_metab$vectors)
vec_carb_metab <- as.data.frame(fit_carb_metab$vectors$arrows*sqrt(fit_carb_metab$vectors$r)*0.15)
p_carb_metab <- as.data.frame(A_carb_metab$pvals)
vec_carb_metab <- cbind(vec_carb_metab, p_carb_metab)
vec_carb_metab <- subset(vec_carb_metab, A_carb_metab$pvals<=0.1)

#Plot using ggplot2
df2c <- as.data.frame(pcoa_carb_metab2.cent.dataframe.trts)

plot2c <- ggplot(df2c, aes(x=V1, y=V2, shape=history_plant, colour=trt)) + geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.002), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.002), colour="black") +   
  geom_point(aes(shape = history_plant, colour=trt), size=5, stroke=2) +  
  theme_bw() +
  scale_color_manual(values=c("black", "#997700", "#004488"), name="Treatment") +
  scale_shape_manual(values = c(7, 0, 15, 13, 1, 19), name="History_Plant") +
  xlab(glue("PCoA 1 ({explainvar1b_carb_metab}%)")) + ylab(glue("PCoA 2 ({explainvar2b_carb_metab}%)")) + 
  geom_segment(data=vec_carb_metab, aes(x=0,xend=Dim1,y=0,yend=Dim2), size=1, arrow = arrow(length = unit(0.2, "cm")),colour="black",inherit.aes=F)+ 
 geom_text(data=vec_carb_metab, aes(x=Dim1+0.02, y=Dim2), label= expression(paste("mg CO"[2],"-C m"^{-2})), size=3, inherit.aes=F)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  theme(panel.background = element_blank()) + 
  theme(axis.title = element_text(size=14), axis.text=element_text(size=14), 
          axis.text.x = element_text(size=14),  axis.text.y = element_text(size=14),
          panel.border = element_rect(colour = "black", linewidth=1.25)) +
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  theme(plot.title=element_text(size=14)) +
  theme(legend.text=element_text(size=14), legend.title = element_text(size=14))+
  guides(shape = guide_legend(override.aes = list(size = 4), order = 1), 
           colour = guide_legend(override.aes = list(pch=16, size = 4), order = 2)) +
    ggtitle("Central Carbohydrate Metabolism")

plot2c
```

PCoA cytochrome C oxidase, all factors, HISTORY AS SHAPE:
```{r PCoA: cytochrome C oxidase; all factors}
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate
pcoa_cytoCox <- cmdscale(cytoCoxBC, k=2, eig=TRUE, add=TRUE)

explainvar1b_cytoCox <- round(pcoa_cytoCox$eig[1] / sum(pcoa_cytoCox$eig), 3) * 100
explainvar2b_cytoCox <- round(pcoa_cytoCox$eig[2] / sum(pcoa_cytoCox$eig), 3) * 100
sum.eigc <- sum(explainvar1b_cytoCox, explainvar2b_cytoCox)

explainvar1b_cytoCox # 29.2
explainvar2b_cytoCox # 22.2

pcoa_cytoCox2.groups <- paste(cytoCoxRA_design$Treat, cytoCoxRA_design$Plant, cytoCoxRA_design$Hist, sep = "_")
pcoa_cytoCox2.points <- data.frame(pcoa_cytoCox$points, group = pcoa_cytoCox2.groups)

# Calculate Centroids (mean and SE)
pcoa_cytoCox2.L.centroids <- reshape2::melt(pcoa_cytoCox2.points, id="group", measure.vars = c("X1", "X2"))
pcoa_cytoCox2.centroids <- acast(pcoa_cytoCox2.L.centroids, variable ~ group, mean)
pcoa_cytoCox2.centroids.se <- acast(pcoa_cytoCox2.L.centroids, variable ~ group, se)
pcoa_cytoCox2.centroids.sd <- acast(pcoa_cytoCox2.L.centroids, variable ~ group, sd)

# Combine
pcoa_cytoCox2.cent.dataframe <- cbind(t(pcoa_cytoCox2.centroids), t(pcoa_cytoCox2.centroids.se))
colnames(pcoa_cytoCox2.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")

# No need to run again if these lists have already been declared:
#trt_H <- c("Baseline", "Baseline", "Dry","Dry", "Dry","Dry","Wet","Wet", "Wet","Wet") 
#plant_H <- c("Baseline", "Baseline", "No Plant","No Plant","Plant", "Plant", "No Plant","No Plant","Plant", "Plant")
#history <- c("Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet")
#trt_plant_combined <- c(paste(trt, plant, sep = "_"))

pcoa_cytoCox2.cent.dataframe.trts <- as.data.frame(pcoa_cytoCox2.cent.dataframe) 
pcoa_cytoCox2.cent.dataframe.trts$trt <- as.factor(trt_H)
pcoa_cytoCox2.cent.dataframe.trts$plant <- as.factor(plant_H)
pcoa_cytoCox2.cent.dataframe.trts$history <- as.factor(history)
pcoa_cytoCox2.cent.dataframe.trts$history_plant<- as.factor(history_plant_combined)

#Run envfit to see which [GHG]s (start or finish) have good fit with ordination:
fit_cytoCox <- envfit(pcoa_cytoCox$points, GHG_metaG, permutations = 10000)
fit_cytoCox

A_cytoCox <-as.list(fit_cytoCox$vectors)
vec_cytoCox <- as.data.frame(fit_cytoCox$vectors$arrows*sqrt(fit_cytoCox$vectors$r)*0.15)
p_cytoCox <- as.data.frame(A_cytoCox$pvals)
vec_cytoCox <- cbind(vec_cytoCox, p_cytoCox)
vec_cytoCox <- subset(vec_cytoCox, A_cytoCox$pvals<=0.1) #No fit with P <= 0.1

#Plot using ggplot2
df2d <- as.data.frame(pcoa_cytoCox2.cent.dataframe.trts)

plot2d <- ggplot(df2d, aes(x=V1, y=V2, shape=history_plant, colour=trt)) + geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.001), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.001), colour="black") +   
  geom_point(aes(shape = history_plant, colour=trt), size=5, stroke=2) +  
  theme_bw() +
  scale_color_manual(values=c("black", "#997700", "#004488"), name="Treatment") +
  scale_shape_manual(values = c(7, 0, 15, 13, 1, 19), name="History_Plant") +
  xlab(glue("PCoA 1 ({explainvar1b_cytoCox}%)")) + ylab(glue("PCoA 2 ({explainvar2b_cytoCox}%)")) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  theme(panel.background = element_blank()) + 
  theme(axis.title = element_text(size=14), axis.text=element_text(size=14), 
          axis.text.x = element_text(size=14),  axis.text.y = element_text(size=14),
          panel.border = element_rect(colour = "black", linewidth=1.25)) +
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  theme(plot.title=element_text(size=14)) +
  theme(legend.text=element_text(size=14), legend.title = element_text(size=14))+
  guides(shape = guide_legend(override.aes = list(size = 4), order = 1), 
           colour = guide_legend(override.aes = list(pch=16, size = 4), order = 2)) +
    ggtitle("Cytochrome C Oxidase")

plot2d
```

Create 4-panel figure of the 4 PCoA plots above:
```{r 4-panel PCoA plot}
#require(ggpubr) #load in library for multi-panel figures (if not already loaded)

# 4 panel metaG PCoA with History
MetaG_pcoa_4panel_H <- ggarrange(plot2a, plot2b, plot2c, plot2d, labels = c("A", "B", "C", "D"), ncol = 2, nrow = 2, legend = "right", common.legend = T)

MetaG_pcoa_4panel_H

#ggsave("../figures/pub/metaG_pcoa4panel_H.tiff", plot = MetaG_pcoa_4panel_H, device="tiff", path=NULL, scale=1, height = 7, width = 10, limitsize = T, bg = "white", dpi = 300)

#ggsave("../figures/pub/metaG_pcoa4panel_H.jpg", plot = MetaG_pcoa_4panel_H, device=NULL, path=NULL, scale=1, height = 7, width = 10, limitsize = TRUE, dpi = 300)
```

