---
title: "metaG_in_depth"
author: "Colin G. Finlay"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document: default
  pdf_document: default
---
# Setup

```{r setup, include=FALSE}
#use to set global options for chunks e.g., echo and warning options will be applied to all chunks:
knitr::opts_chunk$set(echo = TRUE)

# Clear environment:
rm(list=ls())

# Set working directory:
setwd("~/GitHub/WetlandMesocosm_GHG_Timberlake/analyses")

# Standard error (se) and confidence interval (ci):
se <- function(x, ...){sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))}
ci <- function(x, ...){1.96 * sd(x,na.rm = TRUE)}

# Code dependencies:
require("vegan"); require("reshape"); require("ggplot2"): require("tidyr"); require("dplyr"); require("dbstats"); require("reshape2"); require("ggpubr"); require("glue"); require("tidyverse")
```

# Load files

Import KO function files from JGI IMG/MER
```{r Import Files}
#Create vector with desired row order:
sample_order <- c("03NFC", "03PFC", "03Pre", "06NFC", "06PFC", "06Pre", "08NFC", "08PFC", "08Pre", "09NFC", "09PFC", "09Pre", "10NFC", "10PFC", "10Pre", "11NFC", "11PFC", "11Pre", "12NFC", "12PFC", "12Pre", "16NFC", "16PFC", "16Pre")
length(sample_order) #Should be 24

#Denitrification KEGG module (M00529):
denit <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/Redownloaded/CSV/Denit_CF.csv")
#Reorder rows:
denit <- denit[match(sample_order, denit$Sample), ]

#Methanogen Signature KEGG module (M00617):
meth <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/Redownloaded/CSV/Meth_CF.csv")
meth <- meth[match(sample_order, meth$Sample), ]

#All Central Carbohydrate Metabolism KEGG modules:
carb_metab <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/Redownloaded/CSV/Central_carb_metab_CF.csv")
carb_metab <- carb_metab[match(sample_order, carb_metab$Sample), ]

#Cytochrome C Oxidase KEGG module (M00155):
cytoCox <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/Redownloaded/CSV/CytCOx_CF.csv")
cytoCox <- cytoCox[match(sample_order, cytoCox$Sample), ]

#Sulfate-sulfur assimilation (M00616):
SO4_S_Assim <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/IMG_functions_in_depth/M00616_SO4_S_Assimiliation.csv")
#Reorder rows:
SO4_S_Assim <- SO4_S_Assim[match(sample_order, SO4_S_Assim$Sample), ]

#Assimilatory sulfate reduction (M00176):
Assim_SO4_red <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/IMG_functions_in_depth/M00176_Assim_SO4_reduction.csv")
#Reorder rows:
Assim_SO4_red <- Assim_SO4_red[match(sample_order, Assim_SO4_red$Sample), ]

#Dissimilatory sulfate reduction (M00596):
Dissim_SO4_red <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/IMG_functions_in_depth/M00596_Dissim_SO4_reduction.csv")
#Reorder rows:
Dissim_SO4_red <- Dissim_SO4_red[match(sample_order, Dissim_SO4_red$Sample), ]

#Thiosulfate oxidation by SOX complex (M00595):
ThioSOX <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/IMG_functions_in_depth/M00595_Thiosulfate_ox_by_SOX.csv")
#Reorder rows:
ThioSOX <- ThioSOX[match(sample_order, ThioSOX$Sample), ]

#Dissimilatory nitrate reduction (M00530):
DNRA <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/IMG_functions_in_depth/M00530_DNRA.csv")
#Reorder rows:
DNRA <- DNRA[match(sample_order, DNRA$Sample), ]

#Assimilatory nitrate reduction (M00531):
ANRA <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/IMG_functions_in_depth/M00531_ANRA.csv")
#Reorder rows:
ANRA <- ANRA[match(sample_order, ANRA$Sample), ]

#Nitrification (M00528):
Nitrific <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/IMG_functions_in_depth/M00528_Nitrification.csv")
#Reorder rows:
Nitrific <- Nitrific[match(sample_order, Nitrific$Sample), ]

#COMAMMOX (M00804):
COMAMMOX <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/IMG_functions_in_depth/M00804_COMAMMOX.csv")
#Reorder rows:
COMAMMOX <- COMAMMOX[match(sample_order, COMAMMOX$Sample), ]

#Nitrogen fixation (M00175):
Nfix <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/IMG_functions_in_depth/M00175_N_fixation.csv")
#Reorder rows:
Nfix <- Nfix[match(sample_order, Nfix$Sample), ]

#Methane oxidation (M00174):
CH4_ox <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/IMG_functions_in_depth/M00174_MethaneOxidation.csv")
#Reorder rows:
CH4_ox <- CH4_ox[match(sample_order, CH4_ox$Sample), ]

#Fe reduction (mtrBC) (TIGR03507 + TIGR03509):
mtrBC_TIGR <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/IMG_functions_in_depth/FeR_mtrBC.csv")
#Reorder rows:
mtrBC_TIGR <- mtrBC_TIGR[match(sample_order, mtrBC_TIGR$Sample), ]

#Bins, all KOs:

```

```{r Import FeGenie output}
#Gene summary:
FeSummary <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/FeGenie_meta_full/FeGenie-geneSummaryRReady.csv", header = T)

FeHeatmap <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/FeGenie_meta_full/FeGenie-heatmap-data.csv", header = T)
```

```{r Wrangle FeGenie Heatmap}
# read in metagenomic metadata from IMG, that will be used to match genome IDs to Sample IDs:
metaGmetaD <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/metaGmetaData.csv", header = T)

# transpose heatmap:
FeHeatTranspose <- as.data.frame(t(FeHeatmap))
# Fix column names on FeHeatTranspose:
colnames(FeHeatTranspose) <- FeHeatTranspose[1,]
#Now get rid of column 'X' and extra column at bottom:
FeHeatTranspose <- FeHeatTranspose[-c(1,26),]

#String manipulation; cut out extra characters to match IMG taxon oid:
taxon_oid <- as.vector(str_sub(rownames(FeHeatTranspose), start = 2, end = -7))
#Add as new column to FeHeatTranspose:
FeHeatTranspose$taxon_oid <- taxon_oid

#Merge FeHeatTranspose and metaGmetaD by taxon_oid:
FeMerged <- merge(FeHeatTranspose, metaGmetaD)

# Match sample order:
FeMerged <- FeMerged[match(sample_order, FeMerged$Sample), ]
```

```{r wrangle FeSummary}
# purpose: create FeRed dataframe with same layout as KEGG Module data from IMG/MER

# create taxon_oid column
FeSummary$taxon_oid <- as.vector(str_sub(FeSummary$genome.assembly, start = 1, end = -7))

# subset metaGmetaD to get only taxon_oid and Sample:
taxon_oid_sample <- metaGmetaD %>% dplyr::select(c(taxon_oid, Sample))

# create sample column, matched with taxon oid
FeSummary_sample <- merge(FeSummary, taxon_oid_sample, by = "taxon_oid") 

# Filter only Fe Reduction category and Fe Oxidation category: 
FeSummaryRED <- FeSummary %>% dplyr::filter(category == "iron_reduction")
FeSummaryOX <- FeSummary %>% dplyr::filter(category == "iron_oxidation")

# create a df:
FeReduction <- data.frame(matrix(nrow = length(sample_order), ncol = length(unique(FeSummaryRED$HMM))))
rownames(FeReduction) <- sample_order
colnames(FeReduction) <- unique(FeSummaryRED$HMM)

FeOxidation <- data.frame(matrix(nrow = length(sample_order), ncol = length(unique(FeSummaryOX$HMM))))
rownames(FeOxidation) <- sample_order
colnames(FeOxidation) <- unique(FeSummaryOX$HMM)
```

```{r Populate FeReduction}
# Fill data frame with sums that match HMM and sample:

x <- 1 # x is a counter that will be used to index rows, and add one after each loop. Must be set to 1 before loop starts
for(i in rownames(FeReduction)){

   FeReduction$MtrB_TIGR03509[x] <- sum(FeSummary_sample$HMM == "MtrB_TIGR03509" & FeSummary_sample$Sample == i)
   
   FeReduction$MtrA[x] <- sum(FeSummary_sample$HMM == "MtrA" & FeSummary_sample$Sample == i)
   
   FeReduction$MtrC_TIGR03507[x] <- sum(FeSummary_sample$HMM == "MtrC_TIGR03507" & FeSummary_sample$Sample == i)
                                        
   FeReduction$OmcS[x] <- sum(FeSummary_sample$HMM == "OmcS" & FeSummary_sample$Sample == i)
                              
   FeReduction$OmcF[x] <- sum(FeSummary_sample$HMM == "OmcF" & FeSummary_sample$Sample == i)
                              
   FeReduction$DFE_0462[x] <- sum(FeSummary_sample$HMM == "DFE_0462" & FeSummary_sample$Sample == i)
                                  
   FeReduction$DFE_0463[x] <- sum(FeSummary_sample$HMM == "DFE_0463" & FeSummary_sample$Sample == i)
                                  
   FeReduction$DFE_0461[x] <- sum(FeSummary_sample$HMM == "DFE_0461" & FeSummary_sample$Sample == i)
                                  
   FeReduction$t4ap[x] <- sum(FeSummary_sample$HMM == "t4ap" & FeSummary_sample$Sample == i)
                              
   FeReduction$DFE_0465[x] <- sum(FeSummary_sample$HMM == "DFE_0465" & FeSummary_sample$Sample == i)
                                  
   FeReduction$`643638980 YP_002537145 {omc} hypothetical protein [Geobacter sp. FRC-32: NC_011979]`[x] <- sum(FeSummary_sample$HMM == "643638980 YP_002537145 {omc} hypothetical protein [Geobacter sp. FRC-32: NC_011979]" & FeSummary_sample$Sample == i)
                                                                                                               
   FeReduction$`643638979 YP_002537144 {pc} hypothetical protein [Geobacter sp. FRC-32: NC_011979]`[x] <- sum(FeSummary_sample$HMM == "643638979 YP_002537144 {pc} hypothetical protein [Geobacter sp. FRC-32: NC_011979]" & FeSummary_sample$Sample == i)
   
   FeReduction$`637778187 YP_383875 {porin} hypothetical protein [Geobacter metallireducens GS-15: NC_007517]`[x] <- sum(FeSummary_sample$HMM == "637778187 YP_383875 {porin} hypothetical protein [Geobacter metallireducens GS-15: NC_007517]" & FeSummary_sample$Sample == i)
   
   FeReduction$`637127429 NP_953783 {pc} omaB cytochrome c family protein [Geobacter sulfurreducens PCA: NC_002939]`[x] <- sum(FeSummary_sample$HMM == "637127429 NP_953783 {pc} omaB cytochrome c family protein [Geobacter sulfurreducens PCA: NC_002939]" & FeSummary_sample$Sample == i)
   
   FeReduction$`2514223670 filler {omc} Geobacter sulfurreducens CxxxxCH...CXXCH domain-containing protein [Ignavibacterium album Mat9-16; JCM 16511 : NC_017464]`[x] <- sum(FeSummary_sample$HMM == "2514223670 filler {omc} Geobacter sulfurreducens CxxxxCH...CXXCH domain-containing protein [Ignavibacterium album Mat9-16; JCM 16511 : NC_017464]" & FeSummary_sample$Sample == i)
   
   FeReduction$`642764945 YP_002135816 {porin} hypothetical protein [Anaeromyxobacter sp. K: NC_011145]`[x] <- sum(FeSummary_sample$HMM == "642764945 YP_002135816 {porin} hypothetical protein [Anaeromyxobacter sp. K: NC_011145]" & FeSummary_sample$Sample == i)
   
   FeReduction$`642557020 CAJ74788 {pc} hypothetical multiheme cytochrome c protein [Kuenenia stuttgartiensis genome fragment KUST_E (5 of 5).: CT573071]`[x] <- sum(FeSummary_sample$HMM == "642557020 CAJ74788 {pc} hypothetical multiheme cytochrome c protein [Kuenenia stuttgartiensis genome fragment KUST_E (5 of 5).: CT573071]" & FeSummary_sample$Sample == i)
   
   FeReduction$`642557019 CAJ74787 {porin} unknown protein [Kuenenia stuttgartiensis genome fragment KUST_E (5 of 5).: CT573071]`[x] <- sum(FeSummary_sample$HMM == "642557019 CAJ74787 {porin} unknown protein [Kuenenia stuttgartiensis genome fragment KUST_E (5 of 5).: CT573071]" & FeSummary_sample$Sample == i)
   
   FeReduction$`637864947 YP_466596 {pc} hypothetical protein [Anaeromyxobacter dehalogenans 2CP-C: NC_007760]`[x] <- sum(FeSummary_sample$HMM == "637864947 YP_466596 {pc} hypothetical protein [Anaeromyxobacter dehalogenans 2CP-C: NC_007760]" & FeSummary_sample$Sample == i)
   
   FeReduction$`2514223672 filler {porin} hypothetical protein [Ignavibacterium album Mat9-16; JCM 16511 : NC_017464]`[x] <- sum(FeSummary_sample$HMM == "2514223672 filler {porin} hypothetical protein [Ignavibacterium album Mat9-16; JCM 16511 : NC_017464]" & FeSummary_sample$Sample == i)
   
   FeReduction$`2514223671 filler {pc} c(7)-type cytochrome triheme domain-containing protein [Ignavibacterium album Mat9-16; JCM 16511 : NC_017464]`[x] <- sum(FeSummary_sample$HMM == "2514223671 filler {pc} c(7)-type cytochrome triheme domain-containing protein [Ignavibacterium album Mat9-16; JCM 16511 : NC_017464]" & FeSummary_sample$Sample == i)
   
   FeReduction$`637127416 NP_953770 {pc} cytochrome c family protein [Geobacter sulfurreducens PCA: NC_002939]`[x] <- sum(FeSummary_sample$HMM == "637127416 NP_953770 {pc} cytochrome c family protein [Geobacter sulfurreducens PCA: NC_002939]" & FeSummary_sample$Sample == i)
   
   FeReduction$`640549330 YP_001230602 {porin} hypothetical protein [Geobacter uraniumreducens Rf4: NC_009483]`[x] <- sum(FeSummary_sample$HMM == "640549330 YP_001230602 {porin} hypothetical protein [Geobacter uraniumreducens Rf4: NC_009483]" & FeSummary_sample$Sample == i)
   
   FeReduction$`644868482 YP_003020691 {pc} hypothetical protein [Geobacter sp. M21: NC_012918]`[x] <- sum(FeSummary_sample$HMM == "644868482 YP_003020691 {pc} hypothetical protein [Geobacter sp. M21: NC_012918]" & FeSummary_sample$Sample == i)
   
   FeReduction$`637778190 YP_383878 {porin} hypothetical protein [Geobacter metallireducens GS-15: NC_007517]`[x] <- sum(FeSummary_sample$HMM == "637778190 YP_383878 {porin} hypothetical protein [Geobacter metallireducens GS-15: NC_007517]" & FeSummary_sample$Sample == i)
   
   FeReduction$`649931662 YP_004200173 {porin} hypothetical protein [Geobacter sp. M18 chromosome: NC_014973]`[x] <- sum(FeSummary_sample$HMM == "649931662 YP_004200173 {porin} hypothetical protein [Geobacter sp. M18 chromosome: NC_014973]"  & FeSummary_sample$Sample == i)
                                                                                                                         
   FeReduction$`640548487 YP_001229770 {porin} hypothetical protein [Geobacter uraniumreducens Rf4: NC_009483]`[x] <- sum(FeSummary_sample$HMM == "640548487 YP_001229770 {porin} hypothetical protein [Geobacter uraniumreducens Rf4: NC_009483]" & FeSummary_sample$Sample == i)
                                                                                                                          
   FeReduction$`640548488 YP_001229771 {pc} hypothetical protein [Geobacter uraniumreducens Rf4: NC_009483]`[x] <- sum(FeSummary_sample$HMM == "640548488 YP_001229771 {pc} hypothetical protein [Geobacter uraniumreducens Rf4: NC_009483]" & FeSummary_sample$Sample == i)
                                                                                                                       
   FeReduction$`649931668 YP_004200179 {porin} hypothetical protein [Geobacter sp. M18 chromosome: NC_014973]`[x] <- sum(FeSummary_sample$HMM == "649931668 YP_004200179 {porin} hypothetical protein [Geobacter sp. M18 chromosome: NC_014973]" & FeSummary_sample$Sample == i)
   
   FeReduction$`649931661 YP_004200172 {pc} cytochrome c family protein [Geobacter sp. M18 chromosome: NC_014973]`[x] <- sum(FeSummary_sample$HMM == "649931661 YP_004200172 {pc} cytochrome c family protein [Geobacter sp. M18 chromosome: NC_014973]" & FeSummary_sample$Sample == i)
   
   FeReduction$`637864946 YP_466595 {porin} hypothetical protein [Anaeromyxobacter dehalogenans 2CP-C: NC_007760]`[x] <- sum(FeSummary_sample$HMM == "637864946 YP_466595 {porin} hypothetical protein [Anaeromyxobacter dehalogenans 2CP-C: NC_007760]" & FeSummary_sample$Sample == i)
   
   FeReduction$`642764946 YP_002135817 {pc} hypothetical protein [Anaeromyxobacter sp. K: NC_011145]`[x] <- sum(FeSummary_sample$HMM == "642764946 YP_002135817 {pc} hypothetical protein [Anaeromyxobacter sp. K: NC_011145]" & FeSummary_sample$Sample == i)
   
   FeReduction$`637778189 YP_383877 {omc} hypothetical protein [Geobacter metallireducens GS-15: NC_007517]`[x] <- sum(FeSummary_sample$HMM == "637778189 YP_383877 {omc} hypothetical protein [Geobacter metallireducens GS-15: NC_007517]" & FeSummary_sample$Sample == i)
   
   FeReduction$`643638974 YP_002537139 {porin} hypothetical protein [Geobacter sp. FRC-32: NC_011979]`[x] <- sum(FeSummary_sample$HMM == "643638974 YP_002537139 {porin} hypothetical protein [Geobacter sp. FRC-32: NC_011979]" & FeSummary_sample$Sample == i)
   
   FeReduction$`640548483 YP_001229766 {porin} hypothetical protein [Geobacter uraniumreducens Rf4: NC_009483]`[x] <- sum(FeSummary_sample$HMM == "640548483 YP_001229766 {porin} hypothetical protein [Geobacter uraniumreducens Rf4: NC_009483]" & FeSummary_sample$Sample == i)
   
   FeReduction$`637778188 YP_383876 {pc} cytochrome c family protein [Geobacter metallireducens GS-15: NC_007517]`[x] <- sum(FeSummary_sample$HMM == "637778188 YP_383876 {pc} cytochrome c family protein [Geobacter metallireducens GS-15: NC_007517]" & FeSummary_sample$Sample == i)
   
   FeReduction$`642769372 YP_002140172 {porin} hypothetical protein [Geobacter bemidjiensis Bem: NC_011146]`[x] <- sum(FeSummary_sample$HMM == "642769372 YP_002140172 {porin} hypothetical protein [Geobacter bemidjiensis Bem: NC_011146]" & FeSummary_sample$Sample == i)
   
   FeReduction$`643596352 YP_002493935 {pc} hypothetical protein [Anaeromyxobacter dehalogenans 2CP-1: NC_011891]`[x] <- sum(FeSummary_sample$HMM == "643596352 YP_002493935 {pc} hypothetical protein [Anaeromyxobacter dehalogenans 2CP-1: NC_011891]" & FeSummary_sample$Sample == i)
   
   FeReduction$`642769371 YP_002140171 {pc} hypothetical protein [Geobacter bemidjiensis Bem: NC_011146]`[x] <- sum(FeSummary_sample$HMM == "642769371 YP_002140171 {pc} hypothetical protein [Geobacter bemidjiensis Bem: NC_011146]" & FeSummary_sample$Sample == i)

   x <- x + 1
  
}

```

```{r Populate FeOxidation}
# Fill data frame with sums that match HMM and sample:

x <- 1 # x is a counter that will be used to index rows, and add one after each loop. Must be set to 1 before loop starts
for(i in rownames(FeOxidation)){

   FeOxidation$Cyc1[x] <- sum(FeSummary_sample$HMM == "Cyc1" & FeSummary_sample$Sample == i)
   
   FeOxidation$Cyc2_repCluster2[x] <- sum(FeSummary_sample$HMM == "Cyc2_repCluster2" & FeSummary_sample$Sample == i)
   
   FeOxidation$sulfocyanin[x] <- sum(FeSummary_sample$HMM == "sulfocyanin" & FeSummary_sample$Sample == i)
                                        
   FeOxidation$Cyc2_repCluster1[x] <- sum(FeSummary_sample$HMM == "Cyc2_repCluster1" & FeSummary_sample$Sample == i)
                              
   FeOxidation$Cyc2_repCluster3[x] <- sum(FeSummary_sample$HMM == "Cyc2_repCluster3" & FeSummary_sample$Sample == i)
                              
   FeOxidation$FoxE[x] <- sum(FeSummary_sample$HMM == "FoxE" & FeSummary_sample$Sample == i)
                                  
   FeOxidation$FoxY[x] <- sum(FeSummary_sample$HMM == "FoxY" & FeSummary_sample$Sample == i)
                                  
   FeOxidation$CymA[x] <- sum(FeSummary_sample$HMM == "CymA" & FeSummary_sample$Sample == i)
                                  
   x <- x + 1
  
}

```

```{r combined sulfate reduction}
SO4_red_combined <- merge(Assim_SO4_red, Dissim_SO4_red, by=c("Genome", "Sample", "KO.K00958"))
```

```{r Import metadata and greenhouse gas data}
#Design file:
design <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/design.csv")
design2 <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/tmg_design.csv")
#Reorder rows of design:
design <- design[match(sample_order, design$Sample), ]

#Remove mock, and box samples from design2. 
rownames(design2) <- design2$sample
design2 <- design2[!(rownames(design2) %in% "MOCK"),]
design2 <- design2[!(design2$source %in% "b"),]
design2 <- design2[!(design2$source %in% "s"),] # Consider using start samples later (16S data?)

#Bin genome IDs:
Bin_genome_IDs <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/metaG/Bins/Bin_Genome_IDs.csv")

#Greenhouse gas data:
GHG <- read.csv("../data/TMG_GHG-Flux_Pub_3.csv")

# Import soil physicochemical data:
soil <- read.csv("../data/Design_and_soil.csv",  header=T)
```

# Data Wrangling

```{r soil wrangling}
# Start and finish samples, but no field samples (SSF = soil.start.finish)
soil.start.finish <- soil %>%
  filter(time!="field")

SSF.by_time_plant_treatment <- soil.start.finish %>%
  dplyr::select(chamber, time, history, treatment, plant, nh4.mg.l, redox.percent.paint.removed, moisture.percent) %>%
  group_by(chamber)

# No field or start samples (SFO = soil.finishOnly)
soil.finishONLY <- soil %>%
  filter(time!="field")%>%
  filter(time!="start")

SFO.by_time_plant_treatment <- soil.finishONLY %>%
  dplyr::select(-c(g.soil.kcl, no2.no3.mg.l, plant.c.percent, plant.n.percent, plant.dry.mass.g, CEC., HM.)) %>%
  group_by(chamber)
```

```{r Sample IDs to rownames}
#For metagenomic data, make sample IDs rownames in a new data frame, to be further subsetted to numeric only data:

denitNumerical <- denit
rownames(denitNumerical) <- denitNumerical$Sample

methNumerical <- meth
rownames(methNumerical) <- methNumerical$Sample

carb_metabNumerical <- carb_metab
rownames(carb_metabNumerical) <- carb_metabNumerical$Sample

cytoCoxNumerical <- cytoCox
rownames(cytoCoxNumerical) <- cytoCoxNumerical$Sample

SO4_S_AssimNumerical <- SO4_S_Assim
rownames(SO4_S_AssimNumerical) <- SO4_S_AssimNumerical$Sample

Assim_SO4_redNumerical <- Assim_SO4_red
rownames(Assim_SO4_redNumerical) <- Assim_SO4_redNumerical$Sample

Dissim_SO4_redNumerical <- Dissim_SO4_red
rownames(Dissim_SO4_redNumerical) <- Dissim_SO4_redNumerical$Sample

ThioSOXNumerical <- ThioSOX
rownames(ThioSOXNumerical) <- ThioSOXNumerical$Sample

DNRANumerical <- DNRA
rownames(DNRANumerical) <- DNRANumerical$Sample

ANRANumerical <- ANRA
rownames(ANRANumerical) <- ANRANumerical$Sample

NitrificNumerical <- Nitrific
rownames(NitrificNumerical) <- NitrificNumerical$Sample

COMAMMOXNumerical <- COMAMMOX
rownames(COMAMMOXNumerical) <- COMAMMOXNumerical$Sample

NfixNumerical <- Nfix
rownames(NfixNumerical) <- NfixNumerical$Sample

CH4_oxNumerical <- CH4_ox
rownames(CH4_oxNumerical) <- CH4_oxNumerical$Sample

mtrBC_TIGRNumerical <- mtrBC_TIGR
rownames(mtrBC_TIGRNumerical) <- mtrBC_TIGRNumerical$Sample

FeGenieNumerical <- FeMerged
rownames(FeGenieNumerical) <- FeGenieNumerical$Sample 

SO4_red_combinedNumerical <- SO4_red_combined
rownames(SO4_red_combinedNumerical) <- SO4_red_combinedNumerical$Sample
```

```{r make numeric only data frames for functional genes}
denitNumerical <- denitNumerical[,-(1:2)]

methNumerical <- methNumerical[,-(1:2)]

carb_metabNumerical <- carb_metabNumerical[,-(1:2)]

cytoCoxNumerical <- cytoCoxNumerical[,-(1:2)]

SO4_S_AssimNumerical <- SO4_S_AssimNumerical[,-(1:2)]

Assim_SO4_redNumerical <- Assim_SO4_redNumerical[,-(1:2)]

Dissim_SO4_redNumerical <- Dissim_SO4_redNumerical[,-(1:2)]

ThioSOXNumerical <- ThioSOXNumerical[,-(1:2)]

DNRANumerical <- DNRANumerical[,-(1:2)]

ANRANumerical <- ANRANumerical[,-(1:2)]

NitrificNumerical <- NitrificNumerical[,-(1:2)]

COMAMMOXNumerical <- COMAMMOXNumerical[,-(1:2)]

NfixNumerical <- NfixNumerical[,-(1:2)]

CH4_oxNumerical <- CH4_oxNumerical[,-(1:2)]

mtrBC_TIGRNumerical <- mtrBC_TIGRNumerical[,-(1:2)]

FeGenieNumerical <- FeGenieNumerical[, 2:14]
# Must convert characters to numeric for all the data from FeGenie:
#(using a brute force method for now, because struggling to find something more efficient)
FeGenieNumerical$`iron_aquisition-iron_transport` <- as.numeric(FeGenieNumerical$`iron_aquisition-iron_transport`)
FeGenieNumerical$`iron_aquisition-heme_transport` <- as.numeric(FeGenieNumerical$`iron_aquisition-heme_transport`)
FeGenieNumerical$`iron_aquisition-heme_oxygenase` <- as.numeric(FeGenieNumerical$`iron_aquisition-heme_oxygenase`)
FeGenieNumerical$`iron_aquisition-siderophore_synthesis` <- as.numeric(FeGenieNumerical$`iron_aquisition-siderophore_synthesis`)
FeGenieNumerical$`iron_aquisition-siderophore_transport` <- as.numeric(FeGenieNumerical$`iron_aquisition-siderophore_transport`)
FeGenieNumerical$`iron_aquisition-siderophore_transport_potential` <- as.numeric(FeGenieNumerical$`iron_aquisition-siderophore_transport_potential`)
FeGenieNumerical$iron_gene_regulation <- as.numeric(FeGenieNumerical$iron_gene_regulation)
FeGenieNumerical$iron_oxidation <- as.numeric(FeGenieNumerical$iron_oxidation)
FeGenieNumerical$possible_iron_oxidation_and_possible_iron_reduction <- as.numeric(FeGenieNumerical$possible_iron_oxidation_and_possible_iron_reduction)
FeGenieNumerical$probable_iron_reduction <- as.numeric(FeGenieNumerical$probable_iron_reduction)
FeGenieNumerical$iron_reduction <- as.numeric(FeGenieNumerical$iron_reduction)
FeGenieNumerical$iron_storage <- as.numeric(FeGenieNumerical$iron_storage)
FeGenieNumerical$magnetosome_formation <- as.numeric(FeGenieNumerical$magnetosome_formation)

# Combined sulfate reduction numerical
SO4_red_combinedNumerical <- SO4_red_combinedNumerical[,-(1:2)]
```

Note for future analyses:
- use the package "funrar", and function make_relative() to do relative abundance calculations
  - yields same results as our for loops below
  - has publication: https://doi.org/10.1111/ddi.12629 
- install.packages("funrar")
- require("funrar")
- relative_abundance <- make_relative(as.matrix(Numerical))

```{r Convert gene counts to relative abundance}
#Convert denitrification KO counts to relative abundance:
denitRA <- denitNumerical
for(i in 1:dim(denitNumerical)[1]){
  denitRA[i,] <- denitNumerical[i,]/sum(denitNumerical[i,])
}

#Convert methanogen KO counts to relative abundance:
methRA <- methNumerical
for(i in 1:dim(methNumerical)[1]){
  methRA[i,] <- methNumerical[i,]/sum(methNumerical[i,])
}

#Convert central carbohydrate metabolism KO counts to relative abundance:
carb_metabRA <- carb_metabNumerical
for(i in 1:dim(carb_metabNumerical)[1]){
  carb_metabRA[i,] <- carb_metabNumerical[i,]/sum(carb_metabNumerical[i,])
}

#Convert cytochrome C oxidase KO counts to relative abundance:
cytoCoxRA <- cytoCoxNumerical
for(i in 1:dim(cytoCoxNumerical)[1]){
  cytoCoxRA[i,] <- cytoCoxNumerical[i,]/sum(cytoCoxNumerical[i,])
}

SO4_S_AssimRA <- SO4_S_AssimNumerical
for(i in 1:dim(SO4_S_AssimNumerical)[1]){
  SO4_S_AssimRA[i,] <- SO4_S_AssimNumerical[i,]/sum(SO4_S_AssimNumerical[i,])
}

Assim_SO4_redRA <- Assim_SO4_redNumerical
for(i in 1:dim(Assim_SO4_redNumerical)[1]){
  Assim_SO4_redRA[i,] <- Assim_SO4_redNumerical[i,]/sum(Assim_SO4_redNumerical[i,])
}

Dissim_SO4_redRA <- Dissim_SO4_redNumerical
for(i in 1:dim(Dissim_SO4_redNumerical)[1]){
  Dissim_SO4_redRA[i,] <- Dissim_SO4_redNumerical[i,]/sum(Dissim_SO4_redNumerical[i,])
}

ThioSOXRA <- ThioSOXNumerical
for(i in 1:dim(ThioSOXNumerical)[1]){
  ThioSOXRA[i,] <- ThioSOXNumerical[i,]/sum(ThioSOXNumerical[i,])
}

DNRARA <- DNRANumerical
for(i in 1:dim(DNRANumerical)[1]){
  DNRARA[i,] <- DNRANumerical[i,]/sum(DNRANumerical[i,])
}

ANRARA <- ANRANumerical
for(i in 1:dim(ANRANumerical)[1]){
  ANRARA[i,] <- ANRANumerical[i,]/sum(ANRANumerical[i,])
}

NitrificRA <- NitrificNumerical
for(i in 1:dim(NitrificNumerical)[1]){
  NitrificRA[i,] <- NitrificNumerical[i,]/sum(NitrificNumerical[i,])
}

COMAMMOXRA <- COMAMMOXNumerical
for(i in 1:dim(COMAMMOXNumerical)[1]){
  COMAMMOXRA[i,] <- COMAMMOXNumerical[i,]/sum(COMAMMOXNumerical[i,])
}

NfixRA <- NfixNumerical
for(i in 1:dim(NfixNumerical)[1]){
  NfixRA[i,] <- NfixNumerical[i,]/sum(NfixNumerical[i,])
}

CH4_oxRA <- CH4_oxNumerical
for(i in 1:dim(CH4_oxNumerical)[1]){
  CH4_oxRA[i,] <- CH4_oxNumerical[i,]/sum(CH4_oxNumerical[i,])
}

mtrBC_TIGRRA <- mtrBC_TIGRNumerical
for(i in 1:dim(mtrBC_TIGRNumerical)[1]){
  mtrBC_TIGRRA[i,] <- mtrBC_TIGRNumerical[i,]/sum(mtrBC_TIGRNumerical[i,])
}

FeGenieRA <- FeGenieNumerical
for(i in 1:dim(FeGenieNumerical)[1]){
  FeGenieRA[i,] <- FeGenieNumerical[i,]/sum(FeGenieNumerical[i,])
}

SO4_red_combinedRA <- SO4_red_combinedNumerical
for(i in 1:dim(SO4_red_combinedNumerical)[1]){
  SO4_red_combinedRA[i,] <- SO4_red_combinedNumerical[i,]/sum(SO4_red_combinedNumerical[i,])
}

FeReductionRA <- FeReduction
for(i in 1:dim(FeReduction)[1]){
  FeReductionRA[i,] <- FeReduction[i,]/sum(FeReduction[i,])
}

FeOxidationRA <- FeOxidation
for(i in 1:dim(FeOxidation)[1]){
  FeOxidationRA[i,] <- FeOxidation[i,]/sum(FeOxidation[i,])
}
```

```{r lists of start and finish samples}
#Make a list for start and finish, so you don't have to re-write it each time below
start_metaG_samples <- c("03Pre", "06Pre", "08Pre", "09Pre", "10Pre", "11Pre", "12Pre", "16Pre")
finish_metaG_samples <- c("03NFC", "03PFC", "06NFC", "06PFC", "08NFC", "08PFC", "09NFC", "09PFC", "10NFC", "10PFC", "11NFC", "11PFC", "12NFC", "12PFC", "16NFC", "16PFC")
```

```{r Subset start and finish samples from all gene sets}
denitRA_start <- denitRA[start_metaG_samples, ]
denitRA_finish <- denitRA[finish_metaG_samples, ]

methRA_start <- methRA[start_metaG_samples, ]
methRA_finish <- methRA[finish_metaG_samples, ]

carb_metabRA_start <- carb_metabRA[start_metaG_samples, ]
carb_metabRA_finish <- carb_metabRA[finish_metaG_samples, ]

cytoCoxRA_start <- cytoCoxRA[start_metaG_samples, ]
cytoCoxRA_finish <- cytoCoxRA[finish_metaG_samples, ]

SO4_S_AssimRA_start <- SO4_S_AssimRA[start_metaG_samples, ]
SO4_S_AssimRA_finish <- SO4_S_AssimRA[finish_metaG_samples, ]

Assim_SO4_redRA_start <- Assim_SO4_redRA[start_metaG_samples, ]
Assim_SO4_redRA_finish <- Assim_SO4_redRA[finish_metaG_samples, ]

Dissim_SO4_redRA_start <- Dissim_SO4_redRA[start_metaG_samples, ]
Dissim_SO4_redRA_finish <- Dissim_SO4_redRA[finish_metaG_samples, ]

ThioSOXRA_start <- ThioSOXRA[start_metaG_samples, ]
ThioSOXRA_finish <- ThioSOXRA[finish_metaG_samples, ]

DNRARA_start <- DNRARA[start_metaG_samples, ]
DNRARA_finish <- DNRARA[finish_metaG_samples, ]

ANRARA_start <- ANRARA[start_metaG_samples, ]
ANRARA_finish <- ANRARA[finish_metaG_samples, ]

NitrificRA_start <- NitrificRA[start_metaG_samples, ]
NitrificRA_finish <- NitrificRA[finish_metaG_samples, ]

COMAMMOXRA_start <- COMAMMOXRA[start_metaG_samples, ]
COMAMMOXRA_finish <- COMAMMOXRA[finish_metaG_samples, ]

NfixRA_start <- NfixRA[start_metaG_samples, ]
NfixRA_finish <- NfixRA[finish_metaG_samples, ]

CH4_oxRA_start <- CH4_oxRA[start_metaG_samples, ]
CH4_oxRA_finish <- CH4_oxRA[finish_metaG_samples, ]

mtrBC_TIGRRA_start <- mtrBC_TIGRRA[start_metaG_samples, ]
mtrBC_TIGRRA_finish <- mtrBC_TIGRRA[finish_metaG_samples, ]

FeGenieRA_start <- FeGenieRA[start_metaG_samples,]
FeGenieRA_finish <- FeGenieRA[finish_metaG_samples,]

SO4_red_combinedRA_start <- SO4_red_combinedRA[start_metaG_samples,]
SO4_red_combinedRA_finish <- SO4_red_combinedRA[finish_metaG_samples,]

FeReductionRA_start <- FeReductionRA[start_metaG_samples,]
FeReductionRA_finish <- FeReductionRA[finish_metaG_samples,]

FeOxidationRA_start <- FeOxidationRA[start_metaG_samples,]
FeOxidationRA_finish <- FeOxidationRA[finish_metaG_samples,]
```

```{r Dimension check for RA matrices}
#Check dimensions of each relative abundance (RA) matrix:
dim(denitRA) #24 10
dim(methRA) #24 44
dim(carb_metabRA) #24 120
dim(cytoCoxRA) #24 5
dim(SO4_S_AssimRA) #24 14
dim(Assim_SO4_redRA) #24 10
dim(Dissim_SO4_redRA) #24 5
dim(ThioSOXRA) #24 7
dim(DNRARA) #24 9
dim(ANRARA) #24 6
dim(NitrificRA) #24 4
dim(COMAMMOXRA) #24 6
dim(NfixRA) #24 8
dim(CH4_oxRA) #24 12
dim(mtrBC_TIGRRA) #24 2
dim(FeGenieRA) #24 13

#Check dimensions of RA start matrices
dim(denitRA_start) #8 10
dim(methRA_start) #8 44
dim(carb_metabRA_start) #8 120
dim(cytoCoxRA_start) #8 5
dim(SO4_S_AssimRA_start) #8 14
dim(Assim_SO4_redRA_start) #8 10
dim(Dissim_SO4_redRA_start) #8 5
dim(ThioSOXRA_start) #8 7
dim(DNRARA_start) #8 9
dim(ANRARA_start) #8 6
dim(NitrificRA_start) #8 4
dim(COMAMMOXRA_start) #8 6
dim(NfixRA_start) #8 8
dim(CH4_oxRA_start) #8 12
dim(mtrBC_TIGRRA_start) #8 2
dim(FeGenieRA_start) #8 13

#Check dimensions of RA finish matrices
dim(denitRA_finish) #16 10
dim(methRA_finish) #16 44
dim(carb_metabRA_finish) #16 120
dim(cytoCoxRA_finish) #16 5
dim(SO4_S_AssimRA_finish) #16 14
dim(Assim_SO4_redRA_finish) #16 10
dim(Dissim_SO4_redRA_finish) #16 5
dim(ThioSOXRA_finish) #16 7
dim(DNRARA_finish) #16 9
dim(ANRARA_finish) #16 6
dim(NitrificRA_finish) #16 4
dim(COMAMMOXRA_finish) #16 6
dim(NfixRA_finish) #16 8
dim(CH4_oxRA_finish) #16 12
dim(mtrBC_TIGRRA_finish) #16 2
dim(FeGenieRA_finish) #16 13
```

```{r modify 'design' rownames}
#Make metaG sample ID's the rownames on 'design'. Then drop design$Sample:
rownames(design) <- design$Sample
design <- design[,2:4]
```

```{r combine RA and design data frames}
# Make data frames from gene set relative abundance (RA) data frames and design
denitRA_design <- cbind(design, denitRA)
methRA_design <- cbind(design, methRA)
carb_metabRA_design <- cbind(design, carb_metabRA)
cytoCoxRA_design <- cbind(design, cytoCoxRA)
SO4_S_AssimRA_design <- cbind(design, SO4_S_AssimRA)
Assim_SO4_redRA_design <- cbind(design, Assim_SO4_redRA)
Dissim_SO4_redRA_design <- cbind(design, Dissim_SO4_redRA)
ThioSOXRA_design <- cbind(design, ThioSOXRA)
DNRARA_design <- cbind(design, DNRARA)
ANRARA_design <- cbind(design, ANRARA)
NitrificRA_design <- cbind(design, NitrificRA)
COMAMMOXRA_design <- cbind(design, COMAMMOXRA)
NfixRA_design <- cbind(design, NfixRA)
CH4_oxRA_design <- cbind(design, CH4_oxRA)
mtrBC_TIGRRA_design <- cbind(design, mtrBC_TIGRRA)
FeGenieRA_design <- cbind(design, FeGenieRA)

SO4_red_combinedRA_design <- cbind(design, SO4_red_combinedRA)

FeReductionRA_design <- cbind(design, FeReductionRA)

FeOxidationRA_design <- cbind(design, FeOxidationRA)

```

## Bray Curtis dissimilarity

Calculate Bray Curtis dissimilarity:
```{r relative abundance to Bray Curtis dissimilarity}
denitBC <- vegdist(denitRA, method = 'bray')
denitBC_start <- vegdist(denitRA_start, method = 'bray')
denitBC_finish <- vegdist(denitRA_finish, method = 'bray')

methBC <- vegdist(methRA, method = 'bray')
methBC_start <- vegdist(methRA_start, method = 'bray')
methBC_finish <- vegdist(methRA_finish, method = 'bray')

carb_metabBC <- vegdist(carb_metabRA, method = 'bray')
carb_metabBC_start <- vegdist(carb_metabRA_start, method = 'bray')
carb_metabBC_finish <- vegdist(carb_metabRA_finish, method = 'bray')

cytoCoxBC <- vegdist(cytoCoxRA, method = 'bray')
cytoCoxBC_start <- vegdist(cytoCoxRA_start, method = 'bray')
cytoCoxBC_finish <- vegdist(cytoCoxRA_finish, method = 'bray')

SO4_S_AssimBC <- vegdist(SO4_S_AssimRA, method = 'bray')
SO4_S_AssimBC_start <- vegdist(SO4_S_AssimRA_start, method = 'bray')
SO4_S_AssimBC_finish <- vegdist(SO4_S_AssimRA_finish, method = 'bray')

Assim_SO4_redBC <- vegdist(Assim_SO4_redRA, method = 'bray')
Assim_SO4_redBC_start <- vegdist(Assim_SO4_redRA_start, method = 'bray')
Assim_SO4_redBC_finish <- vegdist(Assim_SO4_redRA_finish, method = 'bray')

Dissim_SO4_redBC <- vegdist(Dissim_SO4_redRA, method = 'bray')
Dissim_SO4_redBC_start <- vegdist(Dissim_SO4_redRA_start, method = 'bray')
Dissim_SO4_redBC_finish <- vegdist(Dissim_SO4_redRA_finish, method = 'bray')

ThioSOXBC <- vegdist(ThioSOXRA, method = 'bray')
ThioSOXBC_start <- vegdist(ThioSOXRA_start, method = 'bray')
ThioSOXBC_finish <- vegdist(ThioSOXRA_finish, method = 'bray')

DNRABC <- vegdist(DNRARA, method = 'bray')
DNRABC_start <- vegdist(DNRARA_start, method = 'bray')
DNRABC_finish <- vegdist(DNRARA_finish, method = 'bray')

ANRABC <- vegdist(ANRARA, method = 'bray')
ANRABC_start <- vegdist(ANRARA_start, method = 'bray')
ANRABC_finish <- vegdist(ANRARA_finish, method = 'bray')

NitrificBC <- vegdist(NitrificRA, method = 'bray')
NitrificBC_start <- vegdist(NitrificRA_start, method = 'bray')
NitrificBC_finish <- vegdist(NitrificRA_finish, method = 'bray')

COMAMMOXBC <- vegdist(COMAMMOXRA, method = 'bray')
COMAMMOXBC_start <- vegdist(COMAMMOXRA_start, method = 'bray')
COMAMMOXBC_finish <- vegdist(COMAMMOXRA_finish, method = 'bray')

NfixBC <- vegdist(NfixRA, method = 'bray')
NfixBC_start <- vegdist(NfixRA_start, method = 'bray')
NfixBC_finish <- vegdist(NfixRA_finish, method = 'bray')

CH4_oxBC <- vegdist(CH4_oxRA, method = 'bray')
CH4_oxBC_start <- vegdist(CH4_oxRA_start, method = 'bray')
CH4_oxBC_finish <- vegdist(CH4_oxRA_finish, method = 'bray')

mtrBC_TIGRBC <- vegdist(mtrBC_TIGRRA, method = 'bray')
mtrBC_TIGRBC_start <- vegdist(mtrBC_TIGRRA_start, method = 'bray')
mtrBC_TIGRBC_finish <- vegdist(mtrBC_TIGRRA_finish, method = 'bray')

FeGenieBC <- vegdist(FeGenieRA, method = 'bray')
FeGenieBC_start <- vegdist(FeGenieRA_start, method = 'bray')
FeGenieBC_finish <- vegdist(FeGenieRA_finish, method = 'bray')

# Resume with SO4_red_combined, FeRed, and FeOx:
SO4_red_combinedBC <- vegdist(SO4_red_combinedRA, method = 'bray')
SO4_red_combinedBC_start <- vegdist(SO4_red_combinedRA_start, method = 'bray')
SO4_red_combinedBC_finish <- vegdist(SO4_red_combinedRA_finish, method = 'bray')

FeReductionBC <- vegdist(FeReductionRA, method = 'bray')
FeReductionBC_start <- vegdist(FeReductionRA_start, method = 'bray')
FeReductionBC_finish <- vegdist(FeReductionRA_finish, method = 'bray')

FeOxidationBC <- vegdist(FeOxidationRA, method = 'bray')
FeOxidationBC_start <- vegdist(FeOxidationRA_start, method = 'bray')
FeOxidationBC_finish <- vegdist(FeOxidationRA_finish, method = 'bray')
```

## GHGs (new flux analysis from SoilPhys_GHGs_16S.Rmd)
```{r GHG T1 concentration analysis}
# Version of ghg with labels for T1 through T4 analysis:
ghg2 <- read.csv("../data/TMG_GHG-Flux_Pub_3.csv", header=TRUE)

# FIX c2o TYPO IN ghg2:
colnames(ghg2)[18] <- "co2_ppm"

# ghg2, select T1 through T4, and all ppm measurements:
ghg_ppm <- ghg2 %>%
  dplyr::select('date', 'chamber', 'timepoint', 'plant', 'ch4_ppm', 'co2_ppm', 'n2o_ppm')%>%
  dplyr::filter(date !='7/24/2016')

# Set factors:
ghg_ppm$chamber <- as.factor(ghg_ppm$chamber)
ghg_ppm$date <- as.factor(ghg_ppm$date)

# Change design$box to design$chamber in order to join with ghg$chamber
colnames(meta)[2] <- "chamber"

ghg_ppm <- ghg_ppm %>%
  left_join(meta[,-c(3, 4, 7, 8, 9)], by="chamber")

# Add new date column, to shorten:
ghg_ppm$date_short <- ghg_ppm$date
# remove "/2016" from date_short:
ghg_ppm$date_short <- str_remove(ghg_ppm$date_short, "/2016")

# Separate data frames for "No Plant" and "Plant", for plotting:
ghg_ppm.noplant <- ghg_ppm %>%
  filter(plant=="N")

ghg_ppm.plant <- ghg_ppm %>%
  filter(plant=="P")
```

```{r filter only timepoint 1}
ghg_ppm_filtered <- ghg_ppm %>%
  dplyr::filter(timepoint == "T1")
```

```{r Calculate T0s from WRC static chamber monitoring}
# will attempt to use results from monthly WRC static chamber to calculate average T0. Then calculate a 2 point flux with T30 from TMG mesocosms.

# read in compiled T0 data:
WRC_T0s <- read.csv("../data/WRC_T0s_for_TMG_ambient_calculation.csv")

# Set Sample_Date as factor:
WRC_T0s$Sample_Date <- as.factor(WRC_T0s$Sample_Date)

# calculate averages within each month:
CO2_T0_monthly_avg <- aggregate(T0_CO2_ppm ~ Sample_Date, data = WRC_T0s, FUN = mean, na.rm = TRUE)
CH4_T0_monthly_avg <- aggregate(T0_CH4_ppm ~ Sample_Date, data = WRC_T0s, FUN = mean, na.rm = TRUE)
N2O_T0_monthly_avg <- aggregate(T0_N2O_ppm ~ Sample_Date, data = WRC_T0s, FUN = mean, na.rm = TRUE)

# Average the two years of August data:
CO2_T0_monthly_avg[nrow(CO2_T0_monthly_avg)+1, "T0_CO2_ppm"] <- mean(c(CO2_T0_monthly_avg[1, "T0_CO2_ppm"], CO2_T0_monthly_avg[2, "T0_CO2_ppm"]), na.rm = TRUE)

CH4_T0_monthly_avg[nrow(CH4_T0_monthly_avg)+1, "T0_CH4_ppm"] <- mean(c(CH4_T0_monthly_avg[1, "T0_CH4_ppm"], CH4_T0_monthly_avg[2, "T0_CH4_ppm"]), na.rm = TRUE)

N2O_T0_monthly_avg[nrow(N2O_T0_monthly_avg)+1, "T0_N2O_ppm"] <- mean(c(N2O_T0_monthly_avg[1, "T0_N2O_ppm"], N2O_T0_monthly_avg[2, "T0_N2O_ppm"]), na.rm = TRUE)

# Add in the name:
CO2_T0_monthly_avg$Sample_Date <- factor(c("Aug_2022", "August_2021", "Jul_2022", "June_2022", "August_21_22"))
CH4_T0_monthly_avg$Sample_Date <- factor(c("Aug_2022", "August_2021", "Jul_2022", "June_2022", "August_21_22"))
N2O_T0_monthly_avg$Sample_Date <- factor(c("Aug_2022", "August_2021", "Jul_2022", "June_2022", "August_21_22"))
```

```{r wrangle T0 averages for flux calculation}
# bind together the three GHGs:
T0_averages_combined <- CH4_T0_monthly_avg %>%
  cbind(CO2_T0_monthly_avg$T0_CO2_ppm) %>%
  cbind(N2O_T0_monthly_avg$T0_N2O_ppm)

# drop the two August monthly:
T0_averages_combined <- T0_averages_combined[3:5,]

# Rename 'Sample_Date' in T0_averages_combined to 'date' to match the 'ghg_flux' column
colnames(T0_averages_combined)[1] <- "date"

# Add chamber as NAs:
T0_averages_combined$chamber <- NA

# Rename GHG columns to match ghg_flux:
colnames(T0_averages_combined)[2:4] <- c("ch4_ppm", "co2_ppm", "n2o_ppm")

# Add time columns to match ghg_flux:
T0_averages_combined$time_min <- 0
T0_averages_combined$time <- 0

# Add month column:
T0_averages_combined$month <- c("July", "June", "August")
```


```{r GHG flux wrangling}
# Format a dataframe for 'gasfluxes' package:
ghg_flux <- ghg2[,c(1:2, 12, 15:19)]

# Remove 7/24/2016 from data frame, as per Gina's advice
ghg_flux <- ghg_flux %>%
  dplyr::filter(date !='7/24/2016')

# Combine 'date' and 'chamber' strings to create a unique ID:
ghg_flux$ID <- paste(ghg_flux$date, ghg_flux$chamber)

# convert Volume (cm^3) to Volume (m^3):
ghg_flux$V <- (ghg_flux$Vol..cm3. * (1/1e+06))

# Copy 'Area..m2.' to a new column named 'A':
ghg_flux$A <- ghg_flux$Area..m2.

# Convert time in minutes to time in hours:
ghg_flux$time <- (ghg_flux$time_min * (1/60))

# Bind T0s from WRC:
ghg_flux_w_T0 <- ghg_flux

# Add in month column that takes date strings and matches them to "July", "June", "August":
# Create a vector of month names
month_names <- c("January", "February", "March", "April", "May", "June",
                 "July", "August", "September", "October", "November", "December")

# Extract the month as a number and convert it to a month name
ghg_flux_w_T0 <- ghg_flux_w_T0 %>%
  mutate(
    month = month_names[as.numeric(substr(date, 1, 1))]
  )

# Copy July and June rows because there are two July and June sampling dates:
T0_averages_combined <- rbind(T0_averages_combined, T0_averages_combined[1,])
T0_averages_combined <- rbind(T0_averages_combined, T0_averages_combined[2,])

# Add TMG-style dates to match with ghg_flux:
T0_averages_combined$date_TMG <- c("7/11/2016", "6/13/2016", "8/11/2016", "7/25/2016", "6/28/2016")

# For loop:
for (unique_date in unique(ghg_flux_w_T0$date)) {
  for (unique_chamber in unique(ghg_flux_w_T0$chamber)) {
    
    # Filter the monthly average corresponding to the current date
    month_avg <- T0_averages_combined %>% filter(date_TMG == unique_date)
    
    # Skip if there is no corresponding monthly average for the date
    if (nrow(month_avg) == 0) next
    
    # Create a new row with time = 0 and the averages for the current date and chamber
    new_row <- data.frame(
      date = unique_date,
      chamber = unique_chamber,
      time = month_avg$time,
      time_min = month_avg$time_min,
      ch4_ppm = month_avg$ch4_ppm,
      co2_ppm = month_avg$co2_ppm,
      n2o_ppm = month_avg$n2o_ppm,
      month = month_avg$month
    )
    
    # Bind the new row to ghg_flux
    ghg_flux_w_T0 <- bind_rows(ghg_flux_w_T0, new_row)
  }
}

# Reorder rows by date, chamber, and time:
ghg_flux_w_T0 <- ghg_flux_w_T0 %>%
  arrange(date, chamber, time)

# Copy some of rows following T0s onto T0s (ID, V, A):
# Define the columns with missing values to be filled from the row below
columns_to_fill <- c("ID", "V", "A")

# Loop through the rows from the bottom to the top to fill missing values from the row below
for (i in (nrow(ghg_flux_w_T0) - 1):1) {
  # Check for missing values in the specified columns
  missing_values <- is.na(ghg_flux_w_T0[i, columns_to_fill])
  
  # If there are missing values, copy the values from the row below
  if (any(missing_values)) {
    ghg_flux_w_T0[i, columns_to_fill][missing_values] <- ghg_flux_w_T0[i + 1, columns_to_fill][missing_values]
  }
}



# Convert ppm to mg/m^3, based on https://www.cdc.gov/niosh/docs/2004-101/calc.html (based on 25°C and 1 atm):
# y mg/m3 = (x ppm)(molar mass)/24.45
  # 24.45 is the volume (liters) of a mole (gram molecular weight) of a gas or vapor when the pressure is at 1 atmosphere (760 torr or 760 mm Hg) and at 25°C

# CH4 mg/m3
ghg_flux$C_ch4 <- ((ghg_flux$ch4_ppm * 16.04)/24.45) # C is the package 'gasfluxes' abbreviation for concentration. 16.04 g/mol = molar mass of CH4
ghg_flux_w_T0$C_ch4 <- ((ghg_flux_w_T0$ch4_ppm * 16.04)/24.45)

# CO2 mg/m3
ghg_flux$C_co2 <- ((ghg_flux$co2_ppm * 44.01)/24.45) # molar mass of CO2 = 44.01 g/mol
ghg_flux_w_T0$C_co2 <- ((ghg_flux_w_T0$co2_ppm * 44.01)/24.45)

# N2O mg/m3
ghg_flux$C_n2o <-  ((ghg_flux$n2o_ppm * 44.013)/24.45) # molar mass of N2O = 44.013 g/mol
ghg_flux_w_T0$C_n2o <-  ((ghg_flux_w_T0$n2o_ppm * 44.013)/24.45)

# Split off data frames for each greenhouse gas:
ch4_flux_input <- ghg_flux[,9:13]
ch4_flux_input_w_T0 <- ghg_flux_w_T0[,c(9:12, 14)]

co2_flux_input <- ghg_flux[,c(9:12, 14)]
co2_flux_input_w_T0 <- ghg_flux_w_T0[,c(9:12, 15)]

n2o_flux_input <- ghg_flux[,c(9:12, 15)]
n2o_flux_input_w_T0 <- ghg_flux_w_T0[,c(9:12, 16)]

# Rename C_ghg columns to 'C':
colnames(ch4_flux_input)[5] <- "C"
colnames(ch4_flux_input_w_T0)[5] <- "C"

colnames(co2_flux_input)[5] <- "C"
colnames(co2_flux_input_w_T0)[5] <- "C"

colnames(n2o_flux_input)[5] <- "C"
colnames(n2o_flux_input_w_T0)[5] <- "C"

# Must omit NAs before using gasfluxes():
ch4_flux_input <- na.omit(ch4_flux_input)
ch4_flux_input_w_T0 <- na.omit(ch4_flux_input_w_T0)

co2_flux_input <- na.omit(co2_flux_input)
co2_flux_input_w_T0 <- na.omit(co2_flux_input_w_T0)

# n2o_flux_input has negative values, will need to be converted to NAs, then all NAs omitted
n2o_flux_input$C[n2o_flux_input$C<0] <- NA
n2o_flux_input <- na.omit(n2o_flux_input)

n2o_flux_input_w_T0$C[n2o_flux_input_w_T0$C<0] <- NA
n2o_flux_input_w_T0 <- na.omit(n2o_flux_input_w_T0)
```

```{r Make T0 flux dataframes' V and A numeric}
ch4_flux_input_w_T0$V <- as.numeric(ch4_flux_input_w_T0$V)
ch4_flux_input_w_T0$A <- as.numeric(ch4_flux_input_w_T0$A)

co2_flux_input_w_T0$V <- as.numeric(co2_flux_input_w_T0$V)
co2_flux_input_w_T0$A <- as.numeric(co2_flux_input_w_T0$A)

n2o_flux_input_w_T0$V <- as.numeric(n2o_flux_input_w_T0$V)
n2o_flux_input_w_T0$A <- as.numeric(n2o_flux_input_w_T0$A)
```

```{r flux input DFs with 3 point flux}
# Ambient to T30 is the reviewer recommended approach:
# Filter dataframes to only have time = 0, time = 0.5, and time = 1.0
ch4_flux_input_3point <- ch4_flux_input_w_T0 %>%
  dplyr::filter(time == 0.0 | time == 0.5 | time == 1.0)

co2_flux_input_3point <- co2_flux_input_w_T0 %>%
  dplyr::filter(time == 0.0 | time == 0.5 | time == 1.0)

n2o_flux_input_3point <- n2o_flux_input_w_T0 %>%
  dplyr::filter(time == 0.0 | time == 0.5 | time == 1.0)
```

```{r GHG flux calculation}
ch4_flux.results <- gasfluxes(ch4_flux_input, method = c("linear","robust linear", "HMR"), plot = T)
ch4_flux.results_w_T0 <- gasfluxes(ch4_flux_input_w_T0, method = c("linear","robust linear", "HMR"), plot = T)
ch4_flux.results_3pt <- gasfluxes(ch4_flux_input_3point, method = c("linear","robust linear", "HMR"), plot = T)

co2_flux.results <- gasfluxes(co2_flux_input, method = c("linear","robust linear", "HMR"), plot = T)
co2_flux.results_w_T0 <- gasfluxes(co2_flux_input_w_T0, method = c("linear","robust linear", "HMR"), plot = T)
co2_flux.results_3pt <- gasfluxes(co2_flux_input_3point, method = c("linear","robust linear", "HMR"), plot = T)

n2o_flux.results <- gasfluxes(n2o_flux_input, method = c("linear","robust linear", "HMR"), plot = T)
n2o_flux.results_w_T0 <- gasfluxes(n2o_flux_input_w_T0, method = c("linear","robust linear", "HMR"), plot = T)
n2o_flux.results_3pt <- gasfluxes(n2o_flux_input_3point, method = c("linear","robust linear", "HMR"), plot = T)
```

```{r GHG flux results wrangling}
# If HMR is available, use HMR
# If HMR is not available, use robust linear
# If robust linear is not available use linear
# New DF with all three f0s:
ch4_fluxes_only <- ch4_flux.results[,c(1:2, 11, 17)]
ch4_fluxes_only_w_T0 <- ch4_flux.results_w_T0[,c(1:2, 11, 17)]
ch4_fluxes_only_3pt <- ch4_flux.results_3pt[,c(1:2, 11, 17)]

co2_fluxes_only <- co2_flux.results[,c(1:2, 11, 17)]
co2_fluxes_only_w_T0 <- co2_flux.results_w_T0[,c(1:2, 11, 17)]
co2_fluxes_only_3pt <- co2_flux.results_3pt[,c(1:2, 11, 17)]

n2o_fluxes_only <- n2o_flux.results[,c(1:2, 11, 17)]
n2o_fluxes_only_w_T0 <- n2o_flux.results_w_T0[,c(1:2, 11, 17)]
n2o_fluxes_only_3pt <- n2o_flux.results_3pt[,c(1:2, 11, 17)]

# New column for final flux selection:
ch4_fluxes_only$f0_selected <- ch4_fluxes_only$HMR.f0
ch4_fluxes_only_w_T0$f0_selected <- ch4_fluxes_only_w_T0$HMR.f0
ch4_fluxes_only_3pt$f0_selected <- ch4_fluxes_only_3pt$HMR.f0

co2_fluxes_only$f0_selected <- co2_fluxes_only$HMR.f0
co2_fluxes_only_w_T0$f0_selected <- co2_fluxes_only_w_T0$HMR.f0
co2_fluxes_only_3pt$f0_selected <- co2_fluxes_only_3pt$HMR.f0

n2o_fluxes_only$f0_selected <- n2o_flux.results$HMR.f0
n2o_fluxes_only_w_T0$f0_selected <- n2o_flux.results_w_T0$HMR.f0
n2o_fluxes_only_3pt$f0_selected <- n2o_flux.results_3pt$HMR.f0

# Where HMR = NA, replace with the robust linear value:
ch4_fluxes_only$f0_selected[na.action(na.omit(ch4_fluxes_only$HMR.f0))] <- ch4_fluxes_only$robust.linear.f0[na.action(na.omit(ch4_fluxes_only$HMR.f0))]

ch4_fluxes_only_w_T0$f0_selected[na.action(na.omit(ch4_fluxes_only_w_T0$HMR.f0))] <- ch4_fluxes_only_w_T0$robust.linear.f0[na.action(na.omit(ch4_fluxes_only_w_T0$HMR.f0))]

ch4_fluxes_only_3pt$f0_selected[na.action(na.omit(ch4_fluxes_only_3pt$HMR.f0))] <- ch4_fluxes_only_3pt$robust.linear.f0[na.action(na.omit(ch4_fluxes_only_3pt$HMR.f0))]

co2_fluxes_only$f0_selected[na.action(na.omit(co2_fluxes_only$HMR.f0))] <- co2_fluxes_only$robust.linear.f0[na.action(na.omit(co2_fluxes_only$HMR.f0))]

co2_fluxes_only_w_T0$f0_selected[na.action(na.omit(co2_fluxes_only_w_T0$HMR.f0))] <- co2_fluxes_only_w_T0$robust.linear.f0[na.action(na.omit(co2_fluxes_only_w_T0$HMR.f0))]

co2_fluxes_only_3pt$f0_selected[na.action(na.omit(co2_fluxes_only_3pt$HMR.f0))] <- co2_fluxes_only_3pt$robust.linear.f0[na.action(na.omit(co2_fluxes_only_3pt$HMR.f0))]

n2o_fluxes_only$f0_selected[na.action(na.omit(n2o_fluxes_only$HMR.f0))] <- n2o_fluxes_only$robust.linear.f0[na.action(na.omit(n2o_fluxes_only$HMR.f0))]

n2o_fluxes_only_w_T0$f0_selected[na.action(na.omit(n2o_fluxes_only_w_T0$HMR.f0))] <- n2o_fluxes_only_w_T0$robust.linear.f0[na.action(na.omit(n2o_fluxes_only_w_T0$HMR.f0))]

n2o_fluxes_only_3pt$f0_selected[na.action(na.omit(n2o_fluxes_only_3pt$HMR.f0))] <- n2o_fluxes_only_3pt$robust.linear.f0[na.action(na.omit(n2o_fluxes_only_3pt$HMR.f0))]

# Where f0_selected still has NAs, replace with the linear.f0:
ch4_fluxes_only$f0_selected[na.action(na.omit(ch4_fluxes_only$f0_selected))] <- ch4_fluxes_only$linear.f0[na.action(na.omit(ch4_fluxes_only$f0_selected))]

ch4_fluxes_only_w_T0$f0_selected[na.action(na.omit(ch4_fluxes_only_w_T0$f0_selected))] <- ch4_fluxes_only_w_T0$linear.f0[na.action(na.omit(ch4_fluxes_only_w_T0$f0_selected))]

ch4_fluxes_only_3pt$f0_selected[na.action(na.omit(ch4_fluxes_only_3pt$f0_selected))] <- ch4_fluxes_only_3pt$linear.f0[na.action(na.omit(ch4_fluxes_only_3pt$f0_selected))]

co2_fluxes_only$f0_selected[na.action(na.omit(co2_fluxes_only$f0_selected))] <- co2_fluxes_only$linear.f0[na.action(na.omit(co2_fluxes_only$f0_selected))]

co2_fluxes_only_w_T0$f0_selected[na.action(na.omit(co2_fluxes_only_w_T0$f0_selected))] <- co2_fluxes_only_w_T0$linear.f0[na.action(na.omit(co2_fluxes_only_w_T0$f0_selected))]

co2_fluxes_only_3pt$f0_selected[na.action(na.omit(co2_fluxes_only_3pt$f0_selected))] <- co2_fluxes_only_3pt$linear.f0[na.action(na.omit(co2_fluxes_only_3pt$f0_selected))]

n2o_fluxes_only$f0_selected[na.action(na.omit(n2o_fluxes_only$f0_selected))] <- n2o_fluxes_only$linear.f0[na.action(na.omit(n2o_fluxes_only$f0_selected))]

n2o_fluxes_only_w_T0$f0_selected[na.action(na.omit(n2o_fluxes_only_w_T0$f0_selected))] <- n2o_fluxes_only_w_T0$linear.f0[na.action(na.omit(n2o_fluxes_only_w_T0$f0_selected))]

n2o_fluxes_only_3pt$f0_selected[na.action(na.omit(n2o_fluxes_only_3pt$f0_selected))] <- n2o_fluxes_only_3pt$linear.f0[na.action(na.omit(n2o_fluxes_only_3pt$f0_selected))]

# Combine separate DFs back into one DF for statistics
select_fluxes_combined <- ch4_fluxes_only[,c(1,5)] # start with CH4 IDs and f0 for a base dataframe
select_fluxes_combined_w_T0 <- ch4_fluxes_only_w_T0[,c(1,5)]
select_fluxes_combined_3pt <- ch4_fluxes_only_3pt[,c(1,5)]

colnames(select_fluxes_combined)[2] <- "f0_CH4" # rename to distinguish between GHGs
colnames(select_fluxes_combined_w_T0)[2] <- "f0_CH4"
colnames(select_fluxes_combined_3pt)[2] <- "f0_CH4"

select_fluxes_combined$f0_CO2 <- co2_fluxes_only$f0_selected # add on CO2 fluxes
select_fluxes_combined_w_T0$f0_CO2 <- co2_fluxes_only_w_T0$f0_selected
select_fluxes_combined_3pt$f0_CO2 <- co2_fluxes_only_3pt$f0_selected

select_fluxes_combined$f0_N2O <- n2o_fluxes_only$f0_selected # add on N2O fluxes
select_fluxes_combined_w_T0$f0_N2O <- n2o_fluxes_only_w_T0$f0_selected
select_fluxes_combined_3pt$f0_N2O <- n2o_fluxes_only_3pt$f0_selected

# Change order of select_fluxes_combined_w_T0 to match the other dataframes
select_fluxes_combined_w_T0 <- select_fluxes_combined_w_T0[match(select_fluxes_combined$ID, select_fluxes_combined_w_T0$ID), ]
select_fluxes_combined_3pt <- select_fluxes_combined_3pt[match(select_fluxes_combined$ID, select_fluxes_combined_3pt$ID), ]
# CHECK ROW ORDERS NOW

# Add date_short back onto DFs for plotting
select_fluxes_combined$date_short <- ghg_ppm_filtered$date_short
select_fluxes_combined_w_T0$date_short <- ghg_ppm_filtered$date_short
select_fluxes_combined_3pt$date_short <- ghg_ppm_filtered$date_short

# Add on 'plant'
select_fluxes_combined$plant <- ghg_ppm_filtered$plant
select_fluxes_combined_w_T0$plant <- ghg_ppm_filtered$plant
select_fluxes_combined_3pt$plant <- ghg_ppm_filtered$plant

# Add on 'treatment'
select_fluxes_combined$treatment <- ghg_ppm_filtered$treatment
select_fluxes_combined_w_T0$treatment <- ghg_ppm_filtered$treatment
select_fluxes_combined_3pt$treatment <- ghg_ppm_filtered$treatment

# Add on 'history'
select_fluxes_combined$history <- ghg_ppm_filtered$history
select_fluxes_combined_w_T0$history <- ghg_ppm_filtered$history
select_fluxes_combined_3pt$history <- ghg_ppm_filtered$history
```

## GHG wrangling (previous T30 analysis)

Wrangling greenhouse gas (GHG) data:
```{r wrangling GHG data}
# Fix 'c2o' typo:
colnames(GHG)[18] <- "co2_ppm"

#Create a new GHG data frame with less columns:
#GHG_less_columns <- cbind(GHG[,1:4], GHG[24:32])
GHG_less_columns <- cbind(GHG[,1:4], GHG[17:19])

# Select only T1 rows, and not July 24 (night sampling):
#GHG_less_columns <- GHG_less_columns %>% drop_na()
GHG_less_columns <- GHG_less_columns %>%
  dplyr::filter(timepoint == "T1" & date !='7/24/2016')

#Change column name "chamber" to "box" to match with design2:
colnames(GHG_less_columns)[2] <- "box"

#Convert GHG_less_columns$plant to lowercase to match with design2$plant:
GHG_less_columns$plant <- tolower(GHG_less_columns$plant)

#Subset GHG_less_columns for 'start' time point (6/13/2016) and 'finish' time point (8/11/2016):
GHG_start <- dplyr::filter(GHG_less_columns, date == "6/13/2016")
GHG_finish <- dplyr::filter(GHG_less_columns, date == "8/11/2016")

#Add design2 into GHG 'start' and 'finish':
GHG_start <- left_join(GHG_start, design2, by = c("box", "plant"))
GHG_finish <- left_join(GHG_finish, design2, by = c("box", "plant"))

#Subset GHG_finish and GHG_start to only have metaG sample numbers:
GHG_start_metaG <- rbind(GHG_start[5:6,], GHG_start[11:12,], GHG_start[15:24,], GHG_start[31:32,], make.row.names = FALSE)
GHG_finish_metaG <- rbind(GHG_finish[5:6,], GHG_finish[11:12,], GHG_finish[15:24,], GHG_finish[31:32,], make.row.names = FALSE)

#To align "pre" metaG samples with a single, initial GHG value (from the start GHG measurements on 6/28/206), average together GHG samples in the same box number, then transpose and set row name to corresponding metaG Sample ID:
GHG_start_03 <- data.frame(colMeans(GHG_start_metaG[1:2, 5:7]))
GHG_start_03 <- as.data.frame(t(GHG_start_03))
row.names(GHG_start_03) <- "03Pre"

GHG_start_06 <- data.frame(colMeans(GHG_start_metaG[3:4, 5:7]))
GHG_start_06 <- as.data.frame(t(GHG_start_06))
row.names(GHG_start_06) <- "06Pre"

GHG_start_08 <- data.frame(colMeans(GHG_start_metaG[5:6, 5:7]))
GHG_start_08 <- as.data.frame(t(GHG_start_08))
row.names(GHG_start_08) <- "08Pre"

GHG_start_09 <- data.frame(colMeans(GHG_start_metaG[7:8, 5:7]))
GHG_start_09 <- as.data.frame(t(GHG_start_09))
row.names(GHG_start_09) <- "09Pre"

GHG_start_10 <- data.frame(colMeans(GHG_start_metaG[9:10, 5:7]))
GHG_start_10 <- as.data.frame(t(GHG_start_10))
row.names(GHG_start_10) <- "10Pre"

GHG_start_11 <- data.frame(colMeans(GHG_start_metaG[11:12, 5:7]))
GHG_start_11 <- as.data.frame(t(GHG_start_11))
row.names(GHG_start_11) <- "11Pre"

GHG_start_12 <- data.frame(colMeans(GHG_start_metaG[13:14, 5:7]))
GHG_start_12 <- as.data.frame(t(GHG_start_12))
row.names(GHG_start_12) <- "12Pre"

GHG_start_16 <- data.frame(colMeans(GHG_start_metaG[15:16, 5:7]))
GHG_start_16 <- as.data.frame(t(GHG_start_16))
row.names(GHG_start_16) <- "16Pre"

#Then rbind() all the dfs together
GHG_start_metaG_avg <- rbind(GHG_start_03, GHG_start_06, GHG_start_08, GHG_start_09, GHG_start_10, GHG_start_11, GHG_start_12, GHG_start_16)

#Set the row names of GHG data to the metaG sample names, so that stats functions can compare the GHG and metaG data sets:
row.names(GHG_finish_metaG) <- c("03NFC", "03PFC", "06NFC", "06PFC", "08NFC", "08PFC", "09NFC","09PFC", "10NFC","10PFC", "11NFC", "11PFC", "12NFC", "12PFC", "16NFC", "16PFC")

#Combine GHG_start_metaG_avg and GHG_finish_metaG[,5:7] to have all 24 samples:
GHG_metaG <- rbind(GHG_start_metaG_avg, GHG_finish_metaG[,5:7])
GHG_metaG <- GHG_metaG[match(sample_order, rownames(GHG_metaG)), ]

#Clean up GHG_finish_metaG for envfitting:
GHG_finish_envfit <- GHG_finish_metaG %>%
  dplyr::select(-c(date, timepoint, source, ttreat, hhistory))
```

```{r Combined GHG and Soil FINISH data frames for envfit}
#Subset of SFO.by_time_plant_treatment that is represented in GHG_finish_envfit:
SFO.envfit <- SFO.by_time_plant_treatment %>%
  dplyr::filter(chamber %in% GHG_finish_envfit$sample)

# Change 'box' to 'chamber' to enable merging:
colnames(GHG_finish_envfit)[6] <- "chamber"

# Match lowercase to enable merging
SFO.envfit$plant <- tolower(SFO.envfit$plant)
SFO.envfit$history <- tolower(SFO.envfit$history)
SFO.envfit$treatment <- tolower(SFO.envfit$treatment)
GHG_finish_envfit$treatment <- tolower(GHG_finish_envfit$treatment)
GHG_finish_envfit$history <- tolower(GHG_finish_envfit$history)

# Merge soil and greenhouse gas data:
envfit_FINISH <- merge(GHG_finish_envfit, SFO.envfit)

# Order by chamber, to match with metaG/GHG sample order:
envfit_FINISH <- envfit_FINISH[order(envfit_FINISH$chamber),]

# Set rownames to metaG/GHG rownames; this will be required for envfit
rownames(envfit_FINISH) <- rownames(GHG_finish_envfit)

# Now subset to only numbers:
envfit_FINISH_numerical <- envfit_FINISH %>%
  dplyr::select(where(is.numeric))
```

```{r Combine GHG and Soil START&FINISH for envfit}
#Filter SSF to only get GHG samples ("boxes")
# Using a brute-force approach:
SSF.2.0 <- SSF.by_time_plant_treatment[-c(1:4, 7:10, 13:14, 25:30, 33:40, 43:46, 49:50, 61:66, 69:72),]

# Need to average together 'start' samples again to make 'pre'. Use GHG code:
SSF_start_03 <- data.frame(colMeans(SSF.2.0[17:18, 6:8]))
SSF_start_03 <- as.data.frame(t(SSF_start_03))
row.names(SSF_start_03) <- "03Pre"

SSF_start_06 <- data.frame(colMeans(SSF.2.0[19:20, 6:8]))
SSF_start_06 <- as.data.frame(t(SSF_start_06))
row.names(SSF_start_06) <- "06Pre"

SSF_start_08 <- data.frame(colMeans(SSF.2.0[21:22, 6:8]))
SSF_start_08 <- as.data.frame(t(SSF_start_08))
row.names(SSF_start_08) <- "08Pre"

SSF_start_09 <- data.frame(colMeans(SSF.2.0[23:24, 6:8]))
SSF_start_09 <- as.data.frame(t(SSF_start_09))
row.names(SSF_start_09) <- "09Pre"

SSF_start_10 <- data.frame(colMeans(SSF.2.0[25:26, 6:8]))
SSF_start_10 <- as.data.frame(t(SSF_start_10))
row.names(SSF_start_10) <- "10Pre"

SSF_start_11 <- data.frame(colMeans(SSF.2.0[27:28, 6:8]))
SSF_start_11 <- as.data.frame(t(SSF_start_11))
row.names(SSF_start_11) <- "11Pre"

SSF_start_12 <- data.frame(colMeans(SSF.2.0[29:30, 6:8]))
SSF_start_12 <- as.data.frame(t(SSF_start_12))
row.names(SSF_start_12) <- "12Pre"

SSF_start_16 <- data.frame(colMeans(SSF.2.0[31:32, 6:8]))
SSF_start_16 <- as.data.frame(t(SSF_start_16))
row.names(SSF_start_16) <- "16Pre"

#Then rbind() all the dfs together
SSF_start_metaG_avg <- rbind(SSF_start_03, SSF_start_06, SSF_start_08, SSF_start_09, SSF_start_10, SSF_start_11, SSF_start_12, SSF_start_16)

# Subset Finish samples from SSF.2.0:
SSF.2.0.FINISH <- as.data.frame(SSF.2.0[1:16, ])

# Add column to SSF.2.0.FINISH with the metaG sample IDs:
SSF.2.0.FINISH$sample <- c("03NFC", "03PFC", "06NFC", "06PFC", "08NFC", "08PFC", "09NFC","09PFC", "10NFC","10PFC", "11NFC", "11PFC", "12NFC", "12PFC", "16NFC", "16PFC")

# Make $sample rownames:
rownames(SSF.2.0.FINISH) <- SSF.2.0.FINISH$sample

# Subset SSF.2.0.FINISH to the same dimensions as SSF_start_metaG_avg:
SSF.FINISH.forMerging <- as.data.frame(SSF.2.0.FINISH[,6:8])
rownames(SSF.FINISH.forMerging) <- SSF.2.0.FINISH$sample

#Combine GHG_start_metaG_avg and GHG_finish_metaG[,5:13] to have all 24 samples:
Soil_StartFinish_forEnvfit <- rbind(SSF_start_metaG_avg, SSF.FINISH.forMerging)
Soil_StartFinish_forEnvfit <- Soil_StartFinish_forEnvfit[match(sample_order, rownames(Soil_StartFinish_forEnvfit)), ]

# combine GHG_metaG and SSF
SoilGHG_StartFinish_Envfit <- cbind(GHG_metaG, Soil_StartFinish_forEnvfit)
```


# Statistics

## dbplsr

Distance-based partial least squares regression:
```{r dbplsr}
# CO2 ~ Assimilatory sulfate reduction:
start.Assim_SO4_red.dbplsr <- dbplsr(GHG_start_metaG_avg$co2_ppm ~ as.matrix(Assim_SO4_redBC_start), ncomp = 7, method = "GCV")
summary(start.Assim_SO4_red.dbplsr)
#plot(start.Assim_SO4_red.dbplsr)

finish.Assim_SO4_red.dbplsr <- dbplsr(GHG_finish_metaG$co2_ppm ~ as.matrix(Assim_SO4_redBC_finish), ncomp = 15, method = "GCV")
summary(finish.Assim_SO4_red.dbplsr)
#plot(finish.Assim_SO4_red.dbplsr)

# CH4 ~ Dissimilatory sulfate reduction:
start.Dissim_SO4_red.dbplsr <- dbplsr(GHG_start_metaG_avg$ch4_ppm~ as.matrix(Dissim_SO4_redBC_start), ncomp = 7, method = "GCV")
summary(start.Dissim_SO4_red.dbplsr)
#plot(start.Dissim_SO4_red.dbplsr)

finish.Dissim_SO4_red.dbplsr <- dbplsr(GHG_finish_metaG$ch4_ppm ~ as.matrix(Dissim_SO4_redBC_finish), ncomp = 15, method = "GCV")
summary(finish.Dissim_SO4_red.dbplsr)
#plot(finish.Dissim_SO4_red.dbplsr)

# CH4 ~ Fe reduction:
start.FeReduction.dbplsr <- dbplsr(GHG_start_metaG_avg$ch4_ppm ~ as.matrix(FeReductionBC_start), ncomp = 7, method = "GCV")
summary(start.FeReduction.dbplsr)
#plot(start.FeReduction.dbplsr)

finish.FeReduction.dbplsr <- dbplsr(GHG_finish_metaG$ch4_ppm ~ as.matrix(FeReductionBC_finish), ncomp = 15, method = "GCV")
summary(finish.FeReduction.dbplsr)
#plot(finish.FeReduction.dbplsr)
```

```{r DBPLSR barplots}
# read in formatted table of dbplsr component 2 adjusted R2 results:
dbplsr_comp2 <- read.csv("~/GitHub/WetlandMesocosm_GHG_Timberlake/data/dbplsr_comp2.csv")


# Methane:
dbplsr_meth_modules <- dplyr::filter(dbplsr_comp2, dbplsr_comp2$GHG == "CH4", .preserve = T)
dbplsr_meth_modules <- dbplsr_meth_modules[-c(3,4),] #remove soil and 16S

methane.dbplsr.plot <- ggplot(data = dbplsr_meth_modules, aes(x=Timepoint, y=Comp2_adjR2, fill = Dist_matrix)) +
  geom_bar(stat = "identity", position = position_dodge())+
  scale_fill_manual(values=c("#009E73","#D55E00","#0072B2"), name = "Functional Genes")+
  theme_bw() +
  #Remove plot grid lines
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  #Set axis title and text properties, tick marks, and labels
  theme(text=element_text(size=14),axis.title=element_text(size=14,face="bold"),
          axis.text=element_text(size=14),  
          axis.title.y=element_text(margin=margin(r=10)),
          panel.border = element_rect(colour = "black",size=1),strip.text = element_text(size = 14), legend.position="right", legend.title = element_text(size=14), legend.text=element_text(size=14)) + 
          theme(axis.ticks.length=unit(0.3,"cm")) + 
          labs(x = "Timepoint", y = "% Variance Explained: Methane", title="") +
  theme(rect=element_rect(fill="transparent"))+
  theme(plot.background = element_rect(color=NA))

methane.dbplsr.plot

#ggsave("../figures/Thesis/methane_dbplsr_plot.png", plot=methane.dbplsr.plot, device="png", path=NULL, scale=1, width=7, height=4, dpi=300, limitsize=TRUE, bg = "white")

# CO2:
dbplsr_CO2_modules <- dplyr::filter(dbplsr_comp2, dbplsr_comp2$GHG == "CO2", .preserve = T)
dbplsr_CO2_modules <- dbplsr_CO2_modules[-c(5,6),] #remove soil and 16S

CO2.dbplsr.plot <- ggplot(data = dbplsr_CO2_modules, aes(x=Timepoint, y=Comp2_adjR2, fill = Dist_matrix)) +
  geom_bar(stat = "identity", position = position_dodge())+
  scale_fill_manual(values=c("#F0E442","#56B4E9","#CC79A7"), name = "Functional Genes")+
  theme_bw() +
  #Remove plot grid lines
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  #Set axis title and text properties, tick marks, and labels
  theme(text=element_text(size=14),axis.title=element_text(size=14,face="bold"),
          axis.text=element_text(size=14),  
          axis.title.y=element_text(margin=margin(r=10)),
          panel.border = element_rect(colour = "black",size=1),strip.text = element_text(size = 14), legend.position="right", legend.title = element_text(size=14), legend.text=element_text(size=14)) + 
          theme(axis.ticks.length=unit(0.3,"cm")) + 
          labs(x = "Timepoint", y = "% Variance Explained: Carbon Dioxide", title="") +
  theme(rect=element_rect(fill="transparent"))+
  theme(plot.background = element_rect(color=NA))

CO2.dbplsr.plot

#ggsave("../figures/Thesis/CO2_dbplsr_plot.png", plot=CO2.dbplsr.plot, device="png", path=NULL, scale=1, width=7, height=4, dpi=300, limitsize=TRUE, bg = "white")
```


## PERMANOVA

Permutational multivariate analysis of variance (PERMANOVA):

```{r PERMANOVA with History}
#PERMANOVA with History:
adonis_denit_H = adonis2(denitRA_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = denitRA_design, perm=10000, set.seed=42)
adonis_denit_H

adonis_meth_H = adonis2(methRA_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = methRA_design, perm=10000, set.seed=42)
adonis_meth_H

adonis_carb_metab_H = adonis2(carb_metabRA_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = carb_metabRA_design, perm=10000, set.seed=42)
adonis_carb_metab_H

adonis_cytoCox_H = adonis2(cytoCoxRA_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = cytoCoxRA_design, perm=10000, set.seed=42)
adonis_cytoCox_H

adonis_SO4_S_Assim_H = adonis2(SO4_S_AssimRA_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = SO4_S_AssimRA_design, perm=10000, set.seed=42)
adonis_SO4_S_Assim_H

adonis_Assim_SO4_red_H = adonis2(Assim_SO4_redRA_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = Assim_SO4_redRA_design, perm=10000, set.seed=42)
adonis_Assim_SO4_red_H

adonis_Dissim_SO4_red_H = adonis2(Dissim_SO4_redRA_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = Dissim_SO4_redRA_design, perm=10000, set.seed=42)
adonis_Dissim_SO4_red_H

adonis_ThioSOX_H = adonis2(ThioSOXRA_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = ThioSOXRA_design, perm=10000, set.seed=42)
adonis_ThioSOX_H

adonis_DNRA_H = adonis2(DNRARA_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = DNRARA_design, perm=10000, set.seed=42)
adonis_DNRA_H

adonis_ANRA_H = adonis2(ANRARA_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = ANRARA_design, perm=10000, set.seed=42)
adonis_ANRA_H

adonis_Nitrific_H = adonis2(NitrificRA_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = NitrificRA_design, perm=10000, set.seed=42)
adonis_Nitrific_H

adonis_COMAMMOX_H = adonis2(COMAMMOXRA_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = COMAMMOXRA_design, perm=10000, set.seed=42)
adonis_COMAMMOX_H

adonis_Nfix_H = adonis2(NfixRA_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = NfixRA_design, perm=10000, set.seed=42)
adonis_Nfix_H

adonis_CH4_ox_H = adonis2(CH4_oxRA_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = CH4_oxRA_design, perm=10000, set.seed=42)
adonis_CH4_ox_H

adonis_mtrBC_TIGR_H = adonis2(mtrBC_TIGRRA_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = mtrBC_TIGRRA_design, perm=10000, set.seed=42)
adonis_mtrBC_TIGR_H

adonis_FeGenie_H = adonis2(FeGenieRA_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = FeGenieRA_design, perm=10000, set.seed=42)
adonis_FeGenie_H

adonis_SO4_red_combined_H = adonis2(SO4_red_combinedRA_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = SO4_red_combinedRA_design, perm=10000, set.seed=42)
adonis_SO4_red_combined_H

adonis_FeReduction_H = adonis2(FeReductionRA_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = FeReductionRA_design, perm=10000, set.seed=42)
adonis_FeReduction_H

adonis_FeOxidation_H = adonis2(FeOxidationRA_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = FeOxidationRA_design, perm=10000, set.seed=42)
adonis_FeOxidation_H
```


## PCoA

```{r PCoA: denitrification, start&finish}
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate
pcoa_denit <- cmdscale(denitBC, k=2, eig=TRUE, add=TRUE) # changing k to 2 because we only want 2 dimensions anyway
  # Now trying add = TRUE because of https://www.youtube.com/watch?v=G5Qckqq5Erw

explainvar1b_denit <- round(pcoa_denit$eig[1] / sum(pcoa_denit$eig), 3) * 100
explainvar2b_denit <- round(pcoa_denit$eig[2] / sum(pcoa_denit$eig), 3) * 100
sum.eigb <- sum(explainvar1b_denit, explainvar2b_denit)

explainvar1b_denit # 27.1
explainvar2b_denit # 16.3

pcoa_denit2.groups <- paste(denitRA_design$Treat, denitRA_design$Plant, denitRA_design$Hist, sep = "_")
pcoa_denit2.points <- data.frame(pcoa_denit$points, group = pcoa_denit2.groups)

# Calculate Centroids (mean and SE)
pcoa_denit2.L.centroids <- reshape2::melt(pcoa_denit2.points, id="group", measure.vars = c("X1", "X2"))
pcoa_denit2.centroids <- acast(pcoa_denit2.L.centroids, variable ~ group, mean)
pcoa_denit2.centroids.se <- acast(pcoa_denit2.L.centroids, variable ~ group, se)
pcoa_denit2.centroids.sd <- acast(pcoa_denit2.L.centroids, variable ~ group, sd)

# Combine
pcoa_denit2.cent.dataframe <- cbind(t(pcoa_denit2.centroids), t(pcoa_denit2.centroids.se))
colnames(pcoa_denit2.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")

# Adding columns with the factors. H: hydrologic history
trt_H <- c("Baseline", "Baseline", "Dry","Dry", "Dry","Dry","Wet","Wet", "Wet","Wet") 
plant_H <- c("Baseline", "Baseline", "No Plant","No Plant","Plant", "Plant", "No Plant","No Plant","Plant", "Plant")
history <- c("Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet")
history_plant_combined <- c(paste(history, plant_H, sep = "_"))

pcoa_denit2.cent.dataframe.trts <- as.data.frame(pcoa_denit2.cent.dataframe) 
pcoa_denit2.cent.dataframe.trts$trt <- as.factor(trt_H)
pcoa_denit2.cent.dataframe.trts$plant <- as.factor(plant_H)
pcoa_denit2.cent.dataframe.trts$history <- as.factor(history)
pcoa_denit2.cent.dataframe.trts$history_plant<- as.factor(history_plant_combined)
pcoa_denit2.cent.dataframe.trts$centroid_or_sample <- "centroid"

# Add in points to data frame:
pcoa_denit2.points_ForPlot <- pcoa_denit2.points
colnames(pcoa_denit2.points_ForPlot) <- c("V1", "V2", "group")
# Add labels to match ...cent.dataframe.trts
pcoa_denit2.points_ForPlot$trt <- denitRA_design$Treat %>% str_replace_all(c("w" = "Wet", "_pre" = "Baseline", "d" = "Dry"))
pcoa_denit2.points_ForPlot$plant <- denitRA_design$Plant %>% str_replace_all(c("n" = "No Plant", "y" = "Plant", "_pre" = "Baseline"))
pcoa_denit2.points_ForPlot$history <- denitRA_design$Hist %>% str_replace_all(c("d" = "Dry", "w" = "Wet"))
pcoa_denit2.points_ForPlot$history_plant <- c(paste(pcoa_denit2.points_ForPlot$history, pcoa_denit2.points_ForPlot$plant, sep = "_"))
#remove "group" column
pcoa_denit2.points_ForPlot <- pcoa_denit2.points_ForPlot[, -3]
#add V1e and V2e as zeros:
pcoa_denit2.points_ForPlot$V1e <- numeric(24)
pcoa_denit2.points_ForPlot$V2e <- numeric(24)
# Add column to distinguish sample points from centroids, and later use to set alpha in plot
pcoa_denit2.points_ForPlot$centroid_or_sample <- "sample"

#Merge:
denit_centroids_points_merged <- rbind(pcoa_denit2.cent.dataframe.trts, pcoa_denit2.points_ForPlot)

# Add points data frame (seems to be newer issue, used to be able to use $ on pcoa_denit_FINAL$points)
pcoa_denit_DF <- as.data.frame(pcoa_denit$points)
```
```{r envfit with GHGs ONLY}
#Run envfit to see which [GHG]s (start or finish) have good fit with ordination:
#fit_denit <- envfit(pcoa_denit$points, GHG_metaG, permutations = 10000)
#fit_denit

#A_denit <-as.list(fit_denit$vectors)
#vec_denit <- as.data.frame(fit_denit$vectors$arrows*sqrt(fit_denit$vectors$r)*0.15)
#p_denit <- as.data.frame(A_denit$pvals)
#vec_denit <- cbind(vec_denit, p_denit)
#vec_denit <- subset(vec_denit, A_denit$pvals<=0.1)
```
```{r denit SSF envfit and PCoA}
#Run envfit with new Soil data (SSF = Soil Start & Finish):
fit_denit_SSF <- envfit(pcoa_denit_DF, SoilGHG_StartFinish_Envfit, permutations = 10000, set.seed(42))
fit_denit_SSF

A_denit_SSF <-as.list(fit_denit_SSF$vectors)
vec_denit_SSF <- as.data.frame(fit_denit_SSF$vectors$arrows*sqrt(fit_denit_SSF$vectors$r)*0.15)
p_denit_SSF <- as.data.frame(A_denit_SSF$pvals)
vec_denit_SSF <- cbind(vec_denit_SSF, p_denit_SSF)
vec_denit_SSF <- subset(vec_denit_SSF, A_denit_SSF$pvals<0.05)


#Plot using ggplot2
df2a <- as.data.frame(denit_centroids_points_merged)

plot2a <- ggplot(df2a, aes(x=V1, y=V2, shape=history_plant, colour=trt)) + geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.003), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.003), colour="black") +   
  geom_point(aes(shape = history_plant, colour=trt, alpha = centroid_or_sample, size = centroid_or_sample), stroke=2) +  
  theme_bw() +
  scale_size_manual(values = c(6, 3), name = "Centroid or Sample")+
  scale_alpha_manual(values = c(1, 0.3), name = "Centroid or Sample")+
  scale_color_manual(values=c("black", "#997700", "#004488"), name="Treatment") +
  scale_shape_manual(values = c(7, 0, 15, 13, 1, 19), name="History_Plant") +
  xlab(glue("PCoA 1 ({explainvar1b_denit}%)")) + ylab(glue("PCoA 2 ({explainvar2b_denit}%)")) + 
  geom_segment(data=vec_denit_SSF, aes(x=0,xend=V1,y=0,yend=V2), linewidth=1, arrow = arrow(length = unit(0.2, "cm")),colour="black", alpha = 0.4, inherit.aes=F)+ 
 geom_text(data=vec_denit_SSF, aes(x=V1+0.04, y=V2), label= expression(paste("mg CO")), size=4, inherit.aes=F)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  theme(panel.background = element_blank()) + 
  theme(axis.title = element_text(size=14), axis.text=element_text(size=14), 
          axis.text.x = element_text(size=14),  axis.text.y = element_text(size=14),
          panel.border = element_rect(colour = "black", linewidth=1.25)) +
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  theme(plot.title=element_text(size=14)) +
  theme(legend.text=element_text(size=14), legend.title = element_text(size=14))+
  guides(shape = guide_legend(override.aes = list(size = 4), order = 1), 
           colour = guide_legend(override.aes = list(pch=16, size = 4), order = 2)) +
    ggtitle("Denitrification")

plot2a

#ggsave("~/Desktop/newDenitPCoa.jpeg", plot = plot2a, device="jpeg", path=NULL, scale=1, height = 7, width = 10, limitsize = TRUE, dpi = 300)
```

```{r Denit FINAL}
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate
pcoa_denit_FINAL <- cmdscale(denitBC_finish, k=2, eig=TRUE, add=TRUE) # k = 2 because we only plot 2 dimensions, add=TRUE corrects highly negative eigenvalues


explainvar1_denit_FIN <- round(pcoa_denit_FINAL$eig[1] / sum(pcoa_denit_FINAL$eig), 3) * 100
explainvar2_denit_FIN <- round(pcoa_denit_FINAL$eig[2] / sum(pcoa_denit_FINAL$eig), 3) * 100
sum.eig_denit_FIN <- sum(explainvar1_denit_FIN, explainvar2_denit_FIN)

explainvar1_denit_FIN # 34.8
explainvar2_denit_FIN # 15.9

# Filter 'pre' samples out of *RA_design to get final samples only
denitRA_FIN_design <- denitRA_design %>% dplyr::filter(Treat != "_pre" & Plant != "_pre")

pcoa_denit_FIN.groups <- paste(denitRA_FIN_design$Treat, denitRA_FIN_design$Plant, denitRA_FIN_design$Hist, sep = "_")
pcoa_denit_FIN.points <- data.frame(pcoa_denit_FINAL$points, group = pcoa_denit_FIN.groups)

# Calculate Centroids (mean and SE)
pcoa_denit_FIN.L.centroids <- reshape2::melt(pcoa_denit_FIN.points, id="group", measure.vars = c("X1", "X2"))
pcoa_denit_FIN.centroids <- acast(pcoa_denit_FIN.L.centroids, variable ~ group, mean)
pcoa_denit_FIN.centroids.se <- acast(pcoa_denit_FIN.L.centroids, variable ~ group, se)
pcoa_denit_FIN.centroids.sd <- acast(pcoa_denit_FIN.L.centroids, variable ~ group, sd)

# Combine
pcoa_denit_FIN.cent.dataframe <- cbind(t(pcoa_denit_FIN.centroids), t(pcoa_denit_FIN.centroids.se))
colnames(pcoa_denit_FIN.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")

# Adding columns with the factors.
trt_FIN <- c("Dry","Dry", "Dry","Dry","Wet","Wet", "Wet","Wet") 
plant_FIN <- c("No Plant","No Plant","Plant", "Plant", "No Plant","No Plant","Plant", "Plant")
history_FIN <- c("Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet")
history_plant_combined_FIN <- c(paste(history_FIN, plant_FIN, sep = "_"))

pcoa_denit_FIN.cent.dataframe.trts <- as.data.frame(pcoa_denit_FIN.cent.dataframe) 
pcoa_denit_FIN.cent.dataframe.trts$trt <- as.factor(trt_FIN)
pcoa_denit_FIN.cent.dataframe.trts$plant <- as.factor(plant_FIN)
pcoa_denit_FIN.cent.dataframe.trts$history <- as.factor(history_FIN)
pcoa_denit_FIN.cent.dataframe.trts$history_plant<- as.factor(history_plant_combined_FIN)
pcoa_denit_FIN.cent.dataframe.trts$centroid_or_sample <- "centroid"

# Add in points to data frame:
pcoa_denit_FIN.points_ForPlot <- pcoa_denit_FIN.points
colnames(pcoa_denit_FIN.points_ForPlot) <- c("V1", "V2", "group")
# Add labels to match ...cent.dataframe.trts
pcoa_denit_FIN.points_ForPlot$trt <- denitRA_FIN_design$Treat %>% str_replace_all(c("w" = "Wet", "_pre" = "Baseline", "d" = "Dry"))
pcoa_denit_FIN.points_ForPlot$plant <- denitRA_FIN_design$Plant %>% str_replace_all(c("n" = "No Plant", "y" = "Plant", "_pre" = "Baseline"))
pcoa_denit_FIN.points_ForPlot$history <- denitRA_FIN_design$Hist %>% str_replace_all(c("d" = "Dry", "w" = "Wet"))
pcoa_denit_FIN.points_ForPlot$history_plant <- c(paste(pcoa_denit_FIN.points_ForPlot$history, pcoa_denit_FIN.points_ForPlot$plant, sep = "_"))
#remove "group" column
pcoa_denit_FIN.points_ForPlot <- pcoa_denit_FIN.points_ForPlot[, -3]
#add V1e and V2e as zeros:
pcoa_denit_FIN.points_ForPlot$V1e <- numeric(16)
pcoa_denit_FIN.points_ForPlot$V2e <- numeric(16)
# Add column to distinguish sample points from centroids, and later use to set alpha in plot
pcoa_denit_FIN.points_ForPlot$centroid_or_sample <- "sample"

#Merge:
denit_FIN_centroids_points_merged <- rbind(pcoa_denit_FIN.cent.dataframe.trts, pcoa_denit_FIN.points_ForPlot)

# Add points data frame (seems to be newer issue, used to be able to use $ on pcoa_denit_FINAL$points)
pcoa_denit_FINAL_DF <- as.data.frame(pcoa_denit_FINAL$points)

#Run envfit with new Soil data (SFO = Soil Finish Only):
fit_denit_SFO <- envfit(pcoa_denit_FINAL_DF, envfit_FINISH_numerical, permutations = 10000, set.seed(42))
fit_denit_SFO

A_denit_SFO <-as.list(fit_denit_SFO$vectors)
vec_denit_SFO <- as.data.frame(fit_denit_SFO$vectors$arrows*sqrt(fit_denit_SFO$vectors$r)*0.15)
p_denit_SFO <- as.data.frame(A_denit_SFO$pvals)
vec_denit_SFO <- cbind(vec_denit_SFO, p_denit_SFO)
vec_denit_SFO <- subset(vec_denit_SFO, A_denit_SFO$pvals<0.05)



#Plot using ggplot2
df3a <- as.data.frame(denit_FIN_centroids_points_merged)

plot3a <- ggplot(df3a, aes(x=V1, y=V2, shape=history_plant, colour=trt)) + geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.003), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.003), colour="black") +   
  geom_point(aes(shape = history_plant, colour=trt, alpha = centroid_or_sample, size = centroid_or_sample), stroke=2) +  
  theme_bw() +
  scale_size_manual(values = c(6, 3), name = "Centroid or Sample")+
  scale_alpha_manual(values = c(1, 0.3), name = "Centroid or Sample")+
  scale_color_manual(values=c("#997700", "#004488"), name="Treatment") +
  scale_shape_manual(values = c(0, 15, 1, 19), name="History_Plant") +
  xlab(glue("PCoA 1 ({explainvar1_denit_FIN}%)")) + ylab(glue("PCoA 2 ({explainvar2_denit_FIN}%)")) + 
  geom_segment(data=vec_denit_SFO, aes(x=0,xend=V1,y=0,yend=V2), linewidth=1, arrow = arrow(length = unit(0.2, "cm")), colour="black", alpha = 0.4,inherit.aes=F)+ 
 geom_text(data=vec_denit_SFO, aes(x=V1, y=V2+0.01), label= expression(paste("CO"[2])), size=4, inherit.aes=F)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  theme(panel.background = element_blank()) + 
  theme(axis.title = element_text(size=14), axis.text=element_text(size=14), 
          axis.text.x = element_text(size=14),  axis.text.y = element_text(size=14),
          panel.border = element_rect(colour = "black", linewidth=1.25)) +
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  theme(plot.title=element_text(size=14)) +
  theme(legend.text=element_text(size=14), legend.title = element_text(size=14))+
  guides(shape = guide_legend(override.aes = list(size = 4), order = 1), 
           colour = guide_legend(override.aes = list(pch=16, size = 4), order = 2)) +
    ggtitle("Denitrification")

plot3a
```



PCoA Methanogenesis, all factors, HISTORY AS SHAPE:
```{r PCoA: Methanogenesis; all factors}
pcoa_meth <- cmdscale(methBC, k=2, eig=TRUE, add=TRUE)

explainvar1b_meth <- round(pcoa_meth$eig[1] / sum(pcoa_meth$eig), 3) * 100
explainvar2b_meth <- round(pcoa_meth$eig[2] / sum(pcoa_meth$eig), 3) * 100
sum.eigb <- sum(explainvar1b_meth, explainvar2b_meth)

explainvar1b_meth # 29.7
explainvar2b_meth # 20.8

pcoa_meth2.groups <- paste(methRA_design$Treat, methRA_design$Plant, methRA_design$Hist, sep = "_")
pcoa_meth2.points <- data.frame(pcoa_meth$points, group = pcoa_meth2.groups)

# Calculate Centroids (mean and SE)
pcoa_meth2.L.centroids <- reshape2::melt(pcoa_meth2.points, id="group", measure.vars = c("X1", "X2"))
pcoa_meth2.centroids <- acast(pcoa_meth2.L.centroids, variable ~ group, mean)
pcoa_meth2.centroids.se <- acast(pcoa_meth2.L.centroids, variable ~ group, se)
pcoa_meth2.centroids.sd <- acast(pcoa_meth2.L.centroids, variable ~ group, sd)

# Combine
pcoa_meth2.cent.dataframe <- cbind(t(pcoa_meth2.centroids), t(pcoa_meth2.centroids.se))
colnames(pcoa_meth2.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")

# No need to run again if these lists have already been declared:
#trt_H <- c("Baseline", "Baseline", "Dry","Dry", "Dry","Dry","Wet","Wet", "Wet","Wet") 
#plant_H <- c("Baseline", "Baseline", "No Plant","No Plant","Plant", "Plant", "No Plant","No Plant","Plant", "Plant")
#history <- c("Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet")
#trt_plant_combined <- c(paste(trt, plant, sep = "_"))

pcoa_meth2.cent.dataframe.trts <- as.data.frame(pcoa_meth2.cent.dataframe) 
pcoa_meth2.cent.dataframe.trts$trt <- as.factor(trt_H)
pcoa_meth2.cent.dataframe.trts$plant <- as.factor(plant_H)
pcoa_meth2.cent.dataframe.trts$history <- as.factor(history)
pcoa_meth2.cent.dataframe.trts$history_plant<- as.factor(history_plant_combined)
pcoa_meth2.cent.dataframe.trts$centroid_or_sample <- "centroid"

# Add in points to data frame:
pcoa_meth2.points_ForPlot <- pcoa_meth2.points
colnames(pcoa_meth2.points_ForPlot) <- c("V1", "V2", "group")
# Add labels to match ...cent.dataframe.trts
pcoa_meth2.points_ForPlot$trt <- methRA_design$Treat %>% str_replace_all(c("w" = "Wet", "_pre" = "Baseline", "d" = "Dry"))
pcoa_meth2.points_ForPlot$plant <- methRA_design$Plant %>% str_replace_all(c("n" = "No Plant", "y" = "Plant", "_pre" = "Baseline"))
pcoa_meth2.points_ForPlot$history <- methRA_design$Hist %>% str_replace_all(c("d" = "Dry", "w" = "Wet"))
pcoa_meth2.points_ForPlot$history_plant <- c(paste(pcoa_meth2.points_ForPlot$history, pcoa_meth2.points_ForPlot$plant, sep = "_"))
#remove "group" column
pcoa_meth2.points_ForPlot <- pcoa_meth2.points_ForPlot[, -3]
#add V1e and V2e as zeros:
pcoa_meth2.points_ForPlot$V1e <- numeric(24)
pcoa_meth2.points_ForPlot$V2e <- numeric(24)
# Add column to distinguish sample points from centroids, and later use to set alpha in plot
pcoa_meth2.points_ForPlot$centroid_or_sample <- "sample"

#Merge:
meth_centroids_points_merged <- rbind(pcoa_meth2.cent.dataframe.trts, pcoa_meth2.points_ForPlot)

# Add points data frame (seems to be newer issue, used to be able to use $ on pcoa_denit_FINAL$points)
pcoa_meth_DF <- as.data.frame(pcoa_meth$points)
```
```{r envfit with GHGs ONLY}
#Run envfit to see which [GHG]s (start or finish) have good fit with ordination:
#fit_meth <- envfit(pcoa_meth$points, GHG_metaG, permutations = 10000)
#fit_meth

#A_meth <-as.list(fit_meth$vectors)
#vec_meth <- as.data.frame(fit_meth$vectors$arrows*sqrt(fit_meth$vectors$r)*0.15)
#p_meth <- as.data.frame(A_meth$pvals)
#vec_meth <- cbind(vec_meth, p_meth)
#vec_meth <- subset(vec_meth, A_meth$pvals<=0.1)
```
```{r meth SSF envfit and PCoA}
#Run envfit with new Soil data (SSF = Soil Start & Finish):
fit_meth_SSF <- envfit(pcoa_meth_DF, SoilGHG_StartFinish_Envfit, permutations = 10000, set.seed(42))
fit_meth_SSF

A_meth_SSF <-as.list(fit_meth_SSF$vectors)
vec_meth_SSF <- as.data.frame(fit_meth_SSF$vectors$arrows*sqrt(fit_meth_SSF$vectors$r)*0.15)
p_meth_SSF <- as.data.frame(A_meth_SSF$pvals)
vec_meth_SSF <- cbind(vec_meth_SSF, p_meth_SSF)
vec_meth_SSF <- subset(vec_meth_SSF, A_meth_SSF$pvals<0.05)


#Plot using ggplot2
df2b <- as.data.frame(meth_centroids_points_merged)

plot2b <- ggplot(df2b, aes(x=V1, y=V2, shape=history_plant, colour=trt)) + geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.003), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.003), colour="black") +   
  geom_point(aes(shape = history_plant, colour=trt, alpha = centroid_or_sample, size = centroid_or_sample), stroke=2) +  
  theme_bw() +
  scale_size_manual(values = c(6, 3), name = "Centroid or Sample")+
  scale_alpha_manual(values = c(1, 0.3), name = "Centroid or Sample")+
  scale_color_manual(values=c("black", "#997700", "#004488"), name="Treatment") +
  scale_shape_manual(values = c(7, 0, 15, 13, 1, 19), name="History_Plant") +
  xlab(glue("PCoA 1 ({explainvar1b_meth}%)")) + ylab(glue("PCoA 2 ({explainvar2b_meth}%)")) + 
  #geom_segment(data=vec_meth_SSF, aes(x=0,xend=Dim1,y=0,yend=Dim2), size=1, arrow = arrow(length = unit(0.2, "cm")),colour="black", alpha=0.4, inherit.aes=F)+ 
 #geom_text(data=vec_meth_SSF, aes(x=Dim1, y=Dim2), label= c(expression(paste("mg CH"[4],"-C m"^{-2}), "%moisture")), size=3, inherit.aes=F)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  theme(panel.background = element_blank()) + 
  theme(axis.title = element_text(size=14), axis.text=element_text(size=14), 
          axis.text.x = element_text(size=14),  axis.text.y = element_text(size=14),
          panel.border = element_rect(colour = "black", linewidth=1.25)) +
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  theme(plot.title=element_text(size=14)) +
  theme(legend.text=element_text(size=14), legend.title = element_text(size=14))+
  guides(shape = guide_legend(override.aes = list(size = 4), order = 1), 
           colour = guide_legend(override.aes = list(pch=16, size = 4), order = 2)) +
    ggtitle("Methanogenesis")

plot2b

#ggsave("../figures/Thesis/MethanogenesisPCoA.png", plot = plot2b, path=NULL, device="png", scale=1, height = 7, width = 10, limitsize = TRUE, dpi = 300)
```

```{r meth FINAL PCoA}
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate
pcoa_meth_FINAL <- cmdscale(methBC_finish, k=2, eig=TRUE, add=TRUE) # k = 2 because we only plot 2 dimensions, add=TRUE corrects highly negative eigenvalues


explainvar1_meth_FIN <- round(pcoa_meth_FINAL$eig[1] / sum(pcoa_meth_FINAL$eig), 3) * 100
explainvar2_meth_FIN <- round(pcoa_meth_FINAL$eig[2] / sum(pcoa_meth_FINAL$eig), 3) * 100
sum.eig_meth_FIN <- sum(explainvar1_meth_FIN, explainvar2_meth_FIN)

explainvar1_meth_FIN # 33.2
explainvar2_meth_FIN # 24.8

# Filter 'pre' samples out of *RA_design to get final samples only
methRA_FIN_design <- methRA_design %>% dplyr::filter(Treat != "_pre" & Plant != "_pre")

pcoa_meth_FIN.groups <- paste(methRA_FIN_design$Treat, methRA_FIN_design$Plant, methRA_FIN_design$Hist, sep = "_")
pcoa_meth_FIN.points <- data.frame(pcoa_meth_FINAL$points, group = pcoa_meth_FIN.groups)

# Calculate Centroids (mean and SE)
pcoa_meth_FIN.L.centroids <- reshape2::melt(pcoa_meth_FIN.points, id="group", measure.vars = c("X1", "X2"))
pcoa_meth_FIN.centroids <- acast(pcoa_meth_FIN.L.centroids, variable ~ group, mean)
pcoa_meth_FIN.centroids.se <- acast(pcoa_meth_FIN.L.centroids, variable ~ group, se)
pcoa_meth_FIN.centroids.sd <- acast(pcoa_meth_FIN.L.centroids, variable ~ group, sd)

# Combine
pcoa_meth_FIN.cent.dataframe <- cbind(t(pcoa_meth_FIN.centroids), t(pcoa_meth_FIN.centroids.se))
colnames(pcoa_meth_FIN.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")

# Adding columns with the factors.
trt_FIN <- c("Dry","Dry", "Dry","Dry","Wet","Wet", "Wet","Wet") 
plant_FIN <- c("No Plant","No Plant","Plant", "Plant", "No Plant","No Plant","Plant", "Plant")
history_FIN <- c("Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet")
history_plant_combined_FIN <- c(paste(history_FIN, plant_FIN, sep = "_"))

pcoa_meth_FIN.cent.dataframe.trts <- as.data.frame(pcoa_meth_FIN.cent.dataframe) 
pcoa_meth_FIN.cent.dataframe.trts$trt <- as.factor(trt_FIN)
pcoa_meth_FIN.cent.dataframe.trts$plant <- as.factor(plant_FIN)
pcoa_meth_FIN.cent.dataframe.trts$history <- as.factor(history_FIN)
pcoa_meth_FIN.cent.dataframe.trts$history_plant<- as.factor(history_plant_combined_FIN)
pcoa_meth_FIN.cent.dataframe.trts$centroid_or_sample <- "centroid"

# Add in points to data frame:
pcoa_meth_FIN.points_ForPlot <- pcoa_meth_FIN.points
colnames(pcoa_meth_FIN.points_ForPlot) <- c("V1", "V2", "group")
# Add labels to match ...cent.dataframe.trts
pcoa_meth_FIN.points_ForPlot$trt <- methRA_FIN_design$Treat %>% str_replace_all(c("w" = "Wet", "_pre" = "Baseline", "d" = "Dry"))
pcoa_meth_FIN.points_ForPlot$plant <- methRA_FIN_design$Plant %>% str_replace_all(c("n" = "No Plant", "y" = "Plant", "_pre" = "Baseline"))
pcoa_meth_FIN.points_ForPlot$history <- methRA_FIN_design$Hist %>% str_replace_all(c("d" = "Dry", "w" = "Wet"))
pcoa_meth_FIN.points_ForPlot$history_plant <- c(paste(pcoa_meth_FIN.points_ForPlot$history, pcoa_meth_FIN.points_ForPlot$plant, sep = "_"))
#remove "group" column
pcoa_meth_FIN.points_ForPlot <- pcoa_meth_FIN.points_ForPlot[, -3]
#add V1e and V2e as zeros:
pcoa_meth_FIN.points_ForPlot$V1e <- numeric(16)
pcoa_meth_FIN.points_ForPlot$V2e <- numeric(16)
# Add column to distinguish sample points from centroids, and later use to set alpha in plot
pcoa_meth_FIN.points_ForPlot$centroid_or_sample <- "sample"

#Merge:
meth_FIN_centroids_points_merged <- rbind(pcoa_meth_FIN.cent.dataframe.trts, pcoa_meth_FIN.points_ForPlot)

# Add points data frame (seems to be newer issue, used to be able to use $ on pcoa_denit_FINAL$points)
pcoa_meth_FINAL_DF <- as.data.frame(pcoa_meth_FINAL$points)

#Run envfit with new Soil data (SFO = Soil Finish Only):
fit_meth_SFO <- envfit(pcoa_meth_FINAL_DF, envfit_FINISH_numerical, permutations = 10000, set.seed(42))
fit_meth_SFO

A_meth_SFO <-as.list(fit_meth_SFO$vectors)
vec_meth_SFO <- as.data.frame(fit_meth_SFO$vectors$arrows*sqrt(fit_meth_SFO$vectors$r)*0.15)
p_meth_SFO <- as.data.frame(A_meth_SFO$pvals)
vec_meth_SFO <- cbind(vec_meth_SFO, p_meth_SFO)
vec_meth_SFO <- subset(vec_meth_SFO, A_meth_SFO$pvals<0.05)

# Add column with labels, for plotting:
vec_meth_SFO$labels <- c("P", "Mg", "Ca", "Zn", "Cu", "CEC")


#Plot using ggplot2
df3b <- as.data.frame(meth_FIN_centroids_points_merged)

plot3b <- ggplot(df3b, aes(x=V1, y=V2, shape=history_plant, colour=trt)) + geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.003), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.003), colour="black") +   
  geom_point(aes(shape = history_plant, colour=trt, alpha = centroid_or_sample, size = centroid_or_sample), stroke=2) +  
  theme_bw() +
  scale_size_manual(values = c(6, 3), name = "Centroid or Sample")+
  scale_alpha_manual(values = c(1, 0.3), name = "Centroid or Sample")+
  scale_color_manual(values=c("#997700", "#004488"), name="Treatment") +
  scale_shape_manual(values = c(0, 15, 1, 19), name="History_Plant") +
  xlab(glue("PCoA 1 ({explainvar1_meth_FIN}%)")) + ylab(glue("PCoA 2 ({explainvar2_meth_FIN}%)")) + 
  geom_segment(data=vec_meth_SFO, aes(x=0,xend=V1,y=0,yend=V2), linewidth=1, arrow = arrow(length = unit(0.2, "cm")),colour="black", alpha = 0.4,inherit.aes=F)+ 
 geom_text(data=vec_meth_SFO, aes(x=V1-0.01, y=V2), label=vec_meth_SFO$labels, size=4, inherit.aes=F)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  theme(panel.background = element_blank()) + 
  theme(axis.title = element_text(size=14), axis.text=element_text(size=14), 
          axis.text.x = element_text(size=14),  axis.text.y = element_text(size=14),
          panel.border = element_rect(colour = "black", linewidth=1.25)) +
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  theme(plot.title=element_text(size=14)) +
  theme(legend.text=element_text(size=14), legend.title = element_text(size=14))+
  guides(shape = guide_legend(override.aes = list(size = 4), order = 1), 
           colour = guide_legend(override.aes = list(pch=16, size = 4), order = 2)) +
    ggtitle("Methanogenesis")

plot3b
```


PCoA central carbohydrate metabolism, all factors, HISTORY AS SHAPE:
```{r PCoA: central carbohydrate metabolism; all factors}
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate
pcoa_carb_metab <- cmdscale(carb_metabBC, k=2, eig=TRUE, add=TRUE)

explainvar1b_carb_metab <- format(round(pcoa_carb_metab$eig[1] / sum(pcoa_carb_metab$eig), 3) * 100, nsmall = 1) # Using format() to get .0
explainvar2b_carb_metab <- round(pcoa_carb_metab$eig[2] / sum(pcoa_carb_metab$eig), 3) * 100
#sum.eigc <- sum(explainvar1b_carb_metab, explainvar2b_carb_metab) #Doesn't work when using format()

explainvar1b_carb_metab # 31.0
explainvar2b_carb_metab # 14.9

pcoa_carb_metab2.groups <- paste(carb_metabRA_design$Treat, carb_metabRA_design$Plant, carb_metabRA_design$Hist, sep = "_")
pcoa_carb_metab2.points <- data.frame(pcoa_carb_metab$points, group = pcoa_carb_metab2.groups)

# Calculate Centroids (mean and SE)
pcoa_carb_metab2.L.centroids <- reshape2::melt(pcoa_carb_metab2.points, id="group", measure.vars = c("X1", "X2"))
pcoa_carb_metab2.centroids <- acast(pcoa_carb_metab2.L.centroids, variable ~ group, mean)
pcoa_carb_metab2.centroids.se <- acast(pcoa_carb_metab2.L.centroids, variable ~ group, se)
pcoa_carb_metab2.centroids.sd <- acast(pcoa_carb_metab2.L.centroids, variable ~ group, sd)

# Combine
pcoa_carb_metab2.cent.dataframe <- cbind(t(pcoa_carb_metab2.centroids), t(pcoa_carb_metab2.centroids.se))
colnames(pcoa_carb_metab2.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")

# No need to run again if these lists have already been declared:
#trt_H <- c("Baseline", "Baseline", "Dry","Dry", "Dry","Dry","Wet","Wet", "Wet","Wet") 
#plant_H <- c("Baseline", "Baseline", "No Plant","No Plant","Plant", "Plant", "No Plant","No Plant","Plant", "Plant")
#history <- c("Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet")
#trt_plant_combined <- c(paste(trt, plant, sep = "_"))

pcoa_carb_metab2.cent.dataframe.trts <- as.data.frame(pcoa_carb_metab2.cent.dataframe) 
pcoa_carb_metab2.cent.dataframe.trts$trt <- as.factor(trt_H)
pcoa_carb_metab2.cent.dataframe.trts$plant <- as.factor(plant_H)
pcoa_carb_metab2.cent.dataframe.trts$history <- as.factor(history)
pcoa_carb_metab2.cent.dataframe.trts$history_plant<- as.factor(history_plant_combined)
pcoa_carb_metab2.cent.dataframe.trts$centroid_or_sample <- "centroid"

# Add in points to data frame:
pcoa_carb_metab2.points_ForPlot <- pcoa_carb_metab2.points
colnames(pcoa_carb_metab2.points_ForPlot) <- c("V1", "V2", "group")
# Add labels to match ...cent.dataframe.trts
pcoa_carb_metab2.points_ForPlot$trt <- carb_metabRA_design$Treat %>% str_replace_all(c("w" = "Wet", "_pre" = "Baseline", "d" = "Dry"))
pcoa_carb_metab2.points_ForPlot$plant <- carb_metabRA_design$Plant %>% str_replace_all(c("n" = "No Plant", "y" = "Plant", "_pre" = "Baseline"))
pcoa_carb_metab2.points_ForPlot$history <- carb_metabRA_design$Hist %>% str_replace_all(c("d" = "Dry", "w" = "Wet"))
pcoa_carb_metab2.points_ForPlot$history_plant <- c(paste(pcoa_carb_metab2.points_ForPlot$history, pcoa_carb_metab2.points_ForPlot$plant, sep = "_"))
#remove "group" column
pcoa_carb_metab2.points_ForPlot <- pcoa_carb_metab2.points_ForPlot[, -3]
#add V1e and V2e as zeros:
pcoa_carb_metab2.points_ForPlot$V1e <- numeric(24)
pcoa_carb_metab2.points_ForPlot$V2e <- numeric(24)
# Add column to distinguish sample points from centroids, and later use to set alpha in plot
pcoa_carb_metab2.points_ForPlot$centroid_or_sample <- "sample"

#Merge:
carb_metab_centroids_points_merged <- rbind(pcoa_carb_metab2.cent.dataframe.trts, pcoa_carb_metab2.points_ForPlot)

# Add points data frame (seems to be newer issue, used to be able to use $ on pcoa_denit_FINAL$points)
pcoa_carb_metab_DF <- as.data.frame(pcoa_carb_metab$points)
```
```{r envfit with GHGs ONLY}
#Run envfit to see which [GHG]s (start or finish) have good fit with ordination:
#fit_carb_metab <- envfit(pcoa_carb_metab$points, GHG_metaG, permutations = 10000)
#fit_carb_metab

#A_carb_metab <-as.list(fit_carb_metab$vectors)
#vec_carb_metab <- as.data.frame(fit_carb_metab$vectors$arrows*sqrt(fit_carb_metab$vectors$r)*0.15)
#p_carb_metab <- as.data.frame(A_carb_metab$pvals)
#vec_carb_metab <- cbind(vec_carb_metab, p_carb_metab)
#vec_carb_metab <- subset(vec_carb_metab, A_carb_metab$pvals<=0.1)
```
```{r carb_metab SSF envfit and PCoA}
#Run envfit with new Soil data (SSF = Soil Start & Finish):
fit_carb_metab_SSF <- envfit(pcoa_carb_metab_DF, SoilGHG_StartFinish_Envfit, permutations = 10000, set.seed(42))
fit_carb_metab_SSF

A_carb_metab_SSF <-as.list(fit_carb_metab_SSF$vectors)
vec_carb_metab_SSF <- as.data.frame(fit_carb_metab_SSF$vectors$arrows*sqrt(fit_carb_metab_SSF$vectors$r)*0.15)
p_carb_metab_SSF <- as.data.frame(A_carb_metab_SSF$pvals)
vec_carb_metab_SSF <- cbind(vec_carb_metab_SSF, p_carb_metab_SSF)
vec_carb_metab_SSF <- subset(vec_carb_metab_SSF, A_carb_metab_SSF$pvals<0.05)

#Plot using ggplot2
df2c <- as.data.frame(carb_metab_centroids_points_merged)

plot2c <- ggplot(df2c, aes(x=V1, y=V2, shape=history_plant, colour=trt)) + geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.003), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.003), colour="black") +   
  geom_point(aes(shape = history_plant, colour=trt, alpha = centroid_or_sample, size = centroid_or_sample), stroke=2) +  
  theme_bw() +
  scale_size_manual(values = c(6, 3), name = "Centroid or Sample")+
  scale_alpha_manual(values = c(1, 0.3), name = "Centroid or Sample")+
  scale_color_manual(values=c("black", "#997700", "#004488"), name="Treatment") +
  scale_shape_manual(values = c(7, 0, 15, 13, 1, 19), name="History_Plant") +
  xlab(glue("PCoA 1 ({explainvar1b_carb_metab}%)")) + ylab(glue("PCoA 2 ({explainvar2b_carb_metab}%)")) + 
  geom_segment(data=vec_carb_metab_SSF, aes(x=0,xend=V1,y=0,yend=V2), linewidth=1, arrow = arrow(length = unit(0.2, "cm")),colour="black", alpha = 0.4, inherit.aes=F)+ 
 geom_text(data=vec_carb_metab_SSF, aes(x=V1-0.01, y=V2-0.005), label= expression(paste("CO"[2])), size=4, inherit.aes=F)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  theme(panel.background = element_blank()) + 
  theme(axis.title = element_text(size=14), axis.text=element_text(size=14), 
          axis.text.x = element_text(size=14),  axis.text.y = element_text(size=14),
          panel.border = element_rect(colour = "black", linewidth=1.25)) +
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  theme(plot.title=element_text(size=14)) +
  theme(legend.text=element_text(size=14), legend.title = element_text(size=14))+
  guides(shape = guide_legend(override.aes = list(size = 4), order = 1), 
           colour = guide_legend(override.aes = list(pch=16, size = 4), order = 2)) +
    ggtitle("Central Carbohydrate Metabolism")

plot2c
```

```{r PCoA: central carb metab, FINISH only}
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate
pcoa_carb_metab_FINAL <- cmdscale(carb_metabBC_finish, k=2, eig=TRUE, add=TRUE) # k = 2 because we only plot 2 dimensions, add=TRUE corrects highly negative eigenvalues


explainvar1_carb_metab_FIN <- round(pcoa_carb_metab_FINAL$eig[1] / sum(pcoa_carb_metab_FINAL$eig), 3) * 100
explainvar2_carb_metab_FIN <- round(pcoa_carb_metab_FINAL$eig[2] / sum(pcoa_carb_metab_FINAL$eig), 3) * 100
sum.eig_carb_metab_FIN <- sum(explainvar1_carb_metab_FIN, explainvar2_carb_metab_FIN)

explainvar1_carb_metab_FIN # 33.2
explainvar2_carb_metab_FIN # 24.8

# Filter 'pre' samples out of *RA_design to get final samples only
carb_metabRA_FIN_design <- carb_metabRA_design %>% dplyr::filter(Treat != "_pre" & Plant != "_pre")

pcoa_carb_metab_FIN.groups <- paste(carb_metabRA_FIN_design$Treat, carb_metabRA_FIN_design$Plant, carb_metabRA_FIN_design$Hist, sep = "_")
pcoa_carb_metab_FIN.points <- data.frame(pcoa_carb_metab_FINAL$points, group = pcoa_carb_metab_FIN.groups)

# Calculate Centroids (mean and SE)
pcoa_carb_metab_FIN.L.centroids <- reshape2::melt(pcoa_carb_metab_FIN.points, id="group", measure.vars = c("X1", "X2"))
pcoa_carb_metab_FIN.centroids <- acast(pcoa_carb_metab_FIN.L.centroids, variable ~ group, mean)
pcoa_carb_metab_FIN.centroids.se <- acast(pcoa_carb_metab_FIN.L.centroids, variable ~ group, se)
pcoa_carb_metab_FIN.centroids.sd <- acast(pcoa_carb_metab_FIN.L.centroids, variable ~ group, sd)

# Combine
pcoa_carb_metab_FIN.cent.dataframe <- cbind(t(pcoa_carb_metab_FIN.centroids), t(pcoa_carb_metab_FIN.centroids.se))
colnames(pcoa_carb_metab_FIN.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")

# Adding columns with the factors.
trt_FIN <- c("Dry","Dry", "Dry","Dry","Wet","Wet", "Wet","Wet") 
plant_FIN <- c("No Plant","No Plant","Plant", "Plant", "No Plant","No Plant","Plant", "Plant")
history_FIN <- c("Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet")
history_plant_combined_FIN <- c(paste(history_FIN, plant_FIN, sep = "_"))

pcoa_carb_metab_FIN.cent.dataframe.trts <- as.data.frame(pcoa_carb_metab_FIN.cent.dataframe) 
pcoa_carb_metab_FIN.cent.dataframe.trts$trt <- as.factor(trt_FIN)
pcoa_carb_metab_FIN.cent.dataframe.trts$plant <- as.factor(plant_FIN)
pcoa_carb_metab_FIN.cent.dataframe.trts$history <- as.factor(history_FIN)
pcoa_carb_metab_FIN.cent.dataframe.trts$history_plant<- as.factor(history_plant_combined_FIN)
pcoa_carb_metab_FIN.cent.dataframe.trts$centroid_or_sample <- "centroid"

# Add in points to data frame:
pcoa_carb_metab_FIN.points_ForPlot <- pcoa_carb_metab_FIN.points
colnames(pcoa_carb_metab_FIN.points_ForPlot) <- c("V1", "V2", "group")
# Add labels to match ...cent.dataframe.trts
pcoa_carb_metab_FIN.points_ForPlot$trt <- carb_metabRA_FIN_design$Treat %>% str_replace_all(c("w" = "Wet", "_pre" = "Baseline", "d" = "Dry"))
pcoa_carb_metab_FIN.points_ForPlot$plant <- carb_metabRA_FIN_design$Plant %>% str_replace_all(c("n" = "No Plant", "y" = "Plant", "_pre" = "Baseline"))
pcoa_carb_metab_FIN.points_ForPlot$history <- carb_metabRA_FIN_design$Hist %>% str_replace_all(c("d" = "Dry", "w" = "Wet"))
pcoa_carb_metab_FIN.points_ForPlot$history_plant <- c(paste(pcoa_carb_metab_FIN.points_ForPlot$history, pcoa_carb_metab_FIN.points_ForPlot$plant, sep = "_"))
#remove "group" column
pcoa_carb_metab_FIN.points_ForPlot <- pcoa_carb_metab_FIN.points_ForPlot[, -3]
#add V1e and V2e as zeros:
pcoa_carb_metab_FIN.points_ForPlot$V1e <- numeric(16)
pcoa_carb_metab_FIN.points_ForPlot$V2e <- numeric(16)
# Add column to distinguish sample points from centroids, and later use to set alpha in plot
pcoa_carb_metab_FIN.points_ForPlot$centroid_or_sample <- "sample"

#Merge:
carb_metab_FIN_centroids_points_merged <- rbind(pcoa_carb_metab_FIN.cent.dataframe.trts, pcoa_carb_metab_FIN.points_ForPlot)

# Add points data frame (seems to be newer issue, used to be able to use $ on pcoa_denit_FINAL$points)
pcoa_carb_metab_FINAL_DF <- as.data.frame(pcoa_carb_metab_FINAL$points)


#Run envfit with new Soil data (SFO = Soil Finish Only):
fit_carb_metab_SFO <- envfit(pcoa_carb_metab_FINAL_DF, envfit_FINISH_numerical, permutations = 10000, set.seed(42))
fit_carb_metab_SFO

A_carb_metab_SFO <-as.list(fit_carb_metab_SFO$vectors)
vec_carb_metab_SFO <- as.data.frame(fit_carb_metab_SFO$vectors$arrows*sqrt(fit_carb_metab_SFO$vectors$r)*0.15)
p_carb_metab_SFO <- as.data.frame(A_carb_metab_SFO$pvals)
vec_carb_metab_SFO <- cbind(vec_carb_metab_SFO, p_carb_metab_SFO)
vec_carb_metab_SFO <- subset(vec_carb_metab_SFO, A_carb_metab_SFO$pvals<0.05)

# Add column with labels, for plotting:
vec_carb_metab_SFO$labels <- c("Mg, pH", "Ca", "pH, Mg", "S", "Mn", "Cu", "HM", "CEC")


#Plot using ggplot2
df3c <- as.data.frame(carb_metab_FIN_centroids_points_merged)

plot3c <- ggplot(df3c, aes(x=V1, y=V2, shape=history_plant, colour=trt)) + geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.003), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.003), colour="black") +   
  geom_point(aes(shape = history_plant, colour=trt, alpha = centroid_or_sample, size = centroid_or_sample), stroke=2) +  
  theme_bw() +
  scale_size_manual(values = c(6, 3), name = "Centroid or Sample")+
  scale_alpha_manual(values = c(1, 0.3), name = "Centroid or Sample")+
  scale_color_manual(values=c("#997700", "#004488"), name="Treatment") +
  scale_shape_manual(values = c(0, 15, 1, 19), name="History_Plant") +
  xlab(glue("PCoA 1 ({explainvar1_carb_metab_FIN}%)")) + ylab(glue("PCoA 2 ({explainvar2_carb_metab_FIN}%)")) + 
  geom_segment(data=vec_carb_metab_SFO, aes(x=0,xend=V1,y=0,yend=V2), linewidth=1, arrow = arrow(length = unit(0.2, "cm")),colour="black", alpha = 0.4,inherit.aes=F)+ 
 geom_text(data=vec_carb_metab_SFO, aes(x=V1, y=V2), label=vec_carb_metab_SFO$labels, size=4, check_overlap = T, nudge_x = 0.01, inherit.aes=F)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  theme(panel.background = element_blank()) + 
  theme(axis.title = element_text(size=14), axis.text=element_text(size=14), 
          axis.text.x = element_text(size=14),  axis.text.y = element_text(size=14),
          panel.border = element_rect(colour = "black", linewidth=1.25)) +
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  theme(plot.title=element_text(size=14)) +
  theme(legend.text=element_text(size=14), legend.title = element_text(size=14))+
  guides(shape = guide_legend(override.aes = list(size = 4), order = 1), 
           colour = guide_legend(override.aes = list(pch=16, size = 4), order = 2)) +
    ggtitle("Central Carbohydrate Metabolism")

plot3c
```


PCoA cytochrome C oxidase, all factors, HISTORY AS SHAPE:
```{r PCoA: cytochrome C oxidase; all factors}
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate
pcoa_cytoCox <- cmdscale(cytoCoxBC, k=2, eig=TRUE, add=TRUE)

explainvar1b_cytoCox <- round(pcoa_cytoCox$eig[1] / sum(pcoa_cytoCox$eig), 3) * 100
explainvar2b_cytoCox <- round(pcoa_cytoCox$eig[2] / sum(pcoa_cytoCox$eig), 3) * 100
sum.eigc <- sum(explainvar1b_cytoCox, explainvar2b_cytoCox)

explainvar1b_cytoCox # 29.2
explainvar2b_cytoCox # 22.2

pcoa_cytoCox2.groups <- paste(cytoCoxRA_design$Treat, cytoCoxRA_design$Plant, cytoCoxRA_design$Hist, sep = "_")
pcoa_cytoCox2.points <- data.frame(pcoa_cytoCox$points, group = pcoa_cytoCox2.groups)

# Calculate Centroids (mean and SE)
pcoa_cytoCox2.L.centroids <- reshape2::melt(pcoa_cytoCox2.points, id="group", measure.vars = c("X1", "X2"))
pcoa_cytoCox2.centroids <- acast(pcoa_cytoCox2.L.centroids, variable ~ group, mean)
pcoa_cytoCox2.centroids.se <- acast(pcoa_cytoCox2.L.centroids, variable ~ group, se)
pcoa_cytoCox2.centroids.sd <- acast(pcoa_cytoCox2.L.centroids, variable ~ group, sd)

# Combine
pcoa_cytoCox2.cent.dataframe <- cbind(t(pcoa_cytoCox2.centroids), t(pcoa_cytoCox2.centroids.se))
colnames(pcoa_cytoCox2.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")

# No need to run again if these lists have already been declared:
#trt_H <- c("Baseline", "Baseline", "Dry","Dry", "Dry","Dry","Wet","Wet", "Wet","Wet") 
#plant_H <- c("Baseline", "Baseline", "No Plant","No Plant","Plant", "Plant", "No Plant","No Plant","Plant", "Plant")
#history <- c("Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet")
#trt_plant_combined <- c(paste(trt, plant, sep = "_"))

pcoa_cytoCox2.cent.dataframe.trts <- as.data.frame(pcoa_cytoCox2.cent.dataframe) 
pcoa_cytoCox2.cent.dataframe.trts$trt <- as.factor(trt_H)
pcoa_cytoCox2.cent.dataframe.trts$plant <- as.factor(plant_H)
pcoa_cytoCox2.cent.dataframe.trts$history <- as.factor(history)
pcoa_cytoCox2.cent.dataframe.trts$history_plant<- as.factor(history_plant_combined)
pcoa_cytoCox2.cent.dataframe.trts$centroid_or_sample <- "centroid"

# Add in points to data frame:
pcoa_cytoCox2.points_ForPlot <- pcoa_cytoCox2.points
colnames(pcoa_cytoCox2.points_ForPlot) <- c("V1", "V2", "group")
# Add labels to match ...cent.dataframe.trts
pcoa_cytoCox2.points_ForPlot$trt <- cytoCoxRA_design$Treat %>% str_replace_all(c("w" = "Wet", "_pre" = "Baseline", "d" = "Dry"))
pcoa_cytoCox2.points_ForPlot$plant <- cytoCoxRA_design$Plant %>% str_replace_all(c("n" = "No Plant", "y" = "Plant", "_pre" = "Baseline"))
pcoa_cytoCox2.points_ForPlot$history <- cytoCoxRA_design$Hist %>% str_replace_all(c("d" = "Dry", "w" = "Wet"))
pcoa_cytoCox2.points_ForPlot$history_plant <- c(paste(pcoa_cytoCox2.points_ForPlot$history, pcoa_cytoCox2.points_ForPlot$plant, sep = "_"))
#remove "group" column
pcoa_cytoCox2.points_ForPlot <- pcoa_cytoCox2.points_ForPlot[, -3]
#add V1e and V2e as zeros:
pcoa_cytoCox2.points_ForPlot$V1e <- numeric(24)
pcoa_cytoCox2.points_ForPlot$V2e <- numeric(24)
# Add column to distinguish sample points from centroids, and later use to set alpha in plot
pcoa_cytoCox2.points_ForPlot$centroid_or_sample <- "sample"

#Merge:
cytoCox_centroids_points_merged <- rbind(pcoa_cytoCox2.cent.dataframe.trts, pcoa_cytoCox2.points_ForPlot)

# Add points data frame (seems to be newer issue, used to be able to use $ on pcoa_denit_FINAL$points)
pcoa_cytoCox_DF <- as.data.frame(pcoa_cytoCox$points)
```
```{r envfit with GHGs ONLY}
#Run envfit to see which [GHG]s (start or finish) have good fit with ordination:
#fit_cytoCox <- envfit(pcoa_cytoCox$points, GHG_metaG, permutations = 10000)
#fit_cytoCox

#A_cytoCox <-as.list(fit_cytoCox$vectors)
#vec_cytoCox <- as.data.frame(fit_cytoCox$vectors$arrows*sqrt(fit_cytoCox$vectors$r)*0.15)
#p_cytoCox <- as.data.frame(A_cytoCox$pvals)
#vec_cytoCox <- cbind(vec_cytoCox, p_cytoCox)
#vec_cytoCox <- subset(vec_cytoCox, A_cytoCox$pvals<=0.1)
```
```{r cytoCox SSF envfit and PCoA}
#Run envfit with new Soil data (SSF = Soil Start & Finish):
fit_cytoCox_SSF <- envfit(pcoa_cytoCox_DF, SoilGHG_StartFinish_Envfit, permutations = 10000, set.seed(42))
fit_cytoCox_SSF

A_cytoCox_SSF <-as.list(fit_cytoCox_SSF$vectors)
vec_cytoCox_SSF <- as.data.frame(fit_cytoCox_SSF$vectors$arrows*sqrt(fit_cytoCox_SSF$vectors$r)*0.15)
p_cytoCox_SSF <- as.data.frame(A_cytoCox_SSF$pvals)
vec_cytoCox_SSF <- cbind(vec_cytoCox_SSF, p_cytoCox_SSF)
vec_cytoCox_SSF <- subset(vec_cytoCox_SSF, A_cytoCox_SSF$pvals<0.05)

#Plot using ggplot2
df2d <- as.data.frame(cytoCox_centroids_points_merged)

plot2d <- ggplot(df2d, aes(x=V1, y=V2, shape=history_plant, colour=trt)) + geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.003), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.003), colour="black") +   
  geom_point(aes(shape = history_plant, colour=trt, alpha = centroid_or_sample, size = centroid_or_sample), stroke=2) +  
  theme_bw() +
  scale_size_manual(values = c(6, 3), name = "Centroid or Sample")+
  scale_alpha_manual(values = c(1, 0.3), name = "Centroid or Sample")+
  scale_color_manual(values=c("black", "#997700", "#004488"), name="Treatment") +
  scale_shape_manual(values = c(7, 0, 15, 13, 1, 19), name="History_Plant") +
  xlab(glue("PCoA 1 ({explainvar1b_cytoCox}%)")) + ylab(glue("PCoA 2 ({explainvar2b_cytoCox}%)")) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  theme(panel.background = element_blank()) + 
  theme(axis.title = element_text(size=14), axis.text=element_text(size=14), 
          axis.text.x = element_text(size=14),  axis.text.y = element_text(size=14),
          panel.border = element_rect(colour = "black", linewidth=1.25)) +
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  theme(plot.title=element_text(size=14)) +
  theme(legend.text=element_text(size=14), legend.title = element_text(size=14))+
  guides(shape = guide_legend(override.aes = list(size = 4), order = 1), 
           colour = guide_legend(override.aes = list(pch=16, size = 4), order = 2)) +
    ggtitle("Cytochrome C Oxidase")

plot2d
```

```{r PCoA: cco FINAL}
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate
pcoa_cytoCox_FINAL <- cmdscale(cytoCoxBC_finish, k=2, eig=TRUE, add=TRUE) # k = 2 because we only plot 2 dimensions, add=TRUE corrects highly negative eigenvalues


explainvar1_cytoCox_FIN <- round(pcoa_cytoCox_FINAL$eig[1] / sum(pcoa_cytoCox_FINAL$eig), 3) * 100
explainvar2_cytoCox_FIN <- round(pcoa_cytoCox_FINAL$eig[2] / sum(pcoa_cytoCox_FINAL$eig), 3) * 100
sum.eig_cytoCox_FIN <- sum(explainvar1_cytoCox_FIN, explainvar2_cytoCox_FIN)

explainvar1_cytoCox_FIN # 33.2
explainvar2_cytoCox_FIN # 24.8

# Filter 'pre' samples out of *RA_design to get final samples only
cytoCoxRA_FIN_design <- cytoCoxRA_design %>% dplyr::filter(Treat != "_pre" & Plant != "_pre")

pcoa_cytoCox_FIN.groups <- paste(cytoCoxRA_FIN_design$Treat, cytoCoxRA_FIN_design$Plant, cytoCoxRA_FIN_design$Hist, sep = "_")
pcoa_cytoCox_FIN.points <- data.frame(pcoa_cytoCox_FINAL$points, group = pcoa_cytoCox_FIN.groups)

# Calculate Centroids (mean and SE)
pcoa_cytoCox_FIN.L.centroids <- reshape2::melt(pcoa_cytoCox_FIN.points, id="group", measure.vars = c("X1", "X2"))
pcoa_cytoCox_FIN.centroids <- acast(pcoa_cytoCox_FIN.L.centroids, variable ~ group, mean)
pcoa_cytoCox_FIN.centroids.se <- acast(pcoa_cytoCox_FIN.L.centroids, variable ~ group, se)
pcoa_cytoCox_FIN.centroids.sd <- acast(pcoa_cytoCox_FIN.L.centroids, variable ~ group, sd)

# Combine
pcoa_cytoCox_FIN.cent.dataframe <- cbind(t(pcoa_cytoCox_FIN.centroids), t(pcoa_cytoCox_FIN.centroids.se))
colnames(pcoa_cytoCox_FIN.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")

# Adding columns with the factors.
trt_FIN <- c("Dry","Dry", "Dry","Dry","Wet","Wet", "Wet","Wet") 
plant_FIN <- c("No Plant","No Plant","Plant", "Plant", "No Plant","No Plant","Plant", "Plant")
history_FIN <- c("Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet")
history_plant_combined_FIN <- c(paste(history_FIN, plant_FIN, sep = "_"))

pcoa_cytoCox_FIN.cent.dataframe.trts <- as.data.frame(pcoa_cytoCox_FIN.cent.dataframe) 
pcoa_cytoCox_FIN.cent.dataframe.trts$trt <- as.factor(trt_FIN)
pcoa_cytoCox_FIN.cent.dataframe.trts$plant <- as.factor(plant_FIN)
pcoa_cytoCox_FIN.cent.dataframe.trts$history <- as.factor(history_FIN)
pcoa_cytoCox_FIN.cent.dataframe.trts$history_plant<- as.factor(history_plant_combined_FIN)
pcoa_cytoCox_FIN.cent.dataframe.trts$centroid_or_sample <- "centroid"

# Add in points to data frame:
pcoa_cytoCox_FIN.points_ForPlot <- pcoa_cytoCox_FIN.points
colnames(pcoa_cytoCox_FIN.points_ForPlot) <- c("V1", "V2", "group")
# Add labels to match ...cent.dataframe.trts
pcoa_cytoCox_FIN.points_ForPlot$trt <- cytoCoxRA_FIN_design$Treat %>% str_replace_all(c("w" = "Wet", "_pre" = "Baseline", "d" = "Dry"))
pcoa_cytoCox_FIN.points_ForPlot$plant <- cytoCoxRA_FIN_design$Plant %>% str_replace_all(c("n" = "No Plant", "y" = "Plant", "_pre" = "Baseline"))
pcoa_cytoCox_FIN.points_ForPlot$history <- cytoCoxRA_FIN_design$Hist %>% str_replace_all(c("d" = "Dry", "w" = "Wet"))
pcoa_cytoCox_FIN.points_ForPlot$history_plant <- c(paste(pcoa_cytoCox_FIN.points_ForPlot$history, pcoa_cytoCox_FIN.points_ForPlot$plant, sep = "_"))
#remove "group" column
pcoa_cytoCox_FIN.points_ForPlot <- pcoa_cytoCox_FIN.points_ForPlot[, -3]
#add V1e and V2e as zeros:
pcoa_cytoCox_FIN.points_ForPlot$V1e <- numeric(16)
pcoa_cytoCox_FIN.points_ForPlot$V2e <- numeric(16)
# Add column to distinguish sample points from centroids, and later use to set alpha in plot
pcoa_cytoCox_FIN.points_ForPlot$centroid_or_sample <- "sample"

#Merge:
cytoCox_FIN_centroids_points_merged <- rbind(pcoa_cytoCox_FIN.cent.dataframe.trts, pcoa_cytoCox_FIN.points_ForPlot)

# Add points data frame (seems to be newer issue, used to be able to use $ on pcoa_denit_FINAL$points)
pcoa_cytoCox_FINAL_DF <- as.data.frame(pcoa_cytoCox_FINAL$points)

#Run envfit with new Soil data (SFO = Soil Finish Only):
fit_cytoCox_SFO <- envfit(pcoa_cytoCox_FINAL_DF, envfit_FINISH_numerical, permutations = 10000, set.seed(42))
fit_cytoCox_SFO

A_cytoCox_SFO <-as.list(fit_cytoCox_SFO$vectors)
vec_cytoCox_SFO <- as.data.frame(fit_cytoCox_SFO$vectors$arrows*sqrt(fit_cytoCox_SFO$vectors$r)*0.15)
p_cytoCox_SFO <- as.data.frame(A_cytoCox_SFO$pvals)
vec_cytoCox_SFO <- cbind(vec_cytoCox_SFO, p_cytoCox_SFO)
vec_cytoCox_SFO <- subset(vec_cytoCox_SFO, A_cytoCox_SFO$pvals<0.05)

# Add column with labels, for plotting:
vec_cytoCox_SFO$labels <- c("P", "Zn", "Cu")


#Plot using ggplot2
df3d <- as.data.frame(cytoCox_FIN_centroids_points_merged)

plot3d <- ggplot(df3d, aes(x=V1, y=V2, shape=history_plant, colour=trt)) + geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.003), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.003), colour="black") +   
  geom_point(aes(shape = history_plant, colour=trt, alpha = centroid_or_sample, size = centroid_or_sample), stroke=2) +  
  theme_bw() +
  scale_size_manual(values = c(6, 3), name = "Centroid or Sample")+
  scale_alpha_manual(values = c(1, 0.3), name = "Centroid or Sample")+
  scale_color_manual(values=c("#997700", "#004488"), name="Treatment") +
  scale_shape_manual(values = c(0, 15, 1, 19), name="History_Plant") +
  xlab(glue("PCoA 1 ({explainvar1_cytoCox_FIN}%)")) + ylab(glue("PCoA 2 ({explainvar2_cytoCox_FIN}%)")) + 
  geom_segment(data=vec_cytoCox_SFO, aes(x=0,xend=V1,y=0,yend=V2), linewidth=1, arrow = arrow(length = unit(0.2, "cm")),colour="black", alpha = 0.4,inherit.aes=F)+ 
 geom_text(data=vec_cytoCox_SFO, aes(x=V1-0.005, y=V2), label=vec_cytoCox_SFO$labels, size=4, inherit.aes=F)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  theme(panel.background = element_blank()) + 
  theme(axis.title = element_text(size=14), axis.text=element_text(size=14), 
          axis.text.x = element_text(size=14),  axis.text.y = element_text(size=14),
          panel.border = element_rect(colour = "black", linewidth=1.25)) +
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  theme(plot.title=element_text(size=14)) +
  theme(legend.text=element_text(size=14), legend.title = element_text(size=14))+
  guides(shape = guide_legend(override.aes = list(size = 4), order = 1), 
           colour = guide_legend(override.aes = list(pch=16, size = 4), order = 2)) +
    ggtitle("Cytochrome C Oxidase")

plot3d
```


Create 4-panel figure of the 4 PCoA plots above:
```{r 4-panel PCoA plot}
#require(ggpubr) #load in library for multi-panel figures (if not already loaded)

# 4 panel metaG PCoA with History, start and finish sample (SSF)
MetaG_pcoa_4panel_SSF <- ggarrange(plot2a, plot2b, plot2c, plot2d, labels = c("A", "B", "C", "D"), ncol = 2, nrow = 2, legend = "right", common.legend = T)

MetaG_pcoa_4panel_SSF

#ggsave("../figures/pub/metaG_pcoa4panel_H.tiff", plot = MetaG_pcoa_4panel_H, device="tiff", path=NULL, scale=1, height = 7, width = 10, limitsize = T, bg = "white", dpi = 300)

#ggsave("../figures/Thesis/metaG_pcoa4panel_SSF.jpg", plot = MetaG_pcoa_4panel_SSF, device=NULL, path=NULL, scale=1, height = 7, width = 10, limitsize = TRUE, dpi = 600)

# 4 panel metaG PCoA with History, Finish samples only (SFO)

MetaG_pcoa_4panel_SFO <- ggarrange(plot3a, plot3b, plot3c, plot3d, labels = c("A", "B", "C", "D"), ncol = 2, nrow = 2, legend = "right", common.legend = T)

MetaG_pcoa_4panel_SFO

#ggsave("../figures/Thesis/metaG_pcoa4panel_SFO.jpg", plot = MetaG_pcoa_4panel_SFO, device=NULL, path=NULL, scale=1, height = 7, width = 10, limitsize = TRUE, dpi = 600)
```


```{r PCoA: sulfate-sulfur assimilation, start&finish}
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate
pcoa_SO4_S_Assim <- cmdscale(SO4_S_AssimBC, k=2, eig=TRUE, add=TRUE)

explainvar1b_SO4_S_Assim <- round(pcoa_SO4_S_Assim$eig[1] / sum(pcoa_SO4_S_Assim$eig), 3) * 100
explainvar2b_SO4_S_Assim <- round(pcoa_SO4_S_Assim$eig[2] / sum(pcoa_SO4_S_Assim$eig), 3) * 100
sum.eigc <- sum(explainvar1b_SO4_S_Assim, explainvar2b_SO4_S_Assim)

explainvar1b_SO4_S_Assim # 29.4
explainvar2b_SO4_S_Assim # 13.9

pcoa_SO4_S_Assim2.groups <- paste(SO4_S_AssimRA_design$Treat, SO4_S_AssimRA_design$Plant, SO4_S_AssimRA_design$Hist, sep = "_")
pcoa_SO4_S_Assim2.points <- data.frame(pcoa_SO4_S_Assim$points, group = pcoa_SO4_S_Assim2.groups)

# Calculate Centroids (mean and SE)
pcoa_SO4_S_Assim2.L.centroids <- reshape2::melt(pcoa_SO4_S_Assim2.points, id="group", measure.vars = c("X1", "X2"))
pcoa_SO4_S_Assim2.centroids <- acast(pcoa_SO4_S_Assim2.L.centroids, variable ~ group, mean)
pcoa_SO4_S_Assim2.centroids.se <- acast(pcoa_SO4_S_Assim2.L.centroids, variable ~ group, se)
pcoa_SO4_S_Assim2.centroids.sd <- acast(pcoa_SO4_S_Assim2.L.centroids, variable ~ group, sd)

# Combine
pcoa_SO4_S_Assim2.cent.dataframe <- cbind(t(pcoa_SO4_S_Assim2.centroids), t(pcoa_SO4_S_Assim2.centroids.se))
colnames(pcoa_SO4_S_Assim2.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")

# No need to run again if these lists have already been declared:
#trt_H <- c("Baseline", "Baseline", "Dry","Dry", "Dry","Dry","Wet","Wet", "Wet","Wet") 
#plant_H <- c("Baseline", "Baseline", "No Plant","No Plant","Plant", "Plant", "No Plant","No Plant","Plant", "Plant")
#history <- c("Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet")
#trt_plant_combined <- c(paste(trt, plant, sep = "_"))

pcoa_SO4_S_Assim2.cent.dataframe.trts <- as.data.frame(pcoa_SO4_S_Assim2.cent.dataframe) 
pcoa_SO4_S_Assim2.cent.dataframe.trts$trt <- as.factor(trt_H)
pcoa_SO4_S_Assim2.cent.dataframe.trts$plant <- as.factor(plant_H)
pcoa_SO4_S_Assim2.cent.dataframe.trts$history <- as.factor(history)
pcoa_SO4_S_Assim2.cent.dataframe.trts$history_plant<- as.factor(history_plant_combined)
pcoa_SO4_S_Assim2.cent.dataframe.trts$centroid_or_sample <- "centroid"

# Add in points to data frame:
pcoa_SO4_S_Assim2.points_ForPlot <- pcoa_SO4_S_Assim2.points
colnames(pcoa_SO4_S_Assim2.points_ForPlot) <- c("V1", "V2", "group")
# Add labels to match ...cent.dataframe.trts
pcoa_SO4_S_Assim2.points_ForPlot$trt <- SO4_S_AssimRA_design$Treat %>% str_replace_all(c("w" = "Wet", "_pre" = "Baseline", "d" = "Dry"))
pcoa_SO4_S_Assim2.points_ForPlot$plant <- SO4_S_AssimRA_design$Plant %>% str_replace_all(c("n" = "No Plant", "y" = "Plant", "_pre" = "Baseline"))
pcoa_SO4_S_Assim2.points_ForPlot$history <- SO4_S_AssimRA_design$Hist %>% str_replace_all(c("d" = "Dry", "w" = "Wet"))
pcoa_SO4_S_Assim2.points_ForPlot$history_plant <- c(paste(pcoa_SO4_S_Assim2.points_ForPlot$history, pcoa_SO4_S_Assim2.points_ForPlot$plant, sep = "_"))
#remove "group" column
pcoa_SO4_S_Assim2.points_ForPlot <- pcoa_SO4_S_Assim2.points_ForPlot[, -3]
#add V1e and V2e as zeros:
pcoa_SO4_S_Assim2.points_ForPlot$V1e <- numeric(24)
pcoa_SO4_S_Assim2.points_ForPlot$V2e <- numeric(24)
# Add column to distinguish sample points from centroids, and later use to set alpha in plot
pcoa_SO4_S_Assim2.points_ForPlot$centroid_or_sample <- "sample"

#Merge:
SO4_S_Assim_centroids_points_merged <- rbind(pcoa_SO4_S_Assim2.cent.dataframe.trts, pcoa_SO4_S_Assim2.points_ForPlot)

# Add points data frame (seems to be newer issue, used to be able to use $ on pcoa_denit_FINAL$points)
pcoa_SO4_S_Assim_DF <- as.data.frame(pcoa_SO4_S_Assim$points)
```
```{r SO4_S_Assim envfit and PCoA}
#Run envfit with new Soil data (SSF = Soil Start & Finish):
fit_SO4_S_Assim_SSF <- envfit(pcoa_SO4_S_Assim_DF, SoilGHG_StartFinish_Envfit, permutations = 10000, set.seed(42))
fit_SO4_S_Assim_SSF

A_SO4_S_Assim_SSF <-as.list(fit_SO4_S_Assim_SSF$vectors)
vec_SO4_S_Assim_SSF <- as.data.frame(fit_SO4_S_Assim_SSF$vectors$arrows*sqrt(fit_SO4_S_Assim_SSF$vectors$r)*0.15)
p_SO4_S_Assim_SSF <- as.data.frame(A_SO4_S_Assim_SSF$pvals)
vec_SO4_S_Assim_SSF <- cbind(vec_SO4_S_Assim_SSF, p_SO4_S_Assim_SSF)
vec_SO4_S_Assim_SSF <- subset(vec_SO4_S_Assim_SSF, A_SO4_S_Assim_SSF$pvals<0.05)

# add vector lables for envfit on PCoA:
vec_SO4_S_Assim_SSF$vec_labels <- c(expression(paste("CO"[2])), "%paint removed (redox)")

#Plot using ggplot2
df4a <- as.data.frame(SO4_S_Assim_centroids_points_merged)

plot4a <- ggplot(df4a, aes(x=V1, y=V2, shape=history_plant, colour=trt)) + geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.003), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.003), colour="black") +   
  geom_point(aes(shape = history_plant, colour=trt, alpha = centroid_or_sample, size = centroid_or_sample), stroke=2) +  
  theme_bw() +
  scale_size_manual(values = c(6, 3), name = "Centroid or Sample")+
  scale_alpha_manual(values = c(1, 0.3), name = "Centroid or Sample")+
  scale_color_manual(values=c("black", "#997700", "#004488"), name="Treatment") +
  scale_shape_manual(values = c(7, 0, 15, 13, 1, 19), name="History_Plant") +
  xlab(glue("PCoA 1 ({explainvar1b_SO4_S_Assim}%)")) + ylab(glue("PCoA 2 ({explainvar2b_SO4_S_Assim}%)")) + 
  geom_segment(data=vec_SO4_S_Assim_SSF, aes(x=0,xend=V1,y=0,yend=V2), linewidth=1, arrow = arrow(length = unit(0.2, "cm")),colour="black", alpha = 0.4, inherit.aes=F)+ 
 geom_text(data=vec_SO4_S_Assim_SSF, aes(x=V1-0.004, y=V2-0.003), label= vec_SO4_S_Assim_SSF$vec_labels, size=4, inherit.aes=F)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  theme(panel.background = element_blank()) + 
  theme(axis.title = element_text(size=14), axis.text=element_text(size=14), 
          axis.text.x = element_text(size=14),  axis.text.y = element_text(size=14),
          panel.border = element_rect(colour = "black", linewidth=1.25)) +
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  theme(plot.title=element_text(size=14)) +
  theme(legend.text=element_text(size=14), legend.title = element_text(size=14))+
  guides(shape = guide_legend(override.aes = list(size = 4), order = 1), 
           colour = guide_legend(override.aes = list(pch=16, size = 4), order = 2)) +
    ggtitle("Sulfate-S Assimilation")

plot4a
```

```{r SO4_S_Assim finish only}
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate
pcoa_SO4_S_Assim_FINAL <- cmdscale(SO4_S_AssimBC_finish, k=2, eig=TRUE, add=TRUE) # k = 2 because we only plot 2 dimensions, add=TRUE corrects highly negative eigenvalues


explainvar1_SO4_S_Assim_FIN <- round(pcoa_SO4_S_Assim_FINAL$eig[1] / sum(pcoa_SO4_S_Assim_FINAL$eig), 3) * 100
explainvar2_SO4_S_Assim_FIN <- round(pcoa_SO4_S_Assim_FINAL$eig[2] / sum(pcoa_SO4_S_Assim_FINAL$eig), 3) * 100
sum.eig_SO4_S_Assim_FIN <- sum(explainvar1_SO4_S_Assim_FIN, explainvar2_SO4_S_Assim_FIN)

explainvar1_SO4_S_Assim_FIN # 34.5
explainvar2_SO4_S_Assim_FIN # 18.1

# Filter 'pre' samples out of *RA_design to get final samples only
SO4_S_AssimRA_FIN_design <- SO4_S_AssimRA_design %>% dplyr::filter(Treat != "_pre" & Plant != "_pre")

pcoa_SO4_S_Assim_FIN.groups <- paste(SO4_S_AssimRA_FIN_design$Treat, SO4_S_AssimRA_FIN_design$Plant, SO4_S_AssimRA_FIN_design$Hist, sep = "_")
pcoa_SO4_S_Assim_FIN.points <- data.frame(pcoa_SO4_S_Assim_FINAL$points, group = pcoa_SO4_S_Assim_FIN.groups)

# Calculate Centroids (mean and SE)
pcoa_SO4_S_Assim_FIN.L.centroids <- reshape2::melt(pcoa_SO4_S_Assim_FIN.points, id="group", measure.vars = c("X1", "X2"))
pcoa_SO4_S_Assim_FIN.centroids <- acast(pcoa_SO4_S_Assim_FIN.L.centroids, variable ~ group, mean)
pcoa_SO4_S_Assim_FIN.centroids.se <- acast(pcoa_SO4_S_Assim_FIN.L.centroids, variable ~ group, se)
pcoa_SO4_S_Assim_FIN.centroids.sd <- acast(pcoa_SO4_S_Assim_FIN.L.centroids, variable ~ group, sd)

# Combine
pcoa_SO4_S_Assim_FIN.cent.dataframe <- cbind(t(pcoa_SO4_S_Assim_FIN.centroids), t(pcoa_SO4_S_Assim_FIN.centroids.se))
colnames(pcoa_SO4_S_Assim_FIN.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")

# Adding columns with the factors. (don't need to run again, if already in Global Environment)
#trt_FIN <- c("Dry","Dry", "Dry","Dry","Wet","Wet", "Wet","Wet") 
#plant_FIN <- c("No Plant","No Plant","Plant", "Plant", "No Plant","No Plant","Plant", "Plant")
#history_FIN <- c("Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet")
#history_plant_combined_FIN <- c(paste(history_FIN, plant_FIN, sep = "_"))

pcoa_SO4_S_Assim_FIN.cent.dataframe.trts <- as.data.frame(pcoa_SO4_S_Assim_FIN.cent.dataframe) 
pcoa_SO4_S_Assim_FIN.cent.dataframe.trts$trt <- as.factor(trt_FIN)
pcoa_SO4_S_Assim_FIN.cent.dataframe.trts$plant <- as.factor(plant_FIN)
pcoa_SO4_S_Assim_FIN.cent.dataframe.trts$history <- as.factor(history_FIN)
pcoa_SO4_S_Assim_FIN.cent.dataframe.trts$history_plant<- as.factor(history_plant_combined_FIN)
pcoa_SO4_S_Assim_FIN.cent.dataframe.trts$centroid_or_sample <- "centroid"

# Add in points to data frame:
pcoa_SO4_S_Assim_FIN.points_ForPlot <- pcoa_SO4_S_Assim_FIN.points
colnames(pcoa_SO4_S_Assim_FIN.points_ForPlot) <- c("V1", "V2", "group")
# Add labels to match ...cent.dataframe.trts
pcoa_SO4_S_Assim_FIN.points_ForPlot$trt <- SO4_S_AssimRA_FIN_design$Treat %>% str_replace_all(c("w" = "Wet", "_pre" = "Baseline", "d" = "Dry"))
pcoa_SO4_S_Assim_FIN.points_ForPlot$plant <- SO4_S_AssimRA_FIN_design$Plant %>% str_replace_all(c("n" = "No Plant", "y" = "Plant", "_pre" = "Baseline"))
pcoa_SO4_S_Assim_FIN.points_ForPlot$history <- SO4_S_AssimRA_FIN_design$Hist %>% str_replace_all(c("d" = "Dry", "w" = "Wet"))
pcoa_SO4_S_Assim_FIN.points_ForPlot$history_plant <- c(paste(pcoa_SO4_S_Assim_FIN.points_ForPlot$history, pcoa_SO4_S_Assim_FIN.points_ForPlot$plant, sep = "_"))
#remove "group" column
pcoa_SO4_S_Assim_FIN.points_ForPlot <- pcoa_SO4_S_Assim_FIN.points_ForPlot[, -3]
#add V1e and V2e as zeros:
pcoa_SO4_S_Assim_FIN.points_ForPlot$V1e <- numeric(16)
pcoa_SO4_S_Assim_FIN.points_ForPlot$V2e <- numeric(16)
# Add column to distinguish sample points from centroids, and later use to set alpha in plot
pcoa_SO4_S_Assim_FIN.points_ForPlot$centroid_or_sample <- "sample"

#Merge:
SO4_S_Assim_FIN_centroids_points_merged <- rbind(pcoa_SO4_S_Assim_FIN.cent.dataframe.trts, pcoa_SO4_S_Assim_FIN.points_ForPlot)

# Add points data frame (seems to be newer issue, used to be able to use $ on pcoa_denit_FINAL$points)
pcoa_SO4_S_Assim_FINAL_DF <- as.data.frame(pcoa_SO4_S_Assim_FINAL$points)


#Run envfit with new Soil data (SFO = Soil Finish Only):
fit_SO4_S_Assim_SFO <- envfit(pcoa_SO4_S_Assim_FINAL_DF, envfit_FINISH_numerical, permutations = 10000, set.seed(42))
fit_SO4_S_Assim_SFO

A_SO4_S_Assim_SFO <-as.list(fit_SO4_S_Assim_SFO$vectors)
vec_SO4_S_Assim_SFO <- as.data.frame(fit_SO4_S_Assim_SFO$vectors$arrows*sqrt(fit_SO4_S_Assim_SFO$vectors$r)*0.15)
p_SO4_S_Assim_SFO <- as.data.frame(A_SO4_S_Assim_SFO$pvals)
vec_SO4_S_Assim_SFO <- cbind(vec_SO4_S_Assim_SFO, p_SO4_S_Assim_SFO)
vec_SO4_S_Assim_SFO <- subset(vec_SO4_S_Assim_SFO, A_SO4_S_Assim_SFO$pvals<0.05)

# Add column with labels, for plotting:
vec_SO4_S_Assim_SFO$labels <- c("Mg", "Ca", "S", "Zn", "Mn", "Cu", "CEC")


#Plot using ggplot2
df5a <- as.data.frame(SO4_S_Assim_FIN_centroids_points_merged)

plot5a <- ggplot(df5a, aes(x=V1, y=V2, shape=history_plant, colour=trt)) + geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.003), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.003), colour="black") +   
  geom_point(aes(shape = history_plant, colour=trt, alpha = centroid_or_sample, size = centroid_or_sample), stroke=2) +  
  theme_bw() +
  scale_size_manual(values = c(6, 3), name = "Centroid or Sample")+
  scale_alpha_manual(values = c(1, 0.3), name = "Centroid or Sample")+
  scale_color_manual(values=c("#997700", "#004488"), name="Treatment") +
  scale_shape_manual(values = c(0, 15, 1, 19), name="History_Plant") +
  xlab(glue("PCoA 1 ({explainvar1_SO4_S_Assim_FIN}%)")) + ylab(glue("PCoA 2 ({explainvar2_SO4_S_Assim_FIN}%)")) + 
  geom_segment(data=vec_SO4_S_Assim_SFO, aes(x=0,xend=V1,y=0,yend=V2), linewidth=1, arrow = arrow(length = unit(0.2, "cm")),colour="black", alpha = 0.4,inherit.aes=F)+ 
 geom_text(data=vec_SO4_S_Assim_SFO, aes(x=V1-0.007, y=V2+0.003), label=vec_SO4_S_Assim_SFO$labels, size=4, inherit.aes=F)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  theme(panel.background = element_blank()) + 
  theme(axis.title = element_text(size=14), axis.text=element_text(size=14), 
          axis.text.x = element_text(size=14),  axis.text.y = element_text(size=14),
          panel.border = element_rect(colour = "black", linewidth=1.25)) +
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  theme(plot.title=element_text(size=14)) +
  theme(legend.text=element_text(size=14), legend.title = element_text(size=14))+
  guides(shape = guide_legend(override.aes = list(size = 4), order = 1), 
           colour = guide_legend(override.aes = list(pch=16, size = 4), order = 2)) +
    ggtitle("Sulfate-S Assimilation Week 8")

plot5a
```

```{r PCOA: assimilatory sulfate reduction}
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate
pcoa_Assim_SO4_red <- cmdscale(Assim_SO4_redBC, k=2, eig=TRUE, add=TRUE)

explainvar1b_Assim_SO4_red <- round(pcoa_Assim_SO4_red$eig[1] / sum(pcoa_Assim_SO4_red$eig), 3) * 100
explainvar2b_Assim_SO4_red <- round(pcoa_Assim_SO4_red$eig[2] / sum(pcoa_Assim_SO4_red$eig), 3) * 100
sum.eigc <- sum(explainvar1b_Assim_SO4_red, explainvar2b_Assim_SO4_red)

explainvar1b_Assim_SO4_red # 32.9
explainvar2b_Assim_SO4_red # 15.2

pcoa_Assim_SO4_red2.groups <- paste(Assim_SO4_redRA_design$Treat, Assim_SO4_redRA_design$Plant, Assim_SO4_redRA_design$Hist, sep = "_")
pcoa_Assim_SO4_red2.points <- data.frame(pcoa_Assim_SO4_red$points, group = pcoa_Assim_SO4_red2.groups)

# Calculate Centroids (mean and SE)
pcoa_Assim_SO4_red2.L.centroids <- reshape2::melt(pcoa_Assim_SO4_red2.points, id="group", measure.vars = c("X1", "X2"))
pcoa_Assim_SO4_red2.centroids <- acast(pcoa_Assim_SO4_red2.L.centroids, variable ~ group, mean)
pcoa_Assim_SO4_red2.centroids.se <- acast(pcoa_Assim_SO4_red2.L.centroids, variable ~ group, se)
pcoa_Assim_SO4_red2.centroids.sd <- acast(pcoa_Assim_SO4_red2.L.centroids, variable ~ group, sd)

# Combine
pcoa_Assim_SO4_red2.cent.dataframe <- cbind(t(pcoa_Assim_SO4_red2.centroids), t(pcoa_Assim_SO4_red2.centroids.se))
colnames(pcoa_Assim_SO4_red2.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")

# No need to run again if these lists have already been declared:
#trt_H <- c("Baseline", "Baseline", "Dry","Dry", "Dry","Dry","Wet","Wet", "Wet","Wet") 
#plant_H <- c("Baseline", "Baseline", "No Plant","No Plant","Plant", "Plant", "No Plant","No Plant","Plant", "Plant")
#history <- c("Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet")
#trt_plant_combined <- c(paste(trt, plant, sep = "_"))

pcoa_Assim_SO4_red2.cent.dataframe.trts <- as.data.frame(pcoa_Assim_SO4_red2.cent.dataframe) 
pcoa_Assim_SO4_red2.cent.dataframe.trts$trt <- as.factor(trt_H)
pcoa_Assim_SO4_red2.cent.dataframe.trts$plant <- as.factor(plant_H)
pcoa_Assim_SO4_red2.cent.dataframe.trts$history <- as.factor(history)
pcoa_Assim_SO4_red2.cent.dataframe.trts$history_plant<- as.factor(history_plant_combined)
pcoa_Assim_SO4_red2.cent.dataframe.trts$centroid_or_sample <- "centroid"

# Add in points to data frame:
pcoa_Assim_SO4_red2.points_ForPlot <- pcoa_Assim_SO4_red2.points
colnames(pcoa_Assim_SO4_red2.points_ForPlot) <- c("V1", "V2", "group")
# Add labels to match ...cent.dataframe.trts
pcoa_Assim_SO4_red2.points_ForPlot$trt <- Assim_SO4_redRA_design$Treat %>% str_replace_all(c("w" = "Wet", "_pre" = "Baseline", "d" = "Dry"))
pcoa_Assim_SO4_red2.points_ForPlot$plant <- Assim_SO4_redRA_design$Plant %>% str_replace_all(c("n" = "No Plant", "y" = "Plant", "_pre" = "Baseline"))
pcoa_Assim_SO4_red2.points_ForPlot$history <- Assim_SO4_redRA_design$Hist %>% str_replace_all(c("d" = "Dry", "w" = "Wet"))
pcoa_Assim_SO4_red2.points_ForPlot$history_plant <- c(paste(pcoa_Assim_SO4_red2.points_ForPlot$history, pcoa_Assim_SO4_red2.points_ForPlot$plant, sep = "_"))
#remove "group" column
pcoa_Assim_SO4_red2.points_ForPlot <- pcoa_Assim_SO4_red2.points_ForPlot[, -3]
#add V1e and V2e as zeros:
pcoa_Assim_SO4_red2.points_ForPlot$V1e <- numeric(24)
pcoa_Assim_SO4_red2.points_ForPlot$V2e <- numeric(24)
# Add column to distinguish sample points from centroids, and later use to set alpha in plot
pcoa_Assim_SO4_red2.points_ForPlot$centroid_or_sample <- "sample"

#Merge:
Assim_SO4_red_centroids_points_merged <- rbind(pcoa_Assim_SO4_red2.cent.dataframe.trts, pcoa_Assim_SO4_red2.points_ForPlot)

# Add points data frame (seems to be newer issue, used to be able to use $ on pcoa_denit_FINAL$points)
pcoa_Assim_SO4_red_DF <- as.data.frame(pcoa_Assim_SO4_red$points)
```
```{r Assim_SO4_red envfit and PCoA}
#Run envfit with new Soil data (SSF = Soil Start & Finish):
fit_Assim_SO4_red_SSF <- envfit(pcoa_Assim_SO4_red_DF, SoilGHG_StartFinish_Envfit, permutations = 10000, set.seed(42))
fit_Assim_SO4_red_SSF

A_Assim_SO4_red_SSF <-as.list(fit_Assim_SO4_red_SSF$vectors)
vec_Assim_SO4_red_SSF <- as.data.frame(fit_Assim_SO4_red_SSF$vectors$arrows*sqrt(fit_Assim_SO4_red_SSF$vectors$r)*0.15)
p_Assim_SO4_red_SSF <- as.data.frame(A_Assim_SO4_red_SSF$pvals)
vec_Assim_SO4_red_SSF <- cbind(vec_Assim_SO4_red_SSF, p_Assim_SO4_red_SSF)
vec_Assim_SO4_red_SSF <- subset(vec_Assim_SO4_red_SSF, A_Assim_SO4_red_SSF$pvals<0.05)

# Add a column with nudges for nudge_x and nudge_y:
vec_Assim_SO4_red_SSF$X_nudge <- c(0.01, 0)
vec_Assim_SO4_red_SSF$Y_nudge <- c(-0.0035, 0.005)

# Add column with vector labels for envfit on PCoA:
vec_Assim_SO4_red_SSF$vec_labels <- c(expression(paste("CO"[2])), "%paint removed (redox)")

#Plot using ggplot2
df4b <- as.data.frame(Assim_SO4_red_centroids_points_merged)

plot4b <- ggplot(df4b, aes(x=V1, y=V2, shape=history_plant, colour=trt)) + geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.003), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.003), colour="black") +   
  geom_point(aes(shape = history_plant, colour=trt, alpha = centroid_or_sample, size = centroid_or_sample), stroke=2) +  
  theme_bw() +
  scale_size_manual(values = c(6, 3), name = "Centroid or Sample")+
  scale_alpha_manual(values = c(1, 0.3), name = "Centroid or Sample")+
  scale_color_manual(values=c("black", "#997700", "#004488"), name="Treatment") +
  scale_shape_manual(values = c(7, 0, 15, 13, 1, 19), name="History_Plant") +
  xlab(glue("PCoA 1 ({explainvar1b_Assim_SO4_red}%)")) + ylab(glue("PCoA 2 ({explainvar2b_Assim_SO4_red}%)")) + 
  geom_segment(data=vec_Assim_SO4_red_SSF, aes(x=0,xend=V1,y=0,yend=V2), linewidth=1, arrow = arrow(length = unit(0.2, "cm")),colour="black", alpha = 0.4, inherit.aes=F)+ 
 geom_text(data=vec_Assim_SO4_red_SSF, aes(x=V1, y=V2), label= vec_Assim_SO4_red_SSF$vec_labels, size=4, nudge_x = vec_Assim_SO4_red_SSF$X_nudge, nudge_y = vec_Assim_SO4_red_SSF$Y_nudge, inherit.aes=F)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  theme(panel.background = element_blank()) + 
  theme(axis.title = element_text(size=14), axis.text=element_text(size=14), 
          axis.text.x = element_text(size=14),  axis.text.y = element_text(size=14),
          panel.border = element_rect(colour = "black", linewidth=1.25)) +
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  theme(plot.title=element_text(size=14)) +
  theme(legend.text=element_text(size=14), legend.title = element_text(size=14))+
  guides(shape = guide_legend(override.aes = list(size = 4), order = 1), 
           colour = guide_legend(override.aes = list(pch=16, size = 4), order = 2)) +
    ggtitle("Assimilatory Sulfate Reduction")

plot4b

#ggsave("../figures/Thesis/Assim_sulf_red.png", plot = plot4b, path=NULL, device="png", scale=1, height = 7, width = 10, limitsize = TRUE, dpi = 300)
```

```{r PCoA: Assim_SO4_red finish only}
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate
pcoa_Assim_SO4_red_FINAL <- cmdscale(Assim_SO4_redBC_finish, k=2, eig=TRUE, add=TRUE) # k = 2 because we only plot 2 dimensions, add=TRUE corrects highly negative eigenvalues


explainvar1_Assim_SO4_red_FIN <- round(pcoa_Assim_SO4_red_FINAL$eig[1] / sum(pcoa_Assim_SO4_red_FINAL$eig), 3) * 100
explainvar2_Assim_SO4_red_FIN <- round(pcoa_Assim_SO4_red_FINAL$eig[2] / sum(pcoa_Assim_SO4_red_FINAL$eig), 3) * 100
sum.eig_Assim_SO4_red_FIN <- sum(explainvar1_Assim_SO4_red_FIN, explainvar2_Assim_SO4_red_FIN)

explainvar1_Assim_SO4_red_FIN # 40.8
explainvar2_Assim_SO4_red_FIN # 19.4

# Filter 'pre' samples out of *RA_design to get final samples only
Assim_SO4_redRA_FIN_design <- Assim_SO4_redRA_design %>% dplyr::filter(Treat != "_pre" & Plant != "_pre")

pcoa_Assim_SO4_red_FIN.groups <- paste(Assim_SO4_redRA_FIN_design$Treat, Assim_SO4_redRA_FIN_design$Plant, Assim_SO4_redRA_FIN_design$Hist, sep = "_")
pcoa_Assim_SO4_red_FIN.points <- data.frame(pcoa_Assim_SO4_red_FINAL$points, group = pcoa_Assim_SO4_red_FIN.groups)

# Calculate Centroids (mean and SE)
pcoa_Assim_SO4_red_FIN.L.centroids <- reshape2::melt(pcoa_Assim_SO4_red_FIN.points, id="group", measure.vars = c("X1", "X2"))
pcoa_Assim_SO4_red_FIN.centroids <- acast(pcoa_Assim_SO4_red_FIN.L.centroids, variable ~ group, mean)
pcoa_Assim_SO4_red_FIN.centroids.se <- acast(pcoa_Assim_SO4_red_FIN.L.centroids, variable ~ group, se)
pcoa_Assim_SO4_red_FIN.centroids.sd <- acast(pcoa_Assim_SO4_red_FIN.L.centroids, variable ~ group, sd)

# Combine
pcoa_Assim_SO4_red_FIN.cent.dataframe <- cbind(t(pcoa_Assim_SO4_red_FIN.centroids), t(pcoa_Assim_SO4_red_FIN.centroids.se))
colnames(pcoa_Assim_SO4_red_FIN.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")

# Adding columns with the factors. (No need to run again if already in Global Environment)
#trt_FIN <- c("Dry","Dry", "Dry","Dry","Wet","Wet", "Wet","Wet") 
#plant_FIN <- c("No Plant","No Plant","Plant", "Plant", "No Plant","No Plant","Plant", "Plant")
#history_FIN <- c("Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet")
#history_plant_combined_FIN <- c(paste(history_FIN, plant_FIN, sep = "_"))

pcoa_Assim_SO4_red_FIN.cent.dataframe.trts <- as.data.frame(pcoa_Assim_SO4_red_FIN.cent.dataframe) 
pcoa_Assim_SO4_red_FIN.cent.dataframe.trts$trt <- as.factor(trt_FIN)
pcoa_Assim_SO4_red_FIN.cent.dataframe.trts$plant <- as.factor(plant_FIN)
pcoa_Assim_SO4_red_FIN.cent.dataframe.trts$history <- as.factor(history_FIN)
pcoa_Assim_SO4_red_FIN.cent.dataframe.trts$history_plant<- as.factor(history_plant_combined_FIN)
pcoa_Assim_SO4_red_FIN.cent.dataframe.trts$centroid_or_sample <- "centroid"

# Add in points to data frame:
pcoa_Assim_SO4_red_FIN.points_ForPlot <- pcoa_Assim_SO4_red_FIN.points
colnames(pcoa_Assim_SO4_red_FIN.points_ForPlot) <- c("V1", "V2", "group")
# Add labels to match ...cent.dataframe.trts
pcoa_Assim_SO4_red_FIN.points_ForPlot$trt <- Assim_SO4_redRA_FIN_design$Treat %>% str_replace_all(c("w" = "Wet", "_pre" = "Baseline", "d" = "Dry"))
pcoa_Assim_SO4_red_FIN.points_ForPlot$plant <- Assim_SO4_redRA_FIN_design$Plant %>% str_replace_all(c("n" = "No Plant", "y" = "Plant", "_pre" = "Baseline"))
pcoa_Assim_SO4_red_FIN.points_ForPlot$history <- Assim_SO4_redRA_FIN_design$Hist %>% str_replace_all(c("d" = "Dry", "w" = "Wet"))
pcoa_Assim_SO4_red_FIN.points_ForPlot$history_plant <- c(paste(pcoa_Assim_SO4_red_FIN.points_ForPlot$history, pcoa_Assim_SO4_red_FIN.points_ForPlot$plant, sep = "_"))
#remove "group" column
pcoa_Assim_SO4_red_FIN.points_ForPlot <- pcoa_Assim_SO4_red_FIN.points_ForPlot[, -3]
#add V1e and V2e as zeros:
pcoa_Assim_SO4_red_FIN.points_ForPlot$V1e <- numeric(16)
pcoa_Assim_SO4_red_FIN.points_ForPlot$V2e <- numeric(16)
# Add column to distinguish sample points from centroids, and later use to set alpha in plot
pcoa_Assim_SO4_red_FIN.points_ForPlot$centroid_or_sample <- "sample"

#Merge:
Assim_SO4_red_FIN_centroids_points_merged <- rbind(pcoa_Assim_SO4_red_FIN.cent.dataframe.trts, pcoa_Assim_SO4_red_FIN.points_ForPlot)

# Add points data frame (seems to be newer issue, used to be able to use $ on pcoa_denit_FINAL$points)
pcoa_Assim_SO4_red_FINAL_DF <- as.data.frame(pcoa_Assim_SO4_red_FINAL$points)

#Run envfit with new Soil data (SFO = Soil Finish Only):
fit_Assim_SO4_red_SFO <- envfit(pcoa_Assim_SO4_red_FINAL_DF, envfit_FINISH_numerical, permutations = 10000, set.seed(42))
fit_Assim_SO4_red_SFO

A_Assim_SO4_red_SFO <-as.list(fit_Assim_SO4_red_SFO$vectors)
vec_Assim_SO4_red_SFO <- as.data.frame(fit_Assim_SO4_red_SFO$vectors$arrows*sqrt(fit_Assim_SO4_red_SFO$vectors$r)*0.15)
p_Assim_SO4_red_SFO <- as.data.frame(A_Assim_SO4_red_SFO$pvals)
vec_Assim_SO4_red_SFO <- cbind(vec_Assim_SO4_red_SFO, p_Assim_SO4_red_SFO)
vec_Assim_SO4_red_SFO <- subset(vec_Assim_SO4_red_SFO, A_Assim_SO4_red_SFO$pvals<0.05)

# Add column with labels, for plotting:
vec_Assim_SO4_red_SFO$labels <- c("%C", "%N","Mg", "Ca", "pH","S", "Zn", "Mn", "Cu", "CEC")


#Plot using ggplot2
df5b <- as.data.frame(Assim_SO4_red_FIN_centroids_points_merged)

plot5b <- ggplot(df5b, aes(x=V1, y=V2, shape=history_plant, colour=trt)) + geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.003), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.003), colour="black") +   
  geom_point(aes(shape = history_plant, colour=trt, alpha = centroid_or_sample, size = centroid_or_sample), stroke=2) +  
  theme_bw() +
  scale_size_manual(values = c(6, 3), name = "Centroid or Sample")+
  scale_alpha_manual(values = c(1, 0.3), name = "Centroid or Sample")+
  scale_color_manual(values=c("#997700", "#004488"), name="Treatment") +
  scale_shape_manual(values = c(0, 15, 1, 19), name="History_Plant") +
  xlab(glue("PCoA 1 ({explainvar1_Assim_SO4_red_FIN}%)")) + ylab(glue("PCoA 2 ({explainvar2_Assim_SO4_red_FIN}%)"))+
  geom_segment(data=vec_Assim_SO4_red_SFO, aes(x=0,xend=V1,y=0,yend=V2), linewidth=1, arrow = arrow(length = unit(0.2, "cm")),colour="black", alpha = 0.4,inherit.aes=F)+ 
 geom_text(data=vec_Assim_SO4_red_SFO, aes(x=V1-0.005, y=V2+0.002), label=vec_Assim_SO4_red_SFO$labels, size=4, inherit.aes=F)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  theme(panel.background = element_blank()) + 
  theme(axis.title = element_text(size=14), axis.text=element_text(size=14), 
          axis.text.x = element_text(size=14),  axis.text.y = element_text(size=14),
          panel.border = element_rect(colour = "black", linewidth=1.25)) +
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  theme(plot.title=element_text(size=14)) +
  theme(legend.text=element_text(size=14), legend.title = element_text(size=14))+
  guides(shape = guide_legend(override.aes = list(size = 4), order = 1), 
           colour = guide_legend(override.aes = list(pch=16, size = 4), order = 2)) +
    ggtitle("Assimilatory Sulfate Reduction Week 8")

plot5b
```

```{r PCoA: dissimilatory sulfate reduction}
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate
pcoa_Dissim_SO4_red <- cmdscale(Dissim_SO4_redBC, k=2, eig=TRUE, add=TRUE)

explainvar1b_Dissim_SO4_red <- round(pcoa_Dissim_SO4_red$eig[1] / sum(pcoa_Dissim_SO4_red$eig), 3) * 100
explainvar2b_Dissim_SO4_red <- round(pcoa_Dissim_SO4_red$eig[2] / sum(pcoa_Dissim_SO4_red$eig), 3) * 100
sum.eigc <- sum(explainvar1b_Dissim_SO4_red, explainvar2b_Dissim_SO4_red)

explainvar1b_Dissim_SO4_red # 41.3
explainvar2b_Dissim_SO4_red # 17.6

pcoa_Dissim_SO4_red2.groups <- paste(Dissim_SO4_redRA_design$Treat, Dissim_SO4_redRA_design$Plant, Dissim_SO4_redRA_design$Hist, sep = "_")
pcoa_Dissim_SO4_red2.points <- data.frame(pcoa_Dissim_SO4_red$points, group = pcoa_Dissim_SO4_red2.groups)

# Calculate Centroids (mean and SE)
pcoa_Dissim_SO4_red2.L.centroids <- reshape2::melt(pcoa_Dissim_SO4_red2.points, id="group", measure.vars = c("X1", "X2"))
pcoa_Dissim_SO4_red2.centroids <- acast(pcoa_Dissim_SO4_red2.L.centroids, variable ~ group, mean)
pcoa_Dissim_SO4_red2.centroids.se <- acast(pcoa_Dissim_SO4_red2.L.centroids, variable ~ group, se)
pcoa_Dissim_SO4_red2.centroids.sd <- acast(pcoa_Dissim_SO4_red2.L.centroids, variable ~ group, sd)

# Combine
pcoa_Dissim_SO4_red2.cent.dataframe <- cbind(t(pcoa_Dissim_SO4_red2.centroids), t(pcoa_Dissim_SO4_red2.centroids.se))
colnames(pcoa_Dissim_SO4_red2.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")

# No need to run again if these lists have already been declared:
#trt_H <- c("Baseline", "Baseline", "Dry","Dry", "Dry","Dry","Wet","Wet", "Wet","Wet") 
#plant_H <- c("Baseline", "Baseline", "No Plant","No Plant","Plant", "Plant", "No Plant","No Plant","Plant", "Plant")
#history <- c("Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet")
#trt_plant_combined <- c(paste(trt, plant, sep = "_"))

pcoa_Dissim_SO4_red2.cent.dataframe.trts <- as.data.frame(pcoa_Dissim_SO4_red2.cent.dataframe) 
pcoa_Dissim_SO4_red2.cent.dataframe.trts$trt <- as.factor(trt_H)
pcoa_Dissim_SO4_red2.cent.dataframe.trts$plant <- as.factor(plant_H)
pcoa_Dissim_SO4_red2.cent.dataframe.trts$history <- as.factor(history)
pcoa_Dissim_SO4_red2.cent.dataframe.trts$history_plant<- as.factor(history_plant_combined)
pcoa_Dissim_SO4_red2.cent.dataframe.trts$centroid_or_sample <- "centroid"

# Add in points to data frame:
pcoa_Dissim_SO4_red2.points_ForPlot <- pcoa_Dissim_SO4_red2.points
colnames(pcoa_Dissim_SO4_red2.points_ForPlot) <- c("V1", "V2", "group")
# Add labels to match ...cent.dataframe.trts
pcoa_Dissim_SO4_red2.points_ForPlot$trt <- Dissim_SO4_redRA_design$Treat %>% str_replace_all(c("w" = "Wet", "_pre" = "Baseline", "d" = "Dry"))
pcoa_Dissim_SO4_red2.points_ForPlot$plant <- Dissim_SO4_redRA_design$Plant %>% str_replace_all(c("n" = "No Plant", "y" = "Plant", "_pre" = "Baseline"))
pcoa_Dissim_SO4_red2.points_ForPlot$history <- Dissim_SO4_redRA_design$Hist %>% str_replace_all(c("d" = "Dry", "w" = "Wet"))
pcoa_Dissim_SO4_red2.points_ForPlot$history_plant <- c(paste(pcoa_Dissim_SO4_red2.points_ForPlot$history, pcoa_Dissim_SO4_red2.points_ForPlot$plant, sep = "_"))
#remove "group" column
pcoa_Dissim_SO4_red2.points_ForPlot <- pcoa_Dissim_SO4_red2.points_ForPlot[, -3]
#add V1e and V2e as zeros:
pcoa_Dissim_SO4_red2.points_ForPlot$V1e <- numeric(24)
pcoa_Dissim_SO4_red2.points_ForPlot$V2e <- numeric(24)
# Add column to distinguish sample points from centroids, and later use to set alpha in plot
pcoa_Dissim_SO4_red2.points_ForPlot$centroid_or_sample <- "sample"

#Merge:
Dissim_SO4_red_centroids_points_merged <- rbind(pcoa_Dissim_SO4_red2.cent.dataframe.trts, pcoa_Dissim_SO4_red2.points_ForPlot)

# Add points data frame (seems to be newer issue, used to be able to use $ on pcoa_denit_FINAL$points)
pcoa_Dissim_SO4_red_DF <- as.data.frame(pcoa_Dissim_SO4_red$points)
```
```{r Dissim_SO4_red envfit and PCoA}
#Run envfit with new Soil data (SSF = Soil Start & Finish):
fit_Dissim_SO4_red_SSF <- envfit(pcoa_Dissim_SO4_red_DF, SoilGHG_StartFinish_Envfit, permutations = 10000, set.seed(42))
fit_Dissim_SO4_red_SSF

A_Dissim_SO4_red_SSF <-as.list(fit_Dissim_SO4_red_SSF$vectors)
vec_Dissim_SO4_red_SSF <- as.data.frame(fit_Dissim_SO4_red_SSF$vectors$arrows*sqrt(fit_Dissim_SO4_red_SSF$vectors$r)*0.15)
p_Dissim_SO4_red_SSF <- as.data.frame(A_Dissim_SO4_red_SSF$pvals)
vec_Dissim_SO4_red_SSF <- cbind(vec_Dissim_SO4_red_SSF, p_Dissim_SO4_red_SSF)
vec_Dissim_SO4_red_SSF <- subset(vec_Dissim_SO4_red_SSF, A_Dissim_SO4_red_SSF$pvals<0.05)

#Plot using ggplot2
df4c <- as.data.frame(Dissim_SO4_red_centroids_points_merged)

plot4c <- ggplot(df4c, aes(x=V1, y=V2, shape=history_plant, colour=trt)) + geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.003), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.003), colour="black") +   
  geom_point(aes(shape = history_plant, colour=trt, alpha = centroid_or_sample, size = centroid_or_sample), stroke=2) +  
  theme_bw() +
  scale_size_manual(values = c(6, 3), name = "Centroid or Sample")+
  scale_alpha_manual(values = c(1, 0.3), name = "Centroid or Sample")+
  scale_color_manual(values=c("black", "#997700", "#004488"), name="Treatment") +
  scale_shape_manual(values = c(7, 0, 15, 13, 1, 19), name="History_Plant") +
  xlab(glue("PCoA 1 ({explainvar1b_Dissim_SO4_red}%)")) + ylab(glue("PCoA 2 ({explainvar2b_Dissim_SO4_red}%)")) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  theme(panel.background = element_blank()) + 
  theme(axis.title = element_text(size=14), axis.text=element_text(size=14), 
          axis.text.x = element_text(size=14),  axis.text.y = element_text(size=14),
          panel.border = element_rect(colour = "black", linewidth=1.25)) +
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  theme(plot.title=element_text(size=14)) +
  theme(legend.text=element_text(size=14), legend.title = element_text(size=14))+
  guides(shape = guide_legend(override.aes = list(size = 4), order = 1), 
           colour = guide_legend(override.aes = list(pch=16, size = 4), order = 2)) +
    ggtitle("Dissimilatory Sulfate Reduction")

plot4c
```

```{r PCoA: dissimilatory sulfate reduction, finish only}
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate
pcoa_Dissim_SO4_red_FINAL <- cmdscale(Dissim_SO4_redBC_finish, k=2, eig=TRUE, add=TRUE) # k = 2 because we only plot 2 dimensions, add=TRUE corrects highly negative eigenvalues


explainvar1_Dissim_SO4_red_FIN <- round(pcoa_Dissim_SO4_red_FINAL$eig[1] / sum(pcoa_Dissim_SO4_red_FINAL$eig), 3) * 100
explainvar2_Dissim_SO4_red_FIN <- round(pcoa_Dissim_SO4_red_FINAL$eig[2] / sum(pcoa_Dissim_SO4_red_FINAL$eig), 3) * 100

explainvar1_Dissim_SO4_red_FIN # 49.8
explainvar2_Dissim_SO4_red_FIN # 19.3

# Filter 'pre' samples out of *RA_design to get final samples only
Dissim_SO4_redRA_FIN_design <- Dissim_SO4_redRA_design %>% dplyr::filter(Treat != "_pre" & Plant != "_pre")

pcoa_Dissim_SO4_red_FIN.groups <- paste(Dissim_SO4_redRA_FIN_design$Treat, Dissim_SO4_redRA_FIN_design$Plant, Dissim_SO4_redRA_FIN_design$Hist, sep = "_")
pcoa_Dissim_SO4_red_FIN.points <- data.frame(pcoa_Dissim_SO4_red_FINAL$points, group = pcoa_Dissim_SO4_red_FIN.groups)

# Calculate Centroids (mean and SE)
pcoa_Dissim_SO4_red_FIN.L.centroids <- reshape2::melt(pcoa_Dissim_SO4_red_FIN.points, id="group", measure.vars = c("X1", "X2"))
pcoa_Dissim_SO4_red_FIN.centroids <- acast(pcoa_Dissim_SO4_red_FIN.L.centroids, variable ~ group, mean)
pcoa_Dissim_SO4_red_FIN.centroids.se <- acast(pcoa_Dissim_SO4_red_FIN.L.centroids, variable ~ group, se)
pcoa_Dissim_SO4_red_FIN.centroids.sd <- acast(pcoa_Dissim_SO4_red_FIN.L.centroids, variable ~ group, sd)

# Combine
pcoa_Dissim_SO4_red_FIN.cent.dataframe <- cbind(t(pcoa_Dissim_SO4_red_FIN.centroids), t(pcoa_Dissim_SO4_red_FIN.centroids.se))
colnames(pcoa_Dissim_SO4_red_FIN.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")

# Adding columns with the factors. (No need to run again if already in Global Environment)
#trt_FIN <- c("Dry","Dry", "Dry","Dry","Wet","Wet", "Wet","Wet") 
#plant_FIN <- c("No Plant","No Plant","Plant", "Plant", "No Plant","No Plant","Plant", "Plant")
#history_FIN <- c("Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet")
#history_plant_combined_FIN <- c(paste(history_FIN, plant_FIN, sep = "_"))

pcoa_Dissim_SO4_red_FIN.cent.dataframe.trts <- as.data.frame(pcoa_Dissim_SO4_red_FIN.cent.dataframe) 
pcoa_Dissim_SO4_red_FIN.cent.dataframe.trts$trt <- as.factor(trt_FIN)
pcoa_Dissim_SO4_red_FIN.cent.dataframe.trts$plant <- as.factor(plant_FIN)
pcoa_Dissim_SO4_red_FIN.cent.dataframe.trts$history <- as.factor(history_FIN)
pcoa_Dissim_SO4_red_FIN.cent.dataframe.trts$history_plant<- as.factor(history_plant_combined_FIN)
pcoa_Dissim_SO4_red_FIN.cent.dataframe.trts$centroid_or_sample <- "centroid"

# Add in points to data frame:
pcoa_Dissim_SO4_red_FIN.points_ForPlot <- pcoa_Dissim_SO4_red_FIN.points
colnames(pcoa_Dissim_SO4_red_FIN.points_ForPlot) <- c("V1", "V2", "group")
# Add labels to match ...cent.dataframe.trts
pcoa_Dissim_SO4_red_FIN.points_ForPlot$trt <- Dissim_SO4_redRA_FIN_design$Treat %>% str_replace_all(c("w" = "Wet", "_pre" = "Baseline", "d" = "Dry"))
pcoa_Dissim_SO4_red_FIN.points_ForPlot$plant <- Dissim_SO4_redRA_FIN_design$Plant %>% str_replace_all(c("n" = "No Plant", "y" = "Plant", "_pre" = "Baseline"))
pcoa_Dissim_SO4_red_FIN.points_ForPlot$history <- Dissim_SO4_redRA_FIN_design$Hist %>% str_replace_all(c("d" = "Dry", "w" = "Wet"))
pcoa_Dissim_SO4_red_FIN.points_ForPlot$history_plant <- c(paste(pcoa_Dissim_SO4_red_FIN.points_ForPlot$history, pcoa_Dissim_SO4_red_FIN.points_ForPlot$plant, sep = "_"))
#remove "group" column
pcoa_Dissim_SO4_red_FIN.points_ForPlot <- pcoa_Dissim_SO4_red_FIN.points_ForPlot[, -3]
#add V1e and V2e as zeros:
pcoa_Dissim_SO4_red_FIN.points_ForPlot$V1e <- numeric(16)
pcoa_Dissim_SO4_red_FIN.points_ForPlot$V2e <- numeric(16)
# Add column to distinguish sample points from centroids, and later use to set alpha in plot
pcoa_Dissim_SO4_red_FIN.points_ForPlot$centroid_or_sample <- "sample"

#Merge:
Dissim_SO4_red_FIN_centroids_points_merged <- rbind(pcoa_Dissim_SO4_red_FIN.cent.dataframe.trts, pcoa_Dissim_SO4_red_FIN.points_ForPlot)

# Add points data frame (seems to be newer issue, used to be able to use $ on pcoa_denit_FINAL$points)
pcoa_Dissim_SO4_red_FINAL_DF <- as.data.frame(pcoa_Dissim_SO4_red_FINAL$points)


#Run envfit with new Soil data (SFO = Soil Finish Only):
fit_Dissim_SO4_red_SFO <- envfit(pcoa_Dissim_SO4_red_FINAL_DF, envfit_FINISH_numerical, permutations = 10000, set.seed(42))
fit_Dissim_SO4_red_SFO

A_Dissim_SO4_red_SFO <-as.list(fit_Dissim_SO4_red_SFO$vectors)
vec_Dissim_SO4_red_SFO <- as.data.frame(fit_Dissim_SO4_red_SFO$vectors$arrows*sqrt(fit_Dissim_SO4_red_SFO$vectors$r)*0.15)
p_Dissim_SO4_red_SFO <- as.data.frame(A_Dissim_SO4_red_SFO$pvals)
vec_Dissim_SO4_red_SFO <- cbind(vec_Dissim_SO4_red_SFO, p_Dissim_SO4_red_SFO)
vec_Dissim_SO4_red_SFO <- subset(vec_Dissim_SO4_red_SFO, A_Dissim_SO4_red_SFO$pvals<0.05)

# Add column with labels, for plotting:
vec_Dissim_SO4_red_SFO$labels <- c(expression(paste("CH"[4])), "P", "Ca", "Cu")



#Plot using ggplot2
df5c <- as.data.frame(Dissim_SO4_red_FIN_centroids_points_merged)

plot5c <- ggplot(df5c, aes(x=V1, y=V2, shape=history_plant, colour=trt)) + geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.003), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.003), colour="black") +   
  geom_point(aes(shape = history_plant, colour=trt, alpha = centroid_or_sample, size = centroid_or_sample), stroke=2) +  
  theme_bw() +
  scale_size_manual(values = c(6, 3), name = "Centroid or Sample")+
  scale_alpha_manual(values = c(1, 0.3), name = "Centroid or Sample")+
  scale_color_manual(values=c("#997700", "#004488"), name="Treatment") +
  scale_shape_manual(values = c(0, 15, 1, 19), name="History_Plant") +
  xlab(glue("PCoA 1 ({explainvar1_Dissim_SO4_red_FIN}%)")) + ylab(glue("PCoA 2 ({explainvar2_Dissim_SO4_red_FIN}%)")) + 
  geom_segment(data=vec_Dissim_SO4_red_SFO, aes(x=0,xend=V1,y=0,yend=V2), linewidth=1, arrow = arrow(length = unit(0.2, "cm")),colour="black", alpha = 0.4,inherit.aes=F)+ 
 geom_text(data=vec_Dissim_SO4_red_SFO, aes(x=V1+0.001, y=V2+0.005), label=vec_Dissim_SO4_red_SFO$labels, size=3.5, inherit.aes=F)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  theme(panel.background = element_blank()) + 
  theme(axis.title = element_text(size=14), axis.text=element_text(size=14), 
          axis.text.x = element_text(size=14),  axis.text.y = element_text(size=14),
          panel.border = element_rect(colour = "black", linewidth=1.25)) +
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  theme(plot.title=element_text(size=14)) +
  theme(legend.text=element_text(size=14), legend.title = element_text(size=14))+
  guides(shape = guide_legend(override.aes = list(size = 4), order = 1), 
           colour = guide_legend(override.aes = list(pch=16, size = 4), order = 2)) +
    ggtitle("Dissimilatory Sulfate Reduction Week 8")

plot5c

#ggsave("../figures/Thesis/Dissim_sulf_red_wk8.png", plot = plot5c, path=NULL, device="png", scale=1, height = 7, width = 10, limitsize = TRUE, dpi = 300)
```

```{r PCoA: Thiosulfate oxidation by SOX complex}
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate
pcoa_ThioSOX <- cmdscale(ThioSOXBC, k=2, eig=TRUE, add=TRUE)

explainvar1b_ThioSOX <- round(pcoa_ThioSOX$eig[1] / sum(pcoa_ThioSOX$eig), 3) * 100
explainvar2b_ThioSOX <- round(pcoa_ThioSOX$eig[2] / sum(pcoa_ThioSOX$eig), 3) * 100

explainvar1b_ThioSOX # 49.3
explainvar2b_ThioSOX # 9.7

pcoa_ThioSOX2.groups <- paste(ThioSOXRA_design$Treat, ThioSOXRA_design$Plant, ThioSOXRA_design$Hist, sep = "_")
pcoa_ThioSOX2.points <- data.frame(pcoa_ThioSOX$points, group = pcoa_ThioSOX2.groups)

# Calculate Centroids (mean and SE)
pcoa_ThioSOX2.L.centroids <- reshape2::melt(pcoa_ThioSOX2.points, id="group", measure.vars = c("X1", "X2"))
pcoa_ThioSOX2.centroids <- acast(pcoa_ThioSOX2.L.centroids, variable ~ group, mean)
pcoa_ThioSOX2.centroids.se <- acast(pcoa_ThioSOX2.L.centroids, variable ~ group, se)
pcoa_ThioSOX2.centroids.sd <- acast(pcoa_ThioSOX2.L.centroids, variable ~ group, sd)

# Combine
pcoa_ThioSOX2.cent.dataframe <- cbind(t(pcoa_ThioSOX2.centroids), t(pcoa_ThioSOX2.centroids.se))
colnames(pcoa_ThioSOX2.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")

# No need to run again if these lists have already been declared:
#trt_H <- c("Baseline", "Baseline", "Dry","Dry", "Dry","Dry","Wet","Wet", "Wet","Wet") 
#plant_H <- c("Baseline", "Baseline", "No Plant","No Plant","Plant", "Plant", "No Plant","No Plant","Plant", "Plant")
#history <- c("Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet")
#trt_plant_combined <- c(paste(trt, plant, sep = "_"))

pcoa_ThioSOX2.cent.dataframe.trts <- as.data.frame(pcoa_ThioSOX2.cent.dataframe) 
pcoa_ThioSOX2.cent.dataframe.trts$trt <- as.factor(trt_H)
pcoa_ThioSOX2.cent.dataframe.trts$plant <- as.factor(plant_H)
pcoa_ThioSOX2.cent.dataframe.trts$history <- as.factor(history)
pcoa_ThioSOX2.cent.dataframe.trts$history_plant<- as.factor(history_plant_combined)
pcoa_ThioSOX2.cent.dataframe.trts$centroid_or_sample <- "centroid"

# Add in points to data frame:
pcoa_ThioSOX2.points_ForPlot <- pcoa_ThioSOX2.points
colnames(pcoa_ThioSOX2.points_ForPlot) <- c("V1", "V2", "group")
# Add labels to match ...cent.dataframe.trts
pcoa_ThioSOX2.points_ForPlot$trt <- ThioSOXRA_design$Treat %>% str_replace_all(c("w" = "Wet", "_pre" = "Baseline", "d" = "Dry"))
pcoa_ThioSOX2.points_ForPlot$plant <- ThioSOXRA_design$Plant %>% str_replace_all(c("n" = "No Plant", "y" = "Plant", "_pre" = "Baseline"))
pcoa_ThioSOX2.points_ForPlot$history <- ThioSOXRA_design$Hist %>% str_replace_all(c("d" = "Dry", "w" = "Wet"))
pcoa_ThioSOX2.points_ForPlot$history_plant <- c(paste(pcoa_ThioSOX2.points_ForPlot$history, pcoa_ThioSOX2.points_ForPlot$plant, sep = "_"))
#remove "group" column
pcoa_ThioSOX2.points_ForPlot <- pcoa_ThioSOX2.points_ForPlot[, -3]
#add V1e and V2e as zeros:
pcoa_ThioSOX2.points_ForPlot$V1e <- numeric(24)
pcoa_ThioSOX2.points_ForPlot$V2e <- numeric(24)
# Add column to distinguish sample points from centroids, and later use to set alpha in plot
pcoa_ThioSOX2.points_ForPlot$centroid_or_sample <- "sample"

#Merge:
ThioSOX_centroids_points_merged <- rbind(pcoa_ThioSOX2.cent.dataframe.trts, pcoa_ThioSOX2.points_ForPlot)

# Add points data frame (seems to be newer issue, used to be able to use $ on pcoa_denit_FINAL$points)
pcoa_ThioSOX_DF <- as.data.frame(pcoa_ThioSOX$points)
```
```{r ThioSox envfit and PCoA}
#Run envfit with new Soil data (SSF = Soil Start & Finish):
fit_ThioSOX_SSF <- envfit(pcoa_ThioSOX_DF, SoilGHG_StartFinish_Envfit, permutations = 10000, set.seed(42))
fit_ThioSOX_SSF

A_ThioSOX_SSF <-as.list(fit_ThioSOX_SSF$vectors)
vec_ThioSOX_SSF <- as.data.frame(fit_ThioSOX_SSF$vectors$arrows*sqrt(fit_ThioSOX_SSF$vectors$r)*0.15)
p_ThioSOX_SSF <- as.data.frame(A_ThioSOX_SSF$pvals)
vec_ThioSOX_SSF <- cbind(vec_ThioSOX_SSF, p_ThioSOX_SSF)
vec_ThioSOX_SSF <- subset(vec_ThioSOX_SSF, A_ThioSOX_SSF$pvals<0.05)

#Plot using ggplot2
df4d <- as.data.frame(ThioSOX_centroids_points_merged)

plot4d <- ggplot(df4d, aes(x=V1, y=V2, shape=history_plant, colour=trt)) + geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.003), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.003), colour="black") +   
  geom_point(aes(shape = history_plant, colour=trt, alpha = centroid_or_sample, size = centroid_or_sample), stroke=2) +  
  theme_bw() +
  scale_size_manual(values = c(6, 3), name = "Centroid or Sample")+
  scale_alpha_manual(values = c(1, 0.3), name = "Centroid or Sample")+
  scale_color_manual(values=c("black", "#997700", "#004488"), name="Treatment") +
  scale_shape_manual(values = c(7, 0, 15, 13, 1, 19), name="History_Plant") +
  xlab(glue("PCoA 1 ({explainvar1b_ThioSOX}%)")) + ylab(glue("PCoA 2 ({explainvar2b_ThioSOX}%)")) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  theme(panel.background = element_blank()) + 
  theme(axis.title = element_text(size=14), axis.text=element_text(size=14), 
          axis.text.x = element_text(size=14),  axis.text.y = element_text(size=14),
          panel.border = element_rect(colour = "black", linewidth=1.25)) +
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  theme(plot.title=element_text(size=14)) +
  theme(legend.text=element_text(size=14), legend.title = element_text(size=14))+
  guides(shape = guide_legend(override.aes = list(size = 4), order = 1), 
           colour = guide_legend(override.aes = list(pch=16, size = 4), order = 2)) +
    ggtitle("Thiosulfate Oxidation by SOX")

plot4d
```

```{r PCoA: ThioSox, finish only}
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate
pcoa_ThioSOX_FINAL <- cmdscale(ThioSOXBC_finish, k=2, eig=TRUE, add=TRUE) # k = 2 because we only plot 2 dimensions, add=TRUE corrects highly negative eigenvalues


explainvar1_ThioSOX_FIN <- round(pcoa_ThioSOX_FINAL$eig[1] / sum(pcoa_ThioSOX_FINAL$eig), 3) * 100
explainvar2_ThioSOX_FIN <- round(pcoa_ThioSOX_FINAL$eig[2] / sum(pcoa_ThioSOX_FINAL$eig), 3) * 100

explainvar1_ThioSOX_FIN # 56.2
explainvar2_ThioSOX_FIN # 10.9

# Filter 'pre' samples out of *RA_design to get final samples only
ThioSOXRA_FIN_design <- ThioSOXRA_design %>% dplyr::filter(Treat != "_pre" & Plant != "_pre")

pcoa_ThioSOX_FIN.groups <- paste(ThioSOXRA_FIN_design$Treat, ThioSOXRA_FIN_design$Plant, ThioSOXRA_FIN_design$Hist, sep = "_")
pcoa_ThioSOX_FIN.points <- data.frame(pcoa_ThioSOX_FINAL$points, group = pcoa_ThioSOX_FIN.groups)

# Calculate Centroids (mean and SE)
pcoa_ThioSOX_FIN.L.centroids <- reshape2::melt(pcoa_ThioSOX_FIN.points, id="group", measure.vars = c("X1", "X2"))
pcoa_ThioSOX_FIN.centroids <- acast(pcoa_ThioSOX_FIN.L.centroids, variable ~ group, mean)
pcoa_ThioSOX_FIN.centroids.se <- acast(pcoa_ThioSOX_FIN.L.centroids, variable ~ group, se)
pcoa_ThioSOX_FIN.centroids.sd <- acast(pcoa_ThioSOX_FIN.L.centroids, variable ~ group, sd)

# Combine
pcoa_ThioSOX_FIN.cent.dataframe <- cbind(t(pcoa_ThioSOX_FIN.centroids), t(pcoa_ThioSOX_FIN.centroids.se))
colnames(pcoa_ThioSOX_FIN.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")

# Adding columns with the factors. (No need to run again if already in Global Environment)
#trt_FIN <- c("Dry","Dry", "Dry","Dry","Wet","Wet", "Wet","Wet") 
#plant_FIN <- c("No Plant","No Plant","Plant", "Plant", "No Plant","No Plant","Plant", "Plant")
#history_FIN <- c("Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet")
#history_plant_combined_FIN <- c(paste(history_FIN, plant_FIN, sep = "_"))

pcoa_ThioSOX_FIN.cent.dataframe.trts <- as.data.frame(pcoa_ThioSOX_FIN.cent.dataframe) 
pcoa_ThioSOX_FIN.cent.dataframe.trts$trt <- as.factor(trt_FIN)
pcoa_ThioSOX_FIN.cent.dataframe.trts$plant <- as.factor(plant_FIN)
pcoa_ThioSOX_FIN.cent.dataframe.trts$history <- as.factor(history_FIN)
pcoa_ThioSOX_FIN.cent.dataframe.trts$history_plant<- as.factor(history_plant_combined_FIN)
pcoa_ThioSOX_FIN.cent.dataframe.trts$centroid_or_sample <- "centroid"

# Add in points to data frame:
pcoa_ThioSOX_FIN.points_ForPlot <- pcoa_ThioSOX_FIN.points
colnames(pcoa_ThioSOX_FIN.points_ForPlot) <- c("V1", "V2", "group")
# Add labels to match ...cent.dataframe.trts
pcoa_ThioSOX_FIN.points_ForPlot$trt <- ThioSOXRA_FIN_design$Treat %>% str_replace_all(c("w" = "Wet", "_pre" = "Baseline", "d" = "Dry"))
pcoa_ThioSOX_FIN.points_ForPlot$plant <- ThioSOXRA_FIN_design$Plant %>% str_replace_all(c("n" = "No Plant", "y" = "Plant", "_pre" = "Baseline"))
pcoa_ThioSOX_FIN.points_ForPlot$history <- ThioSOXRA_FIN_design$Hist %>% str_replace_all(c("d" = "Dry", "w" = "Wet"))
pcoa_ThioSOX_FIN.points_ForPlot$history_plant <- c(paste(pcoa_ThioSOX_FIN.points_ForPlot$history, pcoa_ThioSOX_FIN.points_ForPlot$plant, sep = "_"))
#remove "group" column
pcoa_ThioSOX_FIN.points_ForPlot <- pcoa_ThioSOX_FIN.points_ForPlot[, -3]
#add V1e and V2e as zeros:
pcoa_ThioSOX_FIN.points_ForPlot$V1e <- numeric(16)
pcoa_ThioSOX_FIN.points_ForPlot$V2e <- numeric(16)
# Add column to distinguish sample points from centroids, and later use to set alpha in plot
pcoa_ThioSOX_FIN.points_ForPlot$centroid_or_sample <- "sample"

#Merge:
ThioSOX_FIN_centroids_points_merged <- rbind(pcoa_ThioSOX_FIN.cent.dataframe.trts, pcoa_ThioSOX_FIN.points_ForPlot)

# Add points data frame (seems to be newer issue, used to be able to use $ on pcoa_denit_FINAL$points)
pcoa_ThioSOX_FINAL_DF <- as.data.frame(pcoa_ThioSOX_FINAL$points)

#Run envfit with new Soil data (SFO = Soil Finish Only):
fit_ThioSOX_SFO <- envfit(pcoa_ThioSOX_FINAL_DF, envfit_FINISH_numerical, permutations = 10000, set.seed(42))
fit_ThioSOX_SFO

A_ThioSOX_SFO <-as.list(fit_ThioSOX_SFO$vectors)
vec_ThioSOX_SFO <- as.data.frame(fit_ThioSOX_SFO$vectors$arrows*sqrt(fit_ThioSOX_SFO$vectors$r)*0.15)
p_ThioSOX_SFO <- as.data.frame(A_ThioSOX_SFO$pvals)
vec_ThioSOX_SFO <- cbind(vec_ThioSOX_SFO, p_ThioSOX_SFO)
vec_ThioSOX_SFO <- subset(vec_ThioSOX_SFO, A_ThioSOX_SFO$pvals<0.05)

# Add column with labels, for plotting:
vec_ThioSOX_SFO$labels <- c("Mg", "Ca", "pH", "S", "CEC")

# Add a column with nudges for nudge_x and nudge_y:
vec_ThioSOX_SFO$X_nudge <- c(0, 0, 0.007, 0, 0)
vec_ThioSOX_SFO$Y_nudge <- c(0, 0, 0.003, 0, 0)

#Plot using ggplot2
df5d <- as.data.frame(ThioSOX_FIN_centroids_points_merged)

plot5d <- ggplot(df5d, aes(x=V1, y=V2, shape=history_plant, colour=trt)) + geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.003), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.003), colour="black") +   
  geom_point(aes(shape = history_plant, colour=trt, alpha = centroid_or_sample, size = centroid_or_sample), stroke=2) +  
  theme_bw() +
  scale_size_manual(values = c(6, 3), name = "Centroid or Sample")+
  scale_alpha_manual(values = c(1, 0.3), name = "Centroid or Sample")+
  scale_color_manual(values=c("#997700", "#004488"), name="Treatment") +
  scale_shape_manual(values = c(0, 15, 1, 19), name="History_Plant") +
  xlab(glue("PCoA 1 ({explainvar1_ThioSOX_FIN}%)")) + ylab(glue("PCoA 2 ({explainvar2_ThioSOX_FIN}%)")) + 
  geom_segment(data=vec_ThioSOX_SFO, aes(x=0,xend=V1,y=0,yend=V2), linewidth=1, arrow = arrow(length = unit(0.2, "cm")),colour="black", alpha = 0.4,inherit.aes=F)+ 
 geom_text(data=vec_ThioSOX_SFO, aes(x=V1+0.003, y=V2+0.006), label=vec_ThioSOX_SFO$labels, nudge_x = vec_ThioSOX_SFO$X_nudge, nudge_y = vec_ThioSOX_SFO$Y_nudge, size=3.5, inherit.aes=F)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  theme(panel.background = element_blank()) + 
  theme(axis.title = element_text(size=14), axis.text=element_text(size=14), 
          axis.text.x = element_text(size=14),  axis.text.y = element_text(size=14),
          panel.border = element_rect(colour = "black", linewidth=1.25)) +
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  theme(plot.title=element_text(size=14)) +
  theme(legend.text=element_text(size=14), legend.title = element_text(size=14))+
  guides(shape = guide_legend(override.aes = list(size = 4), order = 1), 
           colour = guide_legend(override.aes = list(pch=16, size = 4), order = 2)) +
    ggtitle("Thiosulfate Oxidation by SOX Week 8")

plot5d
```

Create 4-panel figure of the 4 Sulfur PCoA plots above:
```{r 4-panel S PCoA plots}
# 4 panel Sulfur PCoA, start and finish samples (SSF)
S_pcoa_4panel_SSF <- ggarrange(plot4a, plot4b, plot4c, plot4d, labels = c("A", "B", "C", "D"), ncol = 2, nrow = 2, legend = "right", common.legend = T)

S_pcoa_4panel_SSF

#ggsave("../figures/Fe_S_manuscript/S_pcoa4panel_SSF.jpg", plot = S_pcoa_4panel_SSF, device=NULL, path=NULL, scale=1, height = 7, width = 12, limitsize = TRUE, dpi = 600)

# 4 panel Sulfur PCoA, Finish samples only (SFO)

S_pcoa_4panel_SFO <- ggarrange(plot5a, plot5b, plot5c, plot5d, labels = c("A", "B", "C", "D"), ncol = 2, nrow = 2, legend = "right", common.legend = T)

S_pcoa_4panel_SFO

#ggsave("../figures/Fe_S_manuscript/S_pcoa4panel_SFO.jpg", plot = S_pcoa_4panel_SFO, device=NULL, path=NULL, scale=1, height = 7, width = 12, limitsize = TRUE, dpi = 600)
```

```{r FeGenie total: SSF}
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate
pcoa_FeGenie <- cmdscale(FeGenieBC, k=2, eig=TRUE, add=TRUE)

explainvar1_FeGenie <- round(pcoa_FeGenie$eig[1] / sum(pcoa_FeGenie$eig), 3) * 100
explainvar2_FeGenie <- round(pcoa_FeGenie$eig[2] / sum(pcoa_FeGenie$eig), 3) * 100

explainvar1_FeGenie # 43.1
explainvar2_FeGenie # 10.4

pcoa_FeGenie.groups <- paste(FeGenieRA_design$Treat, FeGenieRA_design$Plant, FeGenieRA_design$Hist, sep = "_")
pcoa_FeGenie.points <- data.frame(pcoa_FeGenie$points, group = pcoa_FeGenie.groups)

# Calculate Centroids (mean and SE)
pcoa_FeGenie.L.centroids <- reshape2::melt(pcoa_FeGenie.points, id="group", measure.vars = c("X1", "X2"))
pcoa_FeGenie.centroids <- acast(pcoa_FeGenie.L.centroids, variable ~ group, mean)
pcoa_FeGenie.centroids.se <- acast(pcoa_FeGenie.L.centroids, variable ~ group, se)
pcoa_FeGenie.centroids.sd <- acast(pcoa_FeGenie.L.centroids, variable ~ group, sd)

# Combine
pcoa_FeGenie.cent.dataframe <- cbind(t(pcoa_FeGenie.centroids), t(pcoa_FeGenie.centroids.se))
colnames(pcoa_FeGenie.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")

# No need to run again if these lists have already been declared:
#trt_H <- c("Baseline", "Baseline", "Dry","Dry", "Dry","Dry","Wet","Wet", "Wet","Wet") 
#plant_H <- c("Baseline", "Baseline", "No Plant","No Plant","Plant", "Plant", "No Plant","No Plant","Plant", "Plant")
#history <- c("Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet")
#trt_plant_combined <- c(paste(trt, plant, sep = "_"))

pcoa_FeGenie.cent.dataframe.trts <- as.data.frame(pcoa_FeGenie.cent.dataframe) 
pcoa_FeGenie.cent.dataframe.trts$trt <- as.factor(trt_H)
pcoa_FeGenie.cent.dataframe.trts$plant <- as.factor(plant_H)
pcoa_FeGenie.cent.dataframe.trts$history <- as.factor(history)
pcoa_FeGenie.cent.dataframe.trts$history_plant<- as.factor(history_plant_combined)
pcoa_FeGenie.cent.dataframe.trts$centroid_or_sample <- "centroid"

# Add in points to data frame:
pcoa_FeGenie.points_ForPlot <- pcoa_FeGenie.points
colnames(pcoa_FeGenie.points_ForPlot) <- c("V1", "V2", "group")
# Add labels to match ...cent.dataframe.trts
pcoa_FeGenie.points_ForPlot$trt <- FeGenieRA_design$Treat %>% str_replace_all(c("w" = "Wet", "_pre" = "Baseline", "d" = "Dry"))
pcoa_FeGenie.points_ForPlot$plant <- FeGenieRA_design$Plant %>% str_replace_all(c("n" = "No Plant", "y" = "Plant", "_pre" = "Baseline"))
pcoa_FeGenie.points_ForPlot$history <- FeGenieRA_design$Hist %>% str_replace_all(c("d" = "Dry", "w" = "Wet"))
pcoa_FeGenie.points_ForPlot$history_plant <- c(paste(pcoa_FeGenie.points_ForPlot$history, pcoa_FeGenie.points_ForPlot$plant, sep = "_"))
#remove "group" column
pcoa_FeGenie.points_ForPlot <- pcoa_FeGenie.points_ForPlot[, -3]
#add V1e and V2e as zeros:
pcoa_FeGenie.points_ForPlot$V1e <- numeric(24)
pcoa_FeGenie.points_ForPlot$V2e <- numeric(24)
# Add column to distinguish sample points from centroids, and later use to set alpha in plot
pcoa_FeGenie.points_ForPlot$centroid_or_sample <- "sample"

#Merge:
FeGenie_centroids_points_merged <- rbind(pcoa_FeGenie.cent.dataframe.trts, pcoa_FeGenie.points_ForPlot)

# Add points data frame (seems to be newer issue, used to be able to use $ on pcoa_denit_FINAL$points)
pcoa_FeGenie_DF <- as.data.frame(pcoa_FeGenie$points)
```
```{r FeGenie SSF envit and plot}
#Run envfit with new Soil data (SSF = Soil Start & Finish):
fit_FeGenie_SSF <- envfit(pcoa_FeGenie_DF, SoilGHG_StartFinish_Envfit, permutations = 10000, set.seed(42))
fit_FeGenie_SSF

A_FeGenie_SSF <-as.list(fit_FeGenie_SSF$vectors)
vec_FeGenie_SSF <- as.data.frame(fit_FeGenie_SSF$vectors$arrows*sqrt(fit_FeGenie_SSF$vectors$r)*0.15)
p_FeGenie_SSF <- as.data.frame(A_FeGenie_SSF$pvals)
vec_FeGenie_SSF <- cbind(vec_FeGenie_SSF, p_FeGenie_SSF)
vec_FeGenie_SSF <- subset(vec_FeGenie_SSF, A_FeGenie_SSF$pvals<0.05)


#Plot using ggplot2
df6a <- as.data.frame(FeGenie_centroids_points_merged)

plot6a <- ggplot(df6a, aes(x=V1, y=V2, shape=history_plant, colour=trt)) + geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.003), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.003), colour="black") +   
  geom_point(aes(shape = history_plant, colour=trt, alpha = centroid_or_sample, size = centroid_or_sample), stroke=2) +  
  theme_bw() +
  scale_size_manual(values = c(6, 3), name = "Centroid or Sample")+
  scale_alpha_manual(values = c(1, 0.3), name = "Centroid or Sample")+
  scale_color_manual(values=c("black", "#997700", "#004488"), name="Treatment") +
  scale_shape_manual(values = c(7, 0, 15, 13, 1, 19), name="History_Plant") +
  xlab(glue("PCoA 1 ({explainvar1_FeGenie}%)")) + ylab(glue("PCoA 2 ({explainvar2_FeGenie}%)")) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  theme(panel.background = element_blank()) + 
  theme(axis.title = element_text(size=14), axis.text=element_text(size=14), 
          axis.text.x = element_text(size=14),  axis.text.y = element_text(size=14),
          panel.border = element_rect(colour = "black", linewidth=1.25)) +
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  theme(plot.title=element_text(size=14)) +
  theme(legend.text=element_text(size=14), legend.title = element_text(size=14), legend.position = "none")+
  guides(shape = guide_legend(override.aes = list(size = 4), order = 1), 
           colour = guide_legend(override.aes = list(pch=16, size = 4), order = 2)) +
    ggtitle("Fe Genes Week 0 (Baseline) and Week 8")

plot6a
```

```{r FeGenie total: SFO}
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate
pcoa_FeGenie_FINAL <- cmdscale(FeGenieBC_finish, k=2, eig=TRUE, add=TRUE) # k = 2 because we only plot 2 dimensions, add=TRUE corrects highly negative eigenvalues


explainvar1_FeGenie_FIN <- round(pcoa_FeGenie_FINAL$eig[1] / sum(pcoa_FeGenie_FINAL$eig), 3) * 100
explainvar2_FeGenie_FIN <- round(pcoa_FeGenie_FINAL$eig[2] / sum(pcoa_FeGenie_FINAL$eig), 3) * 100

explainvar1_FeGenie_FIN # 43.5
explainvar2_FeGenie_FIN # 13.5

# Filter 'pre' samples out of *RA_design to get final samples only
FeGenieRA_FIN_design <- FeGenieRA_design %>% dplyr::filter(Treat != "_pre" & Plant != "_pre")

pcoa_FeGenie_FIN.groups <- paste(FeGenieRA_FIN_design$Treat, FeGenieRA_FIN_design$Plant, FeGenieRA_FIN_design$Hist, sep = "_")
pcoa_FeGenie_FIN.points <- data.frame(pcoa_FeGenie_FINAL$points, group = pcoa_FeGenie_FIN.groups)

# Calculate Centroids (mean and SE)
pcoa_FeGenie_FIN.L.centroids <- reshape2::melt(pcoa_FeGenie_FIN.points, id="group", measure.vars = c("X1", "X2"))
pcoa_FeGenie_FIN.centroids <- acast(pcoa_FeGenie_FIN.L.centroids, variable ~ group, mean)
pcoa_FeGenie_FIN.centroids.se <- acast(pcoa_FeGenie_FIN.L.centroids, variable ~ group, se)
pcoa_FeGenie_FIN.centroids.sd <- acast(pcoa_FeGenie_FIN.L.centroids, variable ~ group, sd)

# Combine
pcoa_FeGenie_FIN.cent.dataframe <- cbind(t(pcoa_FeGenie_FIN.centroids), t(pcoa_FeGenie_FIN.centroids.se))
colnames(pcoa_FeGenie_FIN.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")

# Adding columns with the factors. (No need to run again if already in Global Environment)
#trt_FIN <- c("Dry","Dry", "Dry","Dry","Wet","Wet", "Wet","Wet") 
#plant_FIN <- c("No Plant","No Plant","Plant", "Plant", "No Plant","No Plant","Plant", "Plant")
#history_FIN <- c("Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet")
#history_plant_combined_FIN <- c(paste(history_FIN, plant_FIN, sep = "_"))

pcoa_FeGenie_FIN.cent.dataframe.trts <- as.data.frame(pcoa_FeGenie_FIN.cent.dataframe) 
pcoa_FeGenie_FIN.cent.dataframe.trts$trt <- as.factor(trt_FIN)
pcoa_FeGenie_FIN.cent.dataframe.trts$plant <- as.factor(plant_FIN)
pcoa_FeGenie_FIN.cent.dataframe.trts$history <- as.factor(history_FIN)
pcoa_FeGenie_FIN.cent.dataframe.trts$history_plant<- as.factor(history_plant_combined_FIN)
pcoa_FeGenie_FIN.cent.dataframe.trts$centroid_or_sample <- "centroid"

# Add in points to data frame:
pcoa_FeGenie_FIN.points_ForPlot <- pcoa_FeGenie_FIN.points
colnames(pcoa_FeGenie_FIN.points_ForPlot) <- c("V1", "V2", "group")
# Add labels to match ...cent.dataframe.trts
pcoa_FeGenie_FIN.points_ForPlot$trt <- FeGenieRA_FIN_design$Treat %>% str_replace_all(c("w" = "Wet", "_pre" = "Baseline", "d" = "Dry"))
pcoa_FeGenie_FIN.points_ForPlot$plant <- FeGenieRA_FIN_design$Plant %>% str_replace_all(c("n" = "No Plant", "y" = "Plant", "_pre" = "Baseline"))
pcoa_FeGenie_FIN.points_ForPlot$history <- FeGenieRA_FIN_design$Hist %>% str_replace_all(c("d" = "Dry", "w" = "Wet"))
pcoa_FeGenie_FIN.points_ForPlot$history_plant <- c(paste(pcoa_FeGenie_FIN.points_ForPlot$history, pcoa_FeGenie_FIN.points_ForPlot$plant, sep = "_"))
#remove "group" column
pcoa_FeGenie_FIN.points_ForPlot <- pcoa_FeGenie_FIN.points_ForPlot[, -3]
#add V1e and V2e as zeros:
pcoa_FeGenie_FIN.points_ForPlot$V1e <- numeric(16)
pcoa_FeGenie_FIN.points_ForPlot$V2e <- numeric(16)
# Add column to distinguish sample points from centroids, and later use to set alpha in plot
pcoa_FeGenie_FIN.points_ForPlot$centroid_or_sample <- "sample"

#Merge:
FeGenie_FIN_centroids_points_merged <- rbind(pcoa_FeGenie_FIN.cent.dataframe.trts, pcoa_FeGenie_FIN.points_ForPlot)

# Add points data frame (seems to be newer issue, used to be able to use $ on pcoa_denit_FINAL$points)
pcoa_FeGenie_FINAL_DF <- as.data.frame(pcoa_FeGenie_FINAL$points)


#Run envfit with new Soil data (SFO = Soil Finish Only):
fit_FeGenie_SFO <- envfit(pcoa_FeGenie_FINAL_DF, envfit_FINISH_numerical, permutations = 10000, set.seed(42))
fit_FeGenie_SFO

A_FeGenie_SFO <-as.list(fit_FeGenie_SFO$vectors)
vec_FeGenie_SFO <- as.data.frame(fit_FeGenie_SFO$vectors$arrows*sqrt(fit_FeGenie_SFO$vectors$r)*0.15)
p_FeGenie_SFO <- as.data.frame(A_FeGenie_SFO$pvals)
vec_FeGenie_SFO <- cbind(vec_FeGenie_SFO, p_FeGenie_SFO)
vec_FeGenie_SFO <- subset(vec_FeGenie_SFO, A_FeGenie_SFO$pvals<0.05)

# Add column with labels, for plotting:
vec_FeGenie_SFO$labels <- c("Ca", "pH")


#Plot using ggplot2
df6b <- as.data.frame(FeGenie_FIN_centroids_points_merged)

plot6b <- ggplot(df6b, aes(x=V1, y=V2, shape=history_plant, colour=trt)) + geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.003), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.003), colour="black") +   
  geom_point(aes(shape = history_plant, colour=trt, alpha = centroid_or_sample, size = centroid_or_sample), stroke=2) +  
  theme_bw() +
  scale_size_manual(values = c(6, 3), name = "Centroid or Sample")+
  scale_alpha_manual(values = c(1, 0.3), name = "Centroid or Sample")+
  scale_color_manual(values=c("#997700", "#004488"), name="Treatment") +
  scale_shape_manual(values = c(0, 15, 1, 19), name="History_Plant") +
  xlab(glue("PCoA 1 ({explainvar1_FeGenie_FIN}%)")) + ylab(glue("PCoA 2 ({explainvar2_FeGenie_FIN}%)")) + 
  geom_segment(data=vec_FeGenie_SFO, aes(x=0,xend=V1,y=0,yend=V2), linewidth=1, arrow = arrow(length = unit(0.2, "cm")),colour="black", alpha = 0.4,inherit.aes=F)+ 
 geom_text(data=vec_FeGenie_SFO, aes(x=V1, y=V2+0.005), label=vec_FeGenie_SFO$labels, size=4, inherit.aes=F)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  theme(panel.background = element_blank()) + 
  theme(axis.title = element_text(size=14), axis.text=element_text(size=14), 
          axis.text.x = element_text(size=14),  axis.text.y = element_text(size=14),
          panel.border = element_rect(colour = "black", linewidth=1.25)) +
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  theme(plot.title=element_text(size=14)) +
  theme(legend.text=element_text(size=14), legend.title = element_text(size=14), legend.position = "none")+
  guides(shape = guide_legend(override.aes = list(size = 4), order = 1), 
           colour = guide_legend(override.aes = list(pch=16, size = 4), order = 2)) +
    ggtitle("Fe Genes Week 8")

plot6b
```

```{r 2 panel FeGenie total start/finish and finish}
# extract legend that includes baseline:
leg <- get_legend(plot2a, position = "right") #Extract legend as a legend Grob

# panel and add legend grob
FeGenieTotal_2panel <- ggarrange(plot6a, plot6b, ncol = 1, nrow = 2, labels = c("A", "B"), legend.grob = leg, legend = "right")

FeGenieTotal_2panel

#ggsave("../figures/Fe_S_manuscript/FeGenieTotal_2panel.jpg", plot = FeGenieTotal_2panel, device=NULL, path=NULL, scale=1, height = 10, width = 7, limitsize = TRUE, dpi = 600)
```

```{r Fe reduction FeGenie: SSF}
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate
pcoa_FeReduction <- cmdscale(FeReductionBC, k=2, eig=TRUE, add=T)

explainvar1_FeReduction <- round(pcoa_FeReduction$eig[1] / sum(pcoa_FeReduction$eig), 3) * 100
explainvar2_FeReduction <- round(pcoa_FeReduction$eig[2] / sum(pcoa_FeReduction$eig), 3) * 100

explainvar1_FeReduction # 24.4
explainvar2_FeReduction # 16.7

pcoa_FeReduction.groups <- paste(FeReductionRA_design$Treat, FeReductionRA_design$Plant, FeReductionRA_design$Hist, sep = "_")
pcoa_FeReduction.points <- data.frame(pcoa_FeReduction$points, group = pcoa_FeReduction.groups)

# Calculate Centroids (mean and SE)
pcoa_FeReduction.L.centroids <- reshape2::melt(pcoa_FeReduction.points, id="group", measure.vars = c("X1", "X2"))
pcoa_FeReduction.centroids <- acast(pcoa_FeReduction.L.centroids, variable ~ group, mean)
pcoa_FeReduction.centroids.se <- acast(pcoa_FeReduction.L.centroids, variable ~ group, se)
pcoa_FeReduction.centroids.sd <- acast(pcoa_FeReduction.L.centroids, variable ~ group, sd)

# Combine
pcoa_FeReduction.cent.dataframe <- cbind(t(pcoa_FeReduction.centroids), t(pcoa_FeReduction.centroids.se))
colnames(pcoa_FeReduction.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")

# No need to run again if these lists have already been declared:
#trt_H <- c("Baseline", "Baseline", "Dry","Dry", "Dry","Dry","Wet","Wet", "Wet","Wet") 
#plant_H <- c("Baseline", "Baseline", "No Plant","No Plant","Plant", "Plant", "No Plant","No Plant","Plant", "Plant")
#history <- c("Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet")
#trt_plant_combined <- c(paste(trt, plant, sep = "_"))

pcoa_FeReduction.cent.dataframe.trts <- as.data.frame(pcoa_FeReduction.cent.dataframe) 
pcoa_FeReduction.cent.dataframe.trts$trt <- as.factor(trt_H)
pcoa_FeReduction.cent.dataframe.trts$plant <- as.factor(plant_H)
pcoa_FeReduction.cent.dataframe.trts$history <- as.factor(history)
pcoa_FeReduction.cent.dataframe.trts$history_plant<- as.factor(history_plant_combined)
pcoa_FeReduction.cent.dataframe.trts$centroid_or_sample <- "centroid"

# Add in points to data frame:
pcoa_FeReduction.points_ForPlot <- pcoa_FeReduction.points
colnames(pcoa_FeReduction.points_ForPlot) <- c("V1", "V2", "group")
# Add labels to match ...cent.dataframe.trts
pcoa_FeReduction.points_ForPlot$trt <- FeReductionRA_design$Treat %>% str_replace_all(c("w" = "Wet", "_pre" = "Baseline", "d" = "Dry"))
pcoa_FeReduction.points_ForPlot$plant <- FeReductionRA_design$Plant %>% str_replace_all(c("n" = "No Plant", "y" = "Plant", "_pre" = "Baseline"))
pcoa_FeReduction.points_ForPlot$history <- FeReductionRA_design$Hist %>% str_replace_all(c("d" = "Dry", "w" = "Wet"))
pcoa_FeReduction.points_ForPlot$history_plant <- c(paste(pcoa_FeReduction.points_ForPlot$history, pcoa_FeReduction.points_ForPlot$plant, sep = "_"))
#remove "group" column
pcoa_FeReduction.points_ForPlot <- pcoa_FeReduction.points_ForPlot[, -3]
#add V1e and V2e as zeros:
pcoa_FeReduction.points_ForPlot$V1e <- numeric(24)
pcoa_FeReduction.points_ForPlot$V2e <- numeric(24)
# Add column to distinguish sample points from centroids, and later use to set alpha in plot
pcoa_FeReduction.points_ForPlot$centroid_or_sample <- "sample"

#Merge:
FeReduction_centroids_points_merged <- rbind(pcoa_FeReduction.cent.dataframe.trts, pcoa_FeReduction.points_ForPlot)

# Add points data frame (seems to be newer issue, used to be able to use $ on pcoa_denit_FINAL$points)
pcoa_FeReduction_DF <- as.data.frame(pcoa_FeReduction$points)
```
```{r Fe reduction FeGenie SSF envit and plot}
#Run envfit with new Soil data (SSF = Soil Start & Finish):
fit_FeReduction_SSF <- envfit(pcoa_FeReduction_DF, SoilGHG_StartFinish_Envfit, permutations = 10000, set.seed(42))
fit_FeReduction_SSF

A_FeReduction_SSF <-as.list(fit_FeReduction_SSF$vectors)
vec_FeReduction_SSF <- as.data.frame(fit_FeReduction_SSF$vectors$arrows*sqrt(fit_FeReduction_SSF$vectors$r)*0.15)
p_FeReduction_SSF <- as.data.frame(A_FeReduction_SSF$pvals)
vec_FeReduction_SSF <- cbind(vec_FeReduction_SSF, p_FeReduction_SSF)
vec_FeReduction_SSF <- subset(vec_FeReduction_SSF, A_FeReduction_SSF$pvals<0.05)

# Add column with labels, for plotting:
#vec_FeReduction_SSF$labels <- c(expression(paste("mg CH"[4],"-C m"^{-2})), "P", "Ca", "Cu")

#Plot using ggplot2
df7a <- as.data.frame(FeReduction_centroids_points_merged)

plot7a <- ggplot(df7a, aes(x=V1, y=V2, shape=history_plant, colour=trt)) + geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.003), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.003), colour="black") +   
  geom_point(aes(shape = history_plant, colour=trt, alpha = centroid_or_sample, size = centroid_or_sample), stroke=2) +  
  theme_bw() +
  scale_size_manual(values = c(6, 3), name = "Centroid or Sample")+
  scale_alpha_manual(values = c(1, 0.3), name = "Centroid or Sample")+
  scale_color_manual(values=c("black", "#997700", "#004488"), name="Treatment") +
  scale_shape_manual(values = c(7, 0, 15, 13, 1, 19), name="History_Plant") +
  xlab(glue("PCoA 1 ({explainvar1_FeReduction}%)")) + ylab(glue("PCoA 2 ({explainvar2_FeReduction}%)")) + 
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  theme(panel.background = element_blank()) + 
  theme(axis.title = element_text(size=14), axis.text=element_text(size=14), 
          axis.text.x = element_text(size=14),  axis.text.y = element_text(size=14),
          panel.border = element_rect(colour = "black", linewidth=1.25)) +
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  theme(plot.title=element_text(size=14)) +
  theme(legend.text=element_text(size=14), legend.title = element_text(size=14), legend.position = "none")+
  guides(shape = guide_legend(override.aes = list(size = 4), order = 1), 
           colour = guide_legend(override.aes = list(pch=16, size = 4), order = 2)) +
    ggtitle("Fe Reduction Week 0 (Baseline) and Week 8")

plot7a
```

```{r Fe reduction FeGenie: SFO}
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate
pcoa_FeReduction_FINAL <- cmdscale(FeReductionBC_finish, k=2, eig=TRUE, add=TRUE) # k = 2 because we only plot 2 dimensions, add=TRUE corrects highly negative eigenvalues


explainvar1_FeReduction_FIN <- round(pcoa_FeReduction_FINAL$eig[1] / sum(pcoa_FeReduction_FINAL$eig), 3) * 100
explainvar2_FeReduction_FIN <- round(pcoa_FeReduction_FINAL$eig[2] / sum(pcoa_FeReduction_FINAL$eig), 3) * 100

explainvar1_FeReduction_FIN # 25.1
explainvar2_FeReduction_FIN # 21.8

# Filter 'pre' samples out of *RA_design to get final samples only
FeReductionRA_FIN_design <- FeReductionRA_design %>% dplyr::filter(Treat != "_pre" & Plant != "_pre")

pcoa_FeReduction_FIN.groups <- paste(FeReductionRA_FIN_design$Treat, FeReductionRA_FIN_design$Plant, FeReductionRA_FIN_design$Hist, sep = "_")
pcoa_FeReduction_FIN.points <- data.frame(pcoa_FeReduction_FINAL$points, group = pcoa_FeReduction_FIN.groups)

# Calculate Centroids (mean and SE)
pcoa_FeReduction_FIN.L.centroids <- reshape2::melt(pcoa_FeReduction_FIN.points, id="group", measure.vars = c("X1", "X2"))
pcoa_FeReduction_FIN.centroids <- acast(pcoa_FeReduction_FIN.L.centroids, variable ~ group, mean)
pcoa_FeReduction_FIN.centroids.se <- acast(pcoa_FeReduction_FIN.L.centroids, variable ~ group, se)
pcoa_FeReduction_FIN.centroids.sd <- acast(pcoa_FeReduction_FIN.L.centroids, variable ~ group, sd)

# Combine
pcoa_FeReduction_FIN.cent.dataframe <- cbind(t(pcoa_FeReduction_FIN.centroids), t(pcoa_FeReduction_FIN.centroids.se))
colnames(pcoa_FeReduction_FIN.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")

# Adding columns with the factors. (No need to run again if already in Global Environment)
#trt_FIN <- c("Dry","Dry", "Dry","Dry","Wet","Wet", "Wet","Wet") 
#plant_FIN <- c("No Plant","No Plant","Plant", "Plant", "No Plant","No Plant","Plant", "Plant")
#history_FIN <- c("Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet")
#history_plant_combined_FIN <- c(paste(history_FIN, plant_FIN, sep = "_"))

pcoa_FeReduction_FIN.cent.dataframe.trts <- as.data.frame(pcoa_FeReduction_FIN.cent.dataframe) 
pcoa_FeReduction_FIN.cent.dataframe.trts$trt <- as.factor(trt_FIN)
pcoa_FeReduction_FIN.cent.dataframe.trts$plant <- as.factor(plant_FIN)
pcoa_FeReduction_FIN.cent.dataframe.trts$history <- as.factor(history_FIN)
pcoa_FeReduction_FIN.cent.dataframe.trts$history_plant<- as.factor(history_plant_combined_FIN)
pcoa_FeReduction_FIN.cent.dataframe.trts$centroid_or_sample <- "centroid"

# Add in points to data frame:
pcoa_FeReduction_FIN.points_ForPlot <- pcoa_FeReduction_FIN.points
colnames(pcoa_FeReduction_FIN.points_ForPlot) <- c("V1", "V2", "group")
# Add labels to match ...cent.dataframe.trts
pcoa_FeReduction_FIN.points_ForPlot$trt <- FeReductionRA_FIN_design$Treat %>% str_replace_all(c("w" = "Wet", "_pre" = "Baseline", "d" = "Dry"))
pcoa_FeReduction_FIN.points_ForPlot$plant <- FeReductionRA_FIN_design$Plant %>% str_replace_all(c("n" = "No Plant", "y" = "Plant", "_pre" = "Baseline"))
pcoa_FeReduction_FIN.points_ForPlot$history <- FeReductionRA_FIN_design$Hist %>% str_replace_all(c("d" = "Dry", "w" = "Wet"))
pcoa_FeReduction_FIN.points_ForPlot$history_plant <- c(paste(pcoa_FeReduction_FIN.points_ForPlot$history, pcoa_FeReduction_FIN.points_ForPlot$plant, sep = "_"))
#remove "group" column
pcoa_FeReduction_FIN.points_ForPlot <- pcoa_FeReduction_FIN.points_ForPlot[, -3]
#add V1e and V2e as zeros:
pcoa_FeReduction_FIN.points_ForPlot$V1e <- numeric(16)
pcoa_FeReduction_FIN.points_ForPlot$V2e <- numeric(16)
# Add column to distinguish sample points from centroids, and later use to set alpha in plot
pcoa_FeReduction_FIN.points_ForPlot$centroid_or_sample <- "sample"

#Merge:
FeReduction_FIN_centroids_points_merged <- rbind(pcoa_FeReduction_FIN.cent.dataframe.trts, pcoa_FeReduction_FIN.points_ForPlot)

# Add points data frame (seems to be newer issue, used to be able to use $ on pcoa_denit_FINAL$points)
pcoa_FeReduction_FINAL_DF <- as.data.frame(pcoa_FeReduction_FINAL$points)

#Run envfit with new Soil data (SFO = Soil Finish Only):
fit_FeReduction_SFO <- envfit(pcoa_FeReduction_FINAL_DF, envfit_FINISH_numerical, permutations = 10000, set.seed(42))
fit_FeReduction_SFO

A_FeReduction_SFO <-as.list(fit_FeReduction_SFO$vectors)
vec_FeReduction_SFO <- as.data.frame(fit_FeReduction_SFO$vectors$arrows*sqrt(fit_FeReduction_SFO$vectors$r)*0.4) # adjusting to dodge points
p_FeReduction_SFO <- as.data.frame(A_FeReduction_SFO$pvals)
vec_FeReduction_SFO <- cbind(vec_FeReduction_SFO, p_FeReduction_SFO)
vec_FeReduction_SFO <- subset(vec_FeReduction_SFO, A_FeReduction_SFO$pvals<0.05)

# Add column with labels, for plotting:
vec_FeReduction_SFO$labels <- c(expression(paste("CH"[4])), "%C, %N", "%N, %C", "P", "Ca", "Cu", "CEC")
  # %C and %N will overlap below, and one label needs to be removed using check_overlap = T. So, the label for both has been combined into one

#Plot using ggplot2
df7b <- as.data.frame(FeReduction_FIN_centroids_points_merged)

plot7b <- ggplot(df7b, aes(x=V1, y=V2, shape=history_plant, colour=trt)) + geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.003), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.003), colour="black") +   
  geom_point(aes(shape = history_plant, colour=trt, alpha = centroid_or_sample, size = centroid_or_sample), stroke=2) +  
  theme_bw() +
  scale_size_manual(values = c(6, 3), name = "Centroid or Sample")+
  scale_alpha_manual(values = c(1, 0.3), name = "Centroid or Sample")+
  scale_color_manual(values=c("#997700", "#004488"), name="Treatment") +
  scale_shape_manual(values = c(0, 15, 1, 19), name="History_Plant") +
  xlab(glue("PCoA 1 ({explainvar1_FeReduction_FIN}%)")) + ylab(glue("PCoA 2 ({explainvar2_FeReduction_FIN}%)")) + 
  geom_segment(data=vec_FeReduction_SFO, aes(x=0,xend=V1,y=0,yend=V2), linewidth=1, arrow = arrow(length = unit(0.2, "cm")),colour="black", alpha = 0.4,inherit.aes=F)+ 
 geom_text(data=vec_FeReduction_SFO, aes(x=V1, y=V2+0.02), label=vec_FeReduction_SFO$labels, size=4, check_overlap = T,inherit.aes=F)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  theme(panel.background = element_blank()) + 
  theme(axis.title = element_text(size=14), axis.text=element_text(size=14), 
          axis.text.x = element_text(size=14),  axis.text.y = element_text(size=14),
          panel.border = element_rect(colour = "black", linewidth=1.25)) +
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  theme(plot.title=element_text(size=14)) +
  theme(legend.text=element_text(size=14), legend.title = element_text(size=14), legend.position = "none")+
  guides(shape = guide_legend(override.aes = list(size = 4), order = 1), 
           colour = guide_legend(override.aes = list(pch=16, size = 4), order = 2)) +
    ggtitle("Fe Reduction Week 8")

plot7b

#ggsave("../figures/Thesis/Fe_red_wk8.png", plot = plot7b, path=NULL, device="png", scale=1, height = 7, width = 10, limitsize = TRUE, dpi = 300)
```

```{r SO4 reduction combined: SSF}
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate
pcoa_SO4_red_combined <- cmdscale(SO4_red_combinedBC, k=2, eig=TRUE, add=TRUE)

explainvar1_SO4_red_combined <- round(pcoa_SO4_red_combined$eig[1] / sum(pcoa_SO4_red_combined$eig), 3) * 100
explainvar2_SO4_red_combined <- format(round(pcoa_SO4_red_combined$eig[2] / sum(pcoa_SO4_red_combined$eig), 3) * 100, nsmall = 1) # Using format() to get .0

explainvar1_SO4_red_combined # 31.7
explainvar2_SO4_red_combined # 17.0


pcoa_SO4_red_combined.groups <- paste(SO4_red_combinedRA_design$Treat, SO4_red_combinedRA_design$Plant, SO4_red_combinedRA_design$Hist, sep = "_")
pcoa_SO4_red_combined.points <- data.frame(pcoa_SO4_red_combined$points, group = pcoa_SO4_red_combined.groups)

# Calculate Centroids (mean and SE)
pcoa_SO4_red_combined.L.centroids <- reshape2::melt(pcoa_SO4_red_combined.points, id="group", measure.vars = c("X1", "X2"))
pcoa_SO4_red_combined.centroids <- acast(pcoa_SO4_red_combined.L.centroids, variable ~ group, mean)
pcoa_SO4_red_combined.centroids.se <- acast(pcoa_SO4_red_combined.L.centroids, variable ~ group, se)
pcoa_SO4_red_combined.centroids.sd <- acast(pcoa_SO4_red_combined.L.centroids, variable ~ group, sd)

# Combine
pcoa_SO4_red_combined.cent.dataframe <- cbind(t(pcoa_SO4_red_combined.centroids), t(pcoa_SO4_red_combined.centroids.se))
colnames(pcoa_SO4_red_combined.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")

# No need to run again if these lists have already been declared:
#trt_H <- c("Baseline", "Baseline", "Dry","Dry", "Dry","Dry","Wet","Wet", "Wet","Wet") 
#plant_H <- c("Baseline", "Baseline", "No Plant","No Plant","Plant", "Plant", "No Plant","No Plant","Plant", "Plant")
#history <- c("Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet")
#trt_plant_combined <- c(paste(trt, plant, sep = "_"))

pcoa_SO4_red_combined.cent.dataframe.trts <- as.data.frame(pcoa_SO4_red_combined.cent.dataframe) 
pcoa_SO4_red_combined.cent.dataframe.trts$trt <- as.factor(trt_H)
pcoa_SO4_red_combined.cent.dataframe.trts$plant <- as.factor(plant_H)
pcoa_SO4_red_combined.cent.dataframe.trts$history <- as.factor(history)
pcoa_SO4_red_combined.cent.dataframe.trts$history_plant<- as.factor(history_plant_combined)
pcoa_SO4_red_combined.cent.dataframe.trts$centroid_or_sample <- "centroid"

# Add in points to data frame:
pcoa_SO4_red_combined.points_ForPlot <- pcoa_SO4_red_combined.points
colnames(pcoa_SO4_red_combined.points_ForPlot) <- c("V1", "V2", "group")
# Add labels to match ...cent.dataframe.trts
pcoa_SO4_red_combined.points_ForPlot$trt <- SO4_red_combinedRA_design$Treat %>% str_replace_all(c("w" = "Wet", "_pre" = "Baseline", "d" = "Dry"))
pcoa_SO4_red_combined.points_ForPlot$plant <- SO4_red_combinedRA_design$Plant %>% str_replace_all(c("n" = "No Plant", "y" = "Plant", "_pre" = "Baseline"))
pcoa_SO4_red_combined.points_ForPlot$history <- SO4_red_combinedRA_design$Hist %>% str_replace_all(c("d" = "Dry", "w" = "Wet"))
pcoa_SO4_red_combined.points_ForPlot$history_plant <- c(paste(pcoa_SO4_red_combined.points_ForPlot$history, pcoa_SO4_red_combined.points_ForPlot$plant, sep = "_"))
#remove "group" column
pcoa_SO4_red_combined.points_ForPlot <- pcoa_SO4_red_combined.points_ForPlot[, -3]
#add V1e and V2e as zeros:
pcoa_SO4_red_combined.points_ForPlot$V1e <- numeric(24)
pcoa_SO4_red_combined.points_ForPlot$V2e <- numeric(24)
# Add column to distinguish sample points from centroids, and later use to set alpha in plot
pcoa_SO4_red_combined.points_ForPlot$centroid_or_sample <- "sample"

#Merge:
SO4_red_combined_centroids_points_merged <- rbind(pcoa_SO4_red_combined.cent.dataframe.trts, pcoa_SO4_red_combined.points_ForPlot)

# Add points data frame (seems to be newer issue, used to be able to use $ on pcoa_denit_FINAL$points)
pcoa_SO4_red_combined_DF <- as.data.frame(pcoa_SO4_red_combined$points)
```
```{r SO4 reduction combined SSF envit and plot}
#Run envfit with new Soil data (SSF = Soil Start & Finish):
fit_SO4_red_combined_SSF <- envfit(pcoa_SO4_red_combined_DF, SoilGHG_StartFinish_Envfit, permutations = 10000, set.seed(42))
fit_SO4_red_combined_SSF

A_SO4_red_combined_SSF <-as.list(fit_SO4_red_combined_SSF$vectors)
vec_SO4_red_combined_SSF <- as.data.frame(fit_SO4_red_combined_SSF$vectors$arrows*sqrt(fit_SO4_red_combined_SSF$vectors$r)*0.15)
p_SO4_red_combined_SSF <- as.data.frame(A_SO4_red_combined_SSF$pvals)
vec_SO4_red_combined_SSF <- cbind(vec_SO4_red_combined_SSF, p_SO4_red_combined_SSF)
vec_SO4_red_combined_SSF <- subset(vec_SO4_red_combined_SSF, A_SO4_red_combined_SSF$pvals<0.05)

# Add column with labels, for plotting:
vec_SO4_red_combined_SSF$labels <- "%moisture"

#Plot using ggplot2
df7c <- as.data.frame(SO4_red_combined_centroids_points_merged)

plot7c <- ggplot(df7c, aes(x=V1, y=V2, shape=history_plant, colour=trt)) + geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.003), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.003), colour="black") +   
  geom_point(aes(shape = history_plant, colour=trt, alpha = centroid_or_sample, size = centroid_or_sample), stroke=2) +  
  theme_bw() +
  scale_size_manual(values = c(6, 3), name = "Centroid or Sample")+
  scale_alpha_manual(values = c(1, 0.3), name = "Centroid or Sample")+
  scale_color_manual(values=c("black", "#997700", "#004488"), name="Treatment") +
  scale_shape_manual(values = c(7, 0, 15, 13, 1, 19), name="History_Plant") +
  xlab(glue("PCoA 1 ({explainvar1_SO4_red_combined}%)")) + ylab(glue("PCoA 2 ({explainvar2_SO4_red_combined}%)")) + 
  geom_segment(data=vec_SO4_red_combined_SSF, aes(x=0,xend=V1,y=0,yend=V2), linewidth=1, arrow = arrow(length = unit(0.2, "cm")),colour="black", alpha = 0.4,inherit.aes=F)+ 
 geom_text(data=vec_SO4_red_combined_SSF, aes(x=V1, y=V2-0.004), label=vec_SO4_red_combined_SSF$labels, size=4, inherit.aes=F)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  theme(panel.background = element_blank()) + 
  theme(axis.title = element_text(size=14), axis.text=element_text(size=14), 
          axis.text.x = element_text(size=14),  axis.text.y = element_text(size=14),
          panel.border = element_rect(colour = "black", linewidth=1.25)) +
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  theme(plot.title=element_text(size=14)) +
  theme(legend.text=element_text(size=14), legend.title = element_text(size=14), legend.position = "none")+
  guides(shape = guide_legend(override.aes = list(size = 4), order = 1), 
           colour = guide_legend(override.aes = list(pch=16, size = 4), order = 2)) +
    ggtitle("Sulfate Reduction Week 0 (Baseline) and Week 8")

plot7c
```

```{r SO4 reduction combined total: SFO}
# Classical (Metric) Multidimensional Scaling; returns PCoA coordinates
# eig=TRUE returns eigenvalues; k = # of dimensions to calculate
pcoa_SO4_red_combined_FINAL <- cmdscale(SO4_red_combinedBC_finish, k=2, eig=TRUE, add=TRUE) # k = 2 because we only plot 2 dimensions, add=TRUE corrects highly negative eigenvalues


explainvar1_SO4_red_combined_FIN <- round(pcoa_SO4_red_combined_FINAL$eig[1] / sum(pcoa_SO4_red_combined_FINAL$eig), 3) * 100
explainvar2_SO4_red_combined_FIN <- round(pcoa_SO4_red_combined_FINAL$eig[2] / sum(pcoa_SO4_red_combined_FINAL$eig), 3) * 100

explainvar1_SO4_red_combined_FIN # 43.5
explainvar2_SO4_red_combined_FIN # 13.5

# Filter 'pre' samples out of *RA_design to get final samples only
SO4_red_combinedRA_FIN_design <- SO4_red_combinedRA_design %>% dplyr::filter(Treat != "_pre" & Plant != "_pre")

pcoa_SO4_red_combined_FIN.groups <- paste(SO4_red_combinedRA_FIN_design$Treat, SO4_red_combinedRA_FIN_design$Plant, SO4_red_combinedRA_FIN_design$Hist, sep = "_")
pcoa_SO4_red_combined_FIN.points <- data.frame(pcoa_SO4_red_combined_FINAL$points, group = pcoa_SO4_red_combined_FIN.groups)

# Calculate Centroids (mean and SE)
pcoa_SO4_red_combined_FIN.L.centroids <- reshape2::melt(pcoa_SO4_red_combined_FIN.points, id="group", measure.vars = c("X1", "X2"))
pcoa_SO4_red_combined_FIN.centroids <- acast(pcoa_SO4_red_combined_FIN.L.centroids, variable ~ group, mean)
pcoa_SO4_red_combined_FIN.centroids.se <- acast(pcoa_SO4_red_combined_FIN.L.centroids, variable ~ group, se)
pcoa_SO4_red_combined_FIN.centroids.sd <- acast(pcoa_SO4_red_combined_FIN.L.centroids, variable ~ group, sd)

# Combine
pcoa_SO4_red_combined_FIN.cent.dataframe <- cbind(t(pcoa_SO4_red_combined_FIN.centroids), t(pcoa_SO4_red_combined_FIN.centroids.se))
colnames(pcoa_SO4_red_combined_FIN.cent.dataframe) <- c("V1", "V2", "V1e", "V2e")

# Adding columns with the factors. (No need to run again if already in Global Environment)
#trt_FIN <- c("Dry","Dry", "Dry","Dry","Wet","Wet", "Wet","Wet") 
#plant_FIN <- c("No Plant","No Plant","Plant", "Plant", "No Plant","No Plant","Plant", "Plant")
#history_FIN <- c("Dry", "Wet", "Dry", "Wet", "Dry", "Wet", "Dry", "Wet")
#history_plant_combined_FIN <- c(paste(history_FIN, plant_FIN, sep = "_"))

pcoa_SO4_red_combined_FIN.cent.dataframe.trts <- as.data.frame(pcoa_SO4_red_combined_FIN.cent.dataframe) 
pcoa_SO4_red_combined_FIN.cent.dataframe.trts$trt <- as.factor(trt_FIN)
pcoa_SO4_red_combined_FIN.cent.dataframe.trts$plant <- as.factor(plant_FIN)
pcoa_SO4_red_combined_FIN.cent.dataframe.trts$history <- as.factor(history_FIN)
pcoa_SO4_red_combined_FIN.cent.dataframe.trts$history_plant<- as.factor(history_plant_combined_FIN)
pcoa_SO4_red_combined_FIN.cent.dataframe.trts$centroid_or_sample <- "centroid"

# Add in points to data frame:
pcoa_SO4_red_combined_FIN.points_ForPlot <- pcoa_SO4_red_combined_FIN.points
colnames(pcoa_SO4_red_combined_FIN.points_ForPlot) <- c("V1", "V2", "group")
# Add labels to match ...cent.dataframe.trts
pcoa_SO4_red_combined_FIN.points_ForPlot$trt <- SO4_red_combinedRA_FIN_design$Treat %>% str_replace_all(c("w" = "Wet", "_pre" = "Baseline", "d" = "Dry"))
pcoa_SO4_red_combined_FIN.points_ForPlot$plant <- SO4_red_combinedRA_FIN_design$Plant %>% str_replace_all(c("n" = "No Plant", "y" = "Plant", "_pre" = "Baseline"))
pcoa_SO4_red_combined_FIN.points_ForPlot$history <- SO4_red_combinedRA_FIN_design$Hist %>% str_replace_all(c("d" = "Dry", "w" = "Wet"))
pcoa_SO4_red_combined_FIN.points_ForPlot$history_plant <- c(paste(pcoa_SO4_red_combined_FIN.points_ForPlot$history, pcoa_SO4_red_combined_FIN.points_ForPlot$plant, sep = "_"))
#remove "group" column
pcoa_SO4_red_combined_FIN.points_ForPlot <- pcoa_SO4_red_combined_FIN.points_ForPlot[, -3]
#add V1e and V2e as zeros:
pcoa_SO4_red_combined_FIN.points_ForPlot$V1e <- numeric(16)
pcoa_SO4_red_combined_FIN.points_ForPlot$V2e <- numeric(16)
# Add column to distinguish sample points from centroids, and later use to set alpha in plot
pcoa_SO4_red_combined_FIN.points_ForPlot$centroid_or_sample <- "sample"

#Merge:
SO4_red_combined_FIN_centroids_points_merged <- rbind(pcoa_SO4_red_combined_FIN.cent.dataframe.trts, pcoa_SO4_red_combined_FIN.points_ForPlot)

# Add points data frame (seems to be newer issue, used to be able to use $ on pcoa_denit_FINAL$points)
pcoa_SO4_red_combined_FINAL_DF <- as.data.frame(pcoa_SO4_red_combined_FINAL$points)


#Run envfit with new Soil data (SFO = Soil Finish Only):
fit_SO4_red_combined_SFO <- envfit(pcoa_SO4_red_combined_FINAL_DF, envfit_FINISH_numerical, permutations = 10000, set.seed(42))
fit_SO4_red_combined_SFO

A_SO4_red_combined_SFO <-as.list(fit_SO4_red_combined_SFO$vectors)
vec_SO4_red_combined_SFO <- as.data.frame(fit_SO4_red_combined_SFO$vectors$arrows*sqrt(fit_SO4_red_combined_SFO$vectors$r)*0.15)
p_SO4_red_combined_SFO <- as.data.frame(A_SO4_red_combined_SFO$pvals)
vec_SO4_red_combined_SFO <- cbind(vec_SO4_red_combined_SFO, p_SO4_red_combined_SFO)
vec_SO4_red_combined_SFO <- subset(vec_SO4_red_combined_SFO, A_SO4_red_combined_SFO$pvals<0.05)

# Add column with labels, for plotting:
vec_SO4_red_combined_SFO$labels <- c("%C", "Mg", "Ca", "S", "Zn", "Mn", "Cu", "CEC")


#Plot using ggplot2
df7d <- as.data.frame(SO4_red_combined_FIN_centroids_points_merged)

plot7d <- ggplot(df7d, aes(x=V1, y=V2, shape=history_plant, colour=trt)) + geom_errorbarh(aes(xmax=V1+V1e, xmin=V1-V1e, height=0.003), colour="black") +    
  geom_errorbar(aes(ymax=V2+V2e, ymin=V2-V2e, width=0.003), colour="black") +   
  geom_point(aes(shape = history_plant, colour=trt, alpha = centroid_or_sample, size = centroid_or_sample), stroke=2) +  
  theme_bw() +
  scale_size_manual(values = c(6, 3), name = "Centroid or Sample")+
  scale_alpha_manual(values = c(1, 0.3), name = "Centroid or Sample")+
  scale_color_manual(values=c("#997700", "#004488"), name="Treatment") +
  scale_shape_manual(values = c(0, 15, 1, 19), name="History_Plant") +
  xlab(glue("PCoA 1 ({explainvar1_SO4_red_combined_FIN}%)")) + ylab(glue("PCoA 2 ({explainvar2_SO4_red_combined_FIN}%)")) + 
  geom_segment(data=vec_SO4_red_combined_SFO, aes(x=0,xend=V1,y=0,yend=V2), linewidth=1, arrow = arrow(length = unit(0.2, "cm")),colour="black", alpha = 0.4,inherit.aes=F)+ 
 geom_text(data=vec_SO4_red_combined_SFO, aes(x=V1, y=V2+0.005), label=vec_SO4_red_combined_SFO$labels, size=4, inherit.aes=F)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + 
  theme(panel.background = element_blank()) + 
  theme(axis.title = element_text(size=14), axis.text=element_text(size=14), 
          axis.text.x = element_text(size=14),  axis.text.y = element_text(size=14),
          panel.border = element_rect(colour = "black", linewidth=1.25)) +
  theme(axis.ticks.length=unit(0.3,"cm")) + 
  theme(plot.title=element_text(size=14)) +
  theme(legend.text=element_text(size=14), legend.title = element_text(size=14), legend.position = "none")+
  guides(shape = guide_legend(override.aes = list(size = 4), order = 1), 
           colour = guide_legend(override.aes = list(pch=16, size = 4), order = 2)) +
    ggtitle("Sulfate Reduction Week 8")

plot7d
```

```{r four panel Fe and SO4 reduction}
# use same 'leg' for legend grob

# panel and add legend grob
FeRed_SO4Red_4panel <- ggarrange(plot7a, plot7b, plot7c, plot7d, ncol = 2, nrow = 2, labels = c("A", "B", "C", "D"), legend.grob = leg, legend = "right")

FeRed_SO4Red_4panel

#ggsave("../figures/Fe_S_manuscript/FeRed_SO4Red_4panel.jpg", plot = FeRed_SO4Red_4panel, device=NULL, path=NULL, scale=1, height = 7, width = 13, limitsize = TRUE, dpi = 600)
```

```{r 3-panel ASR/DSR/FeR}
ASR_DSR_FeR_3panel <- ggarrange(plot4b, plot4c, plot7a, ncol = 3, nrow = 1, labels = c("A", "B", "C"), common.legend = T, legend = "right")

ASR_DSR_FeR_3panel

#ggsave("../figures/Thesis/ASR_DSR_FeR_3panel.png", plot = ASR_DSR_FeR_3panel, device="png", path=NULL, scale=1, height = 5, width = 20, limitsize = F, dpi = 300, bg = "white")
```

### More stats (PERMANOVA and Mantel)

```{r PERMANOVA on Finish only gene sets}
# this chunk is down here because *RA_FIN_design data frames were made in PCoAs. This chunk, and the creation of *RA_FIN_design data frames may be re positioned later.

# Denit final:
adonis_denit_FIN = adonis2(denitRA_FIN_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = denitRA_FIN_design, perm=10000, set.seed=42)

adonis_denit_FIN

# Methanogenesis final:
adonis_meth_FIN = adonis2(methRA_FIN_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = methRA_FIN_design, perm=10000, set.seed=42)

adonis_meth_FIN

# Central Carb metab final:
adonis_carb_metab_FIN = adonis2(carb_metabRA_FIN_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = carb_metabRA_FIN_design, perm=10000, set.seed=42)

adonis_carb_metab_FIN

# CcO final:
adonis_cytoCox_FIN = adonis2(cytoCoxRA_FIN_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = cytoCoxRA_FIN_design, perm=10000, set.seed=42)

adonis_cytoCox_FIN

# Sulfate-S assim final:
adonis_SO4_S_Assim_FIN = adonis2(SO4_S_AssimRA_FIN_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = SO4_S_AssimRA_FIN_design, perm=10000, set.seed=42)

adonis_SO4_S_Assim_FIN

# Assimilatory sulfate reduction final:
adonis_Assim_SO4_red_FIN = adonis2(Assim_SO4_redRA_FIN_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = Assim_SO4_redRA_FIN_design, perm=10000, set.seed=42)

adonis_Assim_SO4_red_FIN

# Dissimilatory sulfate reduction final:
adonis_Dissim_SO4_red_FIN = adonis2(Dissim_SO4_redRA_FIN_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = Dissim_SO4_redRA_FIN_design, perm=10000, set.seed=42)

adonis_Dissim_SO4_red_FIN

# Thiosulfate oxidaiton by SOX complex final:
adonis_ThioSOX_FIN = adonis2(ThioSOXRA_FIN_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = ThioSOXRA_FIN_design, perm=10000, set.seed=42)

adonis_ThioSOX_FIN

# FeGenie (total Fe genes) final:
adonis_FeGenie_FIN = adonis2(FeGenieRA_FIN_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = FeGenieRA_FIN_design, perm=10000, set.seed=42)

adonis_FeGenie_FIN

# Fe reduction final:
adonis_FeReduction_FIN = adonis2(FeReductionRA_FIN_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = FeReductionRA_FIN_design, perm=10000, set.seed=42)

adonis_FeReduction_FIN

# Sulfate reduction (combined) final:
adonis_SO4_red_combined_FIN = adonis2(SO4_red_combinedRA_FIN_design[,-c(1:3)]~Plant*Hist + Plant*Treat, method = "bray", data = SO4_red_combinedRA_FIN_design, perm=10000, set.seed=42)

adonis_SO4_red_combined_FIN
```

```{r Mantel tests}
# Using Pat Schloss code club as reference: https://www.youtube.com/watch?v=EXNOgmUyPfY

# mantel tests:
mantel(FeReductionBC, SO4_red_combinedBC, permutations = 10000, set.seed(42))

mantel(FeReductionBC, Assim_SO4_redBC, permutations = 10000, set.seed(42))

mantel(FeReductionBC, Dissim_SO4_redBC, permutations = 10000, set.seed(42))

mantel(denitBC, ThioSOXBC, permutations = 10000, set.seed(42))

mantel(DNRABC, ThioSOXBC, permutations = 10000, set.seed(42))

mantel(ANRABC, ThioSOXBC, permutations = 10000, set.seed(42))
```

```{r Wrangling for hydrology-specific Mantel Tests}
#Filter FeReduction and SO4 reduction (combined) for wet history only:
SO4_red_combinedRA_WetHist_design <- SO4_red_combinedRA_design %>% dplyr::filter(Hist == "w")

FeReductionRA_WetHist_design <- FeReductionRA_design %>% dplyr::filter(Hist == "w")

#Filter FeReduction and SO4 reduction (combined) for dry history only:
SO4_red_combinedRA_DryHist_design <- SO4_red_combinedRA_design %>% dplyr::filter(Hist == "d")

FeReductionRA_DryHist_design <- FeReductionRA_design %>% dplyr::filter(Hist == "d")

#Filter FeReduction and SO4 reduction (combined) for wet treatment only:
SO4_red_combinedRA_WetTreat_design <- SO4_red_combinedRA_design %>% dplyr::filter(Treat == "w")

FeReductionRA_WetTreat_design <- FeReductionRA_design %>% dplyr::filter(Treat == "w")

#Filter FeReduction and SO4 reduction (combined) for dry treatment only:
SO4_red_combinedRA_DryTreat_design <- SO4_red_combinedRA_design %>% dplyr::filter(Treat == "d")

FeReductionRA_DryTreat_design <- FeReductionRA_design %>% dplyr::filter(Treat == "d")

# Make Bray-Curtis distance matrices for numeric only data in new RA dfs:
SO4_red_combinedRA_WetHist_BC <- SO4_red_combinedRA_WetHist_design[,-c(1:3)] %>%
  vegdist(method = 'bray')

FeReductionRA_WetHist_BC <- FeReductionRA_WetHist_design[,-c(1:3)] %>%
  vegdist(method = 'bray')

SO4_red_combinedRA_DryHist_BC <- SO4_red_combinedRA_DryHist_design[,-c(1:3)] %>%
  vegdist(method = 'bray')

FeReductionRA_DryHist_BC <- FeReductionRA_DryHist_design[,-c(1:3)] %>%
  vegdist(method = 'bray')

SO4_red_combinedRA_WetTreat_BC <- SO4_red_combinedRA_WetTreat_design[,-c(1:3)] %>%
  vegdist(method = 'bray')

FeReductionRA_WetTreat_BC <- FeReductionRA_WetTreat_design[,-c(1:3)] %>%
  vegdist(method = 'bray')

SO4_red_combinedRA_DryTreat_BC <- SO4_red_combinedRA_DryTreat_design[,-c(1:3)] %>%
  vegdist(method = 'bray')

FeReductionRA_DryTreat_BC <- FeReductionRA_DryTreat_design[,-c(1:3)] %>%
  vegdist(method = 'bray')
```

```{r Hydrology-specific Mantel tests}
# Sulfate reduction (combined) and Fe reduction (FeGenie) in Wet history:
mantel(SO4_red_combinedRA_WetHist_BC, FeReductionRA_WetHist_BC, permutations = 10000, set.seed(42))

# Sulfate reduction (combined) and Fe reduction (FeGenie) in Dry history:
mantel(SO4_red_combinedRA_DryHist_BC, FeReductionRA_DryHist_BC, permutations = 10000, set.seed(42))

# Sulfate reduction (combined) and Fe reduction (FeGenie) in Wet treatment:
mantel(SO4_red_combinedRA_WetTreat_BC, FeReductionRA_WetTreat_BC, permutations = 10000, set.seed(42))

# Sulfate reduction (combined) and Fe reduction (FeGenie) in Dry treatment:
mantel(SO4_red_combinedRA_DryTreat_BC, FeReductionRA_DryTreat_BC, permutations = 10000, set.seed(42))
```



```{r Plotting BC correlation}
# These plots were made in https://www.youtube.com/watch?v=EXNOgmUyPfY, but are not required to run Mantel Test

# Make a tibble of bray-curtis matrix
FeReductionBC_tbl <- FeReductionBC %>%
  as.matrix() %>%
  as_tibble(rownames="A") %>%
  pivot_longer(-A, names_to = "B", values_to = "bray_Fe")

SO4_red_combinedBC_tbl <- SO4_red_combinedBC %>%
  as.matrix() %>%
  as_tibble(rownames="A") %>%
  pivot_longer(-A, names_to = "B", values_to = "bray_SO4")

Assim_SO4_redBC_tbl <- Assim_SO4_redBC %>%
  as.matrix() %>%
  as_tibble(rownames="A") %>%
  pivot_longer(-A, names_to = "B", values_to = "bray_Assim_SO4")

# combine the respective tibbles:
combined_BC_tbls <- inner_join(FeReductionBC_tbl, SO4_red_combinedBC_tbl, by=c("A", "B")) # not sure if I need this.

combined_BC_FeRed_AssSO4 <- inner_join(FeReductionBC_tbl, Assim_SO4_redBC_tbl, by=c("A", "B"))

# plotting:
combined_BC_tbls %>%
  filter(A < B) %>%
  ggplot(aes(x=bray_Fe, y=bray_SO4)) +
  geom_point()+
  geom_smooth(se=FALSE)

combined_BC_FeRed_AssSO4 %>%
  filter(A < B) %>%
  ggplot(aes(x=bray_Fe, y=bray_Assim_SO4)) +
  geom_point()+
  geom_smooth(se=FALSE)
```



END